{"cells":[{"cell_type":"markdown","metadata":{"id":"P5k_WL74Y2Ei"},"source":["#$\\text{L-RAP: Long Read RNA Sequencing Analysis Pipeline}$\n"]},{"cell_type":"markdown","metadata":{"id":"IaygDNNEZnph"},"source":["$\\text{A pipeline to analyze Oxford Nanopore and PacBio third-generation long transcriptomic sequencing reads}$"]},{"cell_type":"markdown","metadata":{"id":"hIjUpSQCZlw1"},"source":["$\\text{Theodore Nelson}$\n","\n","$\\text{Columbia University Irving Medical Center}$\n"]},{"cell_type":"markdown","metadata":{"id":"AUKHTlobUcf_"},"source":["##$\\color{#e74b4b}{\\text{Parameter Input and User Instructions}}$"]},{"cell_type":"markdown","metadata":{"id":"oM-NaZDK63YC"},"source":["Please define where the file structure is within your Google Drive:"]},{"cell_type":"markdown","metadata":{"id":"uc1rzY2x6tvu"},"source":["<ul type=disc>\n","<li><b>PIPELINE_FILE_PATH</b>: file path to location of long-read RNA sequencing analysis pipeline within your Google Colab/Google Drive/local file system - required for most applications. This must be defined and initialized first.</li>\n","</ul>"]},{"cell_type":"code","execution_count":97,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"elapsed":238,"status":"ok","timestamp":1661185052032,"user":{"displayName":"Theodore Maximilian Nelson","userId":"03436156380930975060"},"user_tz":240},"id":"_P4OAFNrzifE","outputId":"6014fb72-9f26-4f3a-ec39-969751595a92"},"outputs":[{"output_type":"stream","name":"stdout","text":["env: PIPELINE_FILE_PATH=/content/long-read-sequencing-pipeline\n"]}],"source":["%env PIPELINE_FILE_PATH=/content/long-read-sequencing-pipeline"]},{"cell_type":"code","source":["! mkdir $PIPELINE_FILE_PATH"],"metadata":{"id":"Hpn2CRCkJTfT","executionInfo":{"status":"ok","timestamp":1661174957362,"user_tz":240,"elapsed":9,"user":{"displayName":"Theodore Maximilian Nelson","userId":"03436156380930975060"}}},"execution_count":2,"outputs":[]},{"cell_type":"code","source":["! cd $PIPELINE_FILE_PATH ; git clone https://github.com/Theo-Nelson/long-read-sequencing-pipeline"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"I24mo0O8JF4P","executionInfo":{"status":"ok","timestamp":1661174958283,"user_tz":240,"elapsed":927,"user":{"displayName":"Theodore Maximilian Nelson","userId":"03436156380930975060"}},"outputId":"cdf8d043-9c64-4930-be01-1dc0d3a5e8d4"},"execution_count":3,"outputs":[{"output_type":"stream","name":"stdout","text":["Cloning into 'long-read-sequencing-pipeline'...\n","remote: Enumerating objects: 117, done.\u001b[K\n","remote: Counting objects: 100% (117/117), done.\u001b[K\n","remote: Compressing objects: 100% (103/103), done.\u001b[K\n","remote: Total 117 (delta 60), reused 32 (delta 10), pack-reused 0\u001b[K\n","Receiving objects: 100% (117/117), 102.70 KiB | 7.90 MiB/s, done.\n","Resolving deltas: 100% (60/60), done.\n"]}]},{"cell_type":"markdown","metadata":{"id":"2USxlBU47A0L"},"source":["Please modify the following parameters within the code box below to fit your own study requirements:  "]},{"cell_type":"markdown","metadata":{"id":"f99vaAE0Z0AG"},"source":["<li><b>ACC</b>: Run accession number for reads within the [European Nucleotide Archive](https://www.ebi.ac.uk/ena/browser/) (SRR...) or file path to location of long-read RNA sequencing data within your Google Drive/general file system - required for most applications</li>\n","<li><b>PLATFORM_FREE</b>: choose either `map-ont` (Nanopore reads) or `map-pb` (PacBio reads) - required for minimap2 featherweight version</li>\n","<li><b>INDEX_FILE_PATH</b>: file path to location of reference genome (e.g. .FASTA) within your Google Drive/general file system - required for most applications</li>\n","<li><b>ANNOTATION_FILE_PATH</b>: file path to location of reference annotation (e.g. .GTF) within your Google Drive/general file system - required for most applications</li>\n","<li><b>PARTITIONED_INDEX_FILE_PATH</b>: choose a filename; select a location to place the partioned reference annotation (e.g. .idx) when generated by the pipeline within your Google Drive/general file system - required for minimap2 featherweight version</li>\n","<li><b>CHROMOSOME</b>: Name of Chromosome of Interest matching the name of the Chromosome within your Reference Annotation - required for svist4get</li>\n","<li><b>CHROMOSOME_START</b>: Starting Location of Interest on the Chromosome - required for svist4get</li>\n","<li><b>CHROMOSOME_FINISH</b>: Ending Location of Interest on the Chromosome - required for svist4get</li>\n","<li><b>REGION_NAME</b>: Gene Name (does not need to match annotation file) - required for svist4get</li>\n","<li><b>HUB_KEYWORD</b>: Short Keyword for your UCSC Track Hub - required for MakeHub</li>\n","<li><b>HUB_NAME</b>: Longer Title for your UCSC Track Hub - required for MakeHub</li>\n","<li><b>HUB_EMAIL</b>: Email for your UCSC Track Hub (if you publish your track hub then this email will be public) - required for MakeHub</li>\n"]},{"cell_type":"code","execution_count":98,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"elapsed":260,"status":"ok","timestamp":1661185057331,"user":{"displayName":"Theodore Maximilian Nelson","userId":"03436156380930975060"},"user_tz":240},"id":"tqGM1OFhUcoo","outputId":"b90db905-224f-4a9d-fcf4-fcb16631ac6b"},"outputs":[{"output_type":"stream","name":"stdout","text":["env: ACC=ERR1951293\n","env: PLATFORM_FREE=map-pb\n","env: INDEX_FILE_PATH=${PIPELINE_FILE_PATH}/long-read-sequencing-pipeline/prebuilt_indices/mm39.fa\n","env: ANNOTATION_FILE_PATH=${PIPELINE_FILE_PATH}/long-read-sequencing-pipeline/prebuilt_indices/mm39.ncbiRefSeq.gtf\n","env: PARTITIONED_INDEX_FILE_PATH=${PIPELINE_FILE_PATH}/long-read-sequencing-pipeline/prebuilt_indices/mm39.idx\n","env: CHROMOSOME=NC_000071.7\n","env: CHROMOSOME_START=118511219\n","env: CHROMOSOME_FINISH=118519017\n","env: REGION_NAME=Gm30604\n","env: HUB_KEYWORD=Gm30604\n","env: HUB_NAME=\"Murine Gm30604\"\n","env: HUB_EMAIL=tmn2126@columbia.edu\n"]}],"source":["%env ACC=ERR1951293  \n","%env PLATFORM_FREE=map-pb\n","%env INDEX_FILE_PATH=${PIPELINE_FILE_PATH}/long-read-sequencing-pipeline/prebuilt_indices/mm39.fa\n","%env ANNOTATION_FILE_PATH=${PIPELINE_FILE_PATH}/long-read-sequencing-pipeline/prebuilt_indices/mm39.ncbiRefSeq.gtf\n","%env PARTITIONED_INDEX_FILE_PATH=${PIPELINE_FILE_PATH}/long-read-sequencing-pipeline/prebuilt_indices/mm39.idx\n","%env CHROMOSOME=NC_000071.7\n","%env CHROMOSOME_START=118511219\n","%env CHROMOSOME_FINISH=118519017\n","%env REGION_NAME=Gm30604\n","%env HUB_KEYWORD=Gm30604\n","%env HUB_NAME=\"Murine Gm30604\"\n","%env HUB_EMAIL=tmn2126@columbia.edu"]},{"cell_type":"markdown","metadata":{"id":"RrTcAlHtbWfg"},"source":["##$\\color{#e74b4b}{\\text{Mounting your Google Drive}}$"]},{"cell_type":"markdown","metadata":{"id":"C5_iJ_KkbYt3"},"source":["This step allows for permanent storage of your bioinformatics analysis in Google Drive"]},{"cell_type":"markdown","metadata":{"id":"rbS0P9JcJnnJ"},"source":["<ul type=disc>\n","<li><b>STORAGE_FILE_PATH</b>: file path to location where you would wish to store output of long-read RNA sequencing analysis pipeline - required to export data from Google Colab. </li>\n","</ul>"]},{"cell_type":"code","execution_count":5,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"elapsed":10,"status":"ok","timestamp":1661174959115,"user":{"displayName":"Theodore Maximilian Nelson","userId":"03436156380930975060"},"user_tz":240},"outputId":"9bd9da0e-ab54-4576-a3fa-4b643982d109","id":"igrdHc7NJ3Zr"},"outputs":[{"output_type":"stream","name":"stdout","text":["env: STORAGE_FILE_PATH=/content/drive/MyDrive/long-read-sequencing-pipeline\n"]}],"source":["%env STORAGE_FILE_PATH=/content/drive/MyDrive/long-read-sequencing-pipeline"]},{"cell_type":"code","execution_count":6,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"elapsed":20167,"status":"ok","timestamp":1661174979275,"user":{"displayName":"Theodore Maximilian Nelson","userId":"03436156380930975060"},"user_tz":240},"id":"aQOezcA5ZrFA","outputId":"f4a4ba48-92ca-4f78-8d15-2505bc1f2e55"},"outputs":[{"output_type":"stream","name":"stdout","text":["Mounted at /content/drive\n"]}],"source":["from google.colab import drive\n","drive.mount('/content/drive')"]},{"cell_type":"markdown","metadata":{"id":"24vu4E_8vbKa"},"source":["##$\\color{#e74b4b}{\\text{Managing Software via BioConda}}$"]},{"cell_type":"markdown","metadata":{"id":"lz3I3YU06E6z"},"source":["BioConda is a software environment and package manager, providing acess to over 8,000 different software packages related to bioinformatics (documentation: [BioConda](https://bioconda.github.io/user/install.html) and [Managing Environments via Conda](https://docs.conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html)). "]},{"cell_type":"code","execution_count":7,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"elapsed":14923,"status":"ok","timestamp":1661174995665,"user":{"displayName":"Theodore Maximilian Nelson","userId":"03436156380930975060"},"user_tz":240},"id":"oxEPAO-_wpbz","outputId":"43d98856-428b-4fb7-b322-973f13207539"},"outputs":[{"output_type":"stream","name":"stdout","text":["--2022-08-22 13:29:40--  https://repo.anaconda.com/miniconda/Miniconda3-py37_4.8.2-Linux-x86_64.sh\n","Resolving repo.anaconda.com (repo.anaconda.com)... 104.16.131.3, 104.16.130.3, 2606:4700::6810:8203, ...\n","Connecting to repo.anaconda.com (repo.anaconda.com)|104.16.131.3|:443... connected.\n","HTTP request sent, awaiting response... 200 OK\n","Length: 85055499 (81M) [application/x-sh]\n","Saving to: ‘Miniconda3-py37_4.8.2-Linux-x86_64.sh’\n","\n","Miniconda3-py37_4.8 100%[===================>]  81.12M  94.9MB/s    in 0.9s    \n","\n","2022-08-22 13:29:41 (94.9 MB/s) - ‘Miniconda3-py37_4.8.2-Linux-x86_64.sh’ saved [85055499/85055499]\n","\n","PREFIX=/usr/local\n","Unpacking payload ...\n","Collecting package metadata (current_repodata.json): - \b\b\\ \b\b| \b\bdone\n","Solving environment: - \b\b\\ \b\bdone\n","\n","## Package Plan ##\n","\n","  environment location: /usr/local\n","\n","  added / updated specs:\n","    - _libgcc_mutex==0.1=main\n","    - asn1crypto==1.3.0=py37_0\n","    - ca-certificates==2020.1.1=0\n","    - certifi==2019.11.28=py37_0\n","    - cffi==1.14.0=py37h2e261b9_0\n","    - chardet==3.0.4=py37_1003\n","    - conda-package-handling==1.6.0=py37h7b6447c_0\n","    - conda==4.8.2=py37_0\n","    - cryptography==2.8=py37h1ba5d50_0\n","    - idna==2.8=py37_0\n","    - ld_impl_linux-64==2.33.1=h53a641e_7\n","    - libedit==3.1.20181209=hc058e9b_0\n","    - libffi==3.2.1=hd88cf55_4\n","    - libgcc-ng==9.1.0=hdf63c60_0\n","    - libstdcxx-ng==9.1.0=hdf63c60_0\n","    - ncurses==6.2=he6710b0_0\n","    - openssl==1.1.1d=h7b6447c_4\n","    - pip==20.0.2=py37_1\n","    - pycosat==0.6.3=py37h7b6447c_0\n","    - pycparser==2.19=py37_0\n","    - pyopenssl==19.1.0=py37_0\n","    - pysocks==1.7.1=py37_0\n","    - python==3.7.6=h0371630_2\n","    - readline==7.0=h7b6447c_5\n","    - requests==2.22.0=py37_1\n","    - ruamel_yaml==0.15.87=py37h7b6447c_0\n","    - setuptools==45.2.0=py37_0\n","    - six==1.14.0=py37_0\n","    - sqlite==3.31.1=h7b6447c_0\n","    - tk==8.6.8=hbc83047_0\n","    - tqdm==4.42.1=py_0\n","    - urllib3==1.25.8=py37_0\n","    - wheel==0.34.2=py37_0\n","    - xz==5.2.4=h14c3975_4\n","    - yaml==0.1.7=had09818_2\n","    - zlib==1.2.11=h7b6447c_3\n","\n","\n","The following NEW packages will be INSTALLED:\n","\n","  _libgcc_mutex      pkgs/main/linux-64::_libgcc_mutex-0.1-main\n","  asn1crypto         pkgs/main/linux-64::asn1crypto-1.3.0-py37_0\n","  ca-certificates    pkgs/main/linux-64::ca-certificates-2020.1.1-0\n","  certifi            pkgs/main/linux-64::certifi-2019.11.28-py37_0\n","  cffi               pkgs/main/linux-64::cffi-1.14.0-py37h2e261b9_0\n","  chardet            pkgs/main/linux-64::chardet-3.0.4-py37_1003\n","  conda              pkgs/main/linux-64::conda-4.8.2-py37_0\n","  conda-package-han~ pkgs/main/linux-64::conda-package-handling-1.6.0-py37h7b6447c_0\n","  cryptography       pkgs/main/linux-64::cryptography-2.8-py37h1ba5d50_0\n","  idna               pkgs/main/linux-64::idna-2.8-py37_0\n","  ld_impl_linux-64   pkgs/main/linux-64::ld_impl_linux-64-2.33.1-h53a641e_7\n","  libedit            pkgs/main/linux-64::libedit-3.1.20181209-hc058e9b_0\n","  libffi             pkgs/main/linux-64::libffi-3.2.1-hd88cf55_4\n","  libgcc-ng          pkgs/main/linux-64::libgcc-ng-9.1.0-hdf63c60_0\n","  libstdcxx-ng       pkgs/main/linux-64::libstdcxx-ng-9.1.0-hdf63c60_0\n","  ncurses            pkgs/main/linux-64::ncurses-6.2-he6710b0_0\n","  openssl            pkgs/main/linux-64::openssl-1.1.1d-h7b6447c_4\n","  pip                pkgs/main/linux-64::pip-20.0.2-py37_1\n","  pycosat            pkgs/main/linux-64::pycosat-0.6.3-py37h7b6447c_0\n","  pycparser          pkgs/main/linux-64::pycparser-2.19-py37_0\n","  pyopenssl          pkgs/main/linux-64::pyopenssl-19.1.0-py37_0\n","  pysocks            pkgs/main/linux-64::pysocks-1.7.1-py37_0\n","  python             pkgs/main/linux-64::python-3.7.6-h0371630_2\n","  readline           pkgs/main/linux-64::readline-7.0-h7b6447c_5\n","  requests           pkgs/main/linux-64::requests-2.22.0-py37_1\n","  ruamel_yaml        pkgs/main/linux-64::ruamel_yaml-0.15.87-py37h7b6447c_0\n","  setuptools         pkgs/main/linux-64::setuptools-45.2.0-py37_0\n","  six                pkgs/main/linux-64::six-1.14.0-py37_0\n","  sqlite             pkgs/main/linux-64::sqlite-3.31.1-h7b6447c_0\n","  tk                 pkgs/main/linux-64::tk-8.6.8-hbc83047_0\n","  tqdm               pkgs/main/noarch::tqdm-4.42.1-py_0\n","  urllib3            pkgs/main/linux-64::urllib3-1.25.8-py37_0\n","  wheel              pkgs/main/linux-64::wheel-0.34.2-py37_0\n","  xz                 pkgs/main/linux-64::xz-5.2.4-h14c3975_4\n","  yaml               pkgs/main/linux-64::yaml-0.1.7-had09818_2\n","  zlib               pkgs/main/linux-64::zlib-1.2.11-h7b6447c_3\n","\n","\n","Preparing transaction: / \b\b- \b\b\\ \b\b| \b\bdone\n","Executing transaction: - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\bdone\n","installation finished.\n","WARNING:\n","    You currently have a PYTHONPATH environment variable set. This may cause\n","    unexpected behavior when running the Python interpreter in Miniconda3.\n","    For best results, please verify that your PYTHONPATH only points to\n","    directories of packages that are compatible with the Python interpreter\n","    in Miniconda3: /usr/local\n"]}],"source":["! wget https://repo.anaconda.com/miniconda/Miniconda3-py37_4.8.2-Linux-x86_64.sh\n","! chmod +x Miniconda3-py37_4.8.2-Linux-x86_64.sh\n","! bash ./Miniconda3-py37_4.8.2-Linux-x86_64.sh -b -f -p /usr/local\n","import sys\n","sys.path.append('/usr/local/lib/python3.7/site-packages/')"]},{"cell_type":"markdown","metadata":{"id":"z1lysGn4hO-i"},"source":["##$\\color{#ff00d5}{\\text{Kingfisher: procurement of sequence files - installation}}$"]},{"cell_type":"markdown","metadata":{"id":"X4iuuuqPkKcT"},"source":["The Kingfisher program allows for sequence files to be downloaded from the European Nucleotide Archive (documentation: https://github.com/wwood/kingfisher-download)."]},{"cell_type":"code","execution_count":8,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"elapsed":2473,"status":"ok","timestamp":1661175003671,"user":{"displayName":"Theodore Maximilian Nelson","userId":"03436156380930975060"},"user_tz":240},"id":"_Kn0KYlyOi2r","outputId":"a7d81380-2644-4507-b1a2-7e9a07546a9c"},"outputs":[{"output_type":"stream","name":"stdout","text":["Cloning into 'kingfisher-download'...\n","remote: Enumerating objects: 696, done.\u001b[K\n","remote: Counting objects: 100% (69/69), done.\u001b[K\n","remote: Compressing objects: 100% (19/19), done.\u001b[K\n","remote: Total 696 (delta 60), reused 50 (delta 50), pack-reused 627\u001b[K\n","Receiving objects: 100% (696/696), 8.36 MiB | 5.48 MiB/s, done.\n","Resolving deltas: 100% (396/396), done.\n"]}],"source":["! git clone https://github.com/MakeTheBrainHappy/kingfisher-download"]},{"cell_type":"code","execution_count":9,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"elapsed":271600,"status":"ok","timestamp":1661175275266,"user":{"displayName":"Theodore Maximilian Nelson","userId":"03436156380930975060"},"user_tz":240},"id":"NNtGMks8OkmZ","outputId":"9e7b3ca6-e71b-4158-d5f5-4a208656a4d3"},"outputs":[{"output_type":"stream","name":"stdout","text":["Collecting package metadata (repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\bdone\n","Solving environment: / \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\bdone\n","\n","\n","==> WARNING: A newer version of conda exists. <==\n","  current version: 4.8.2\n","  latest version: 4.13.0\n","\n","Please update conda by running\n","\n","    $ conda update -n base -c defaults conda\n","\n","\n","\n","Downloading and Extracting Packages\n","pandas-1.4.2         | 12.5 MB   | : 100% 1.0/1 [00:03<00:00, 52.00s/it]               \n","libgfortran5-12.1.0  | 1.8 MB    | : 100% 1.0/1 [00:00<00:00,  2.35it/s]\n","openssl-1.1.1o       | 2.1 MB    | : 100% 1.0/1 [00:00<00:00,  2.08it/s]               \n","conda-4.14.0         | 1011 KB   | : 100% 1.0/1 [00:00<00:00,  2.54it/s]\n","libblas-3.9.0        | 12 KB     | : 100% 1.0/1 [00:00<00:00, 20.46it/s]\n","ca-certificates-2022 | 149 KB    | : 100% 1.0/1 [00:00<00:00, 15.85it/s]\n","pysocks-1.7.1        | 28 KB     | : 100% 1.0/1 [00:00<00:00, 17.13it/s]\n","sracat-0.2           | 1.5 MB    | : 100% 1.0/1 [00:00<00:00,  1.62it/s]              \n","certifi-2022.6.15    | 155 KB    | : 100% 1.0/1 [00:00<00:00, 13.78it/s]\n","cffi-1.14.4          | 227 KB    | : 100% 1.0/1 [00:00<00:00,  1.58s/it]                \n","c-ares-1.18.1        | 113 KB    | : 100% 1.0/1 [00:00<00:00, 18.57it/s]\n","setuptools-65.2.0    | 1.4 MB    | : 100% 1.0/1 [00:00<00:00,  2.17it/s]\n","pip-20.2             | 1.1 MB    | : 100% 1.0/1 [00:00<00:00,  4.17s/it]               \n","libstdcxx-ng-12.1.0  | 4.3 MB    | : 100% 1.0/1 [00:00<00:00,  1.25it/s]\n","libxml2-2.9.12       | 772 KB    | : 100% 1.0/1 [00:00<00:00,  4.56it/s]\n","colorama-0.4.4       | 18 KB     | : 100% 1.0/1 [00:00<00:00, 25.58it/s]\n","idna-3.3             | 55 KB     | : 100% 1.0/1 [00:00<00:00, 19.77it/s]\n","curl-7.76.1          | 149 KB    | : 100% 1.0/1 [00:00<00:00, 15.57it/s]\n","bird_tool_utils_pyth | 27 KB     | : 100% 1.0/1 [00:00<00:00,  6.65it/s]               \n","libssh2-1.10.0       | 233 KB    | : 100% 1.0/1 [00:00<00:00,  8.38it/s]\n","pycosat-0.6.3        | 108 KB    | : 100% 1.0/1 [00:00<00:00, 14.18it/s]\n","brotlipy-0.7.0       | 342 KB    | : 100% 1.0/1 [00:00<00:00, 11.32it/s]\n","libgfortran-ng-12.1. | 23 KB     | : 100% 1.0/1 [00:00<00:00, 12.63it/s]\n","tzdata-2022c         | 119 KB    | : 100% 1.0/1 [00:00<00:00,  7.49it/s]\n","awscli-1.25.34       | 1.9 MB    | : 100% 1.0/1 [00:01<00:00, 10.18s/it]               \n","jmespath-1.0.1       | 21 KB     | : 100% 1.0/1 [00:00<00:00, 13.11it/s]\n","numpy-1.22.3         | 6.8 MB    | : 100% 1.0/1 [00:01<00:00,  1.02s/it]                \n","libcblas-3.9.0       | 12 KB     | : 100% 1.0/1 [00:00<00:00, 14.29it/s]\n","pycparser-2.21       | 100 KB    | : 100% 1.0/1 [00:00<00:00,  8.26it/s]\n","cryptography-37.0.2  | 1.5 MB    | : 100% 1.0/1 [00:00<00:00,  8.66s/it]                 \n","urllib3-1.26.11      | 102 KB    | : 100% 1.0/1 [00:00<00:00, 18.10it/s]\n","_openmp_mutex-5.1    | 21 KB     | : 100% 1.0/1 [00:00<00:00,  5.20it/s]\n","pigz-2.6             | 87 KB     | : 100% 1.0/1 [00:00<00:00,  6.01it/s]               \n","sqlite-3.36.0        | 1.4 MB    | : 100% 1.0/1 [00:00<00:00,  1.48it/s]               \n","icu-68.2             | 13.1 MB   | : 100% 1.0/1 [00:02<00:00,  1.52s/it]               \n","nose-1.3.7           | 118 KB    | : 100% 1.0/1 [00:00<00:00, 15.67it/s]\n","libopenblas-0.3.20   | 10.1 MB   | : 100% 1.0/1 [00:02<00:00,  1.57s/it]               \n","extern-0.4.1         | 8 KB      | : 100% 1.0/1 [00:00<00:00,  8.35it/s]\n","libev-4.33           | 104 KB    | : 100% 1.0/1 [00:00<00:00, 12.23it/s]\n","python-dateutil-2.8. | 240 KB    | : 100% 1.0/1 [00:00<00:00, 15.49it/s]\n","yaml-0.2.5           | 87 KB     | : 100% 1.0/1 [00:00<00:00, 22.23it/s]\n","krb5-1.17.1          | 1.5 MB    | : 100% 1.0/1 [00:00<00:00, 12.21s/it]                 \n","tk-8.6.11            | 3.3 MB    | : 100% 1.0/1 [00:00<00:00,  1.48it/s]\n","wheel-0.37.1         | 31 KB     | : 100% 1.0/1 [00:00<00:00, 21.65it/s]\n","conda-package-handli | 1.0 MB    | : 100% 1.0/1 [00:00<00:00,  5.66it/s]\n","sra-tools-2.8.0      | 99.7 MB   | : 100% 1.0/1 [00:20<00:00, 64.57s/it]               \n","pyyaml-5.4.1         | 196 KB    | : 100% 1.0/1 [00:00<00:00,  2.64it/s]               \n","rsa-4.7.2            | 28 KB     | : 100% 1.0/1 [00:00<00:00, 15.86it/s]\n","argparse-manpage-bir | 16 KB     | : 100% 1.0/1 [00:00<00:00,  7.18it/s]\n","python_abi-3.9       | 4 KB      | : 100% 1.0/1 [00:00<00:00, 16.73it/s]\n","ruamel_yaml-0.15.80  | 268 KB    | : 100% 1.0/1 [00:00<00:00,  4.38it/s]               \n","pyasn1-0.4.8         | 53 KB     | : 100% 1.0/1 [00:00<00:00, 14.72it/s]\n","botocore-1.27.34     | 5.5 MB    | : 100% 1.0/1 [00:03<00:00,  1.63s/it]               \n","toolz-0.12.0         | 48 KB     | : 100% 1.0/1 [00:00<00:00, 19.96it/s]\n","liblapack-3.9.0      | 12 KB     | : 100% 1.0/1 [00:00<00:00, 19.36it/s]\n","aria2-1.36.0         | 1.5 MB    | : 100% 1.0/1 [00:00<00:00,  9.34s/it]                \n","pytz-2022.2.1        | 224 KB    | : 100% 1.0/1 [00:00<00:00,  7.25it/s]\n","libnghttp2-1.43.0    | 790 KB    | : 100% 1.0/1 [00:00<00:00,  4.03it/s]\n","six-1.16.0           | 14 KB     | : 100% 1.0/1 [00:00<00:00, 20.85it/s]\n","docutils-0.16        | 737 KB    | : 100% 1.0/1 [00:00<00:00,  2.99s/it]               \n","readline-8.1         | 295 KB    | : 100% 1.0/1 [00:00<00:00, 10.85it/s]\n","xz-5.2.5             | 343 KB    | : 100% 1.0/1 [00:00<00:00,  9.26it/s]\n","python-3.9.0         | 24.0 MB   | : 100% 1.0/1 [00:06<00:00,  6.04s/it]               \n","libgomp-11.2.0       | 474 KB    | : 100% 1.0/1 [00:00<00:00,  2.39it/s]                \n","pyopenssl-22.0.0     | 49 KB     | : 100% 1.0/1 [00:00<00:00, 16.02it/s]\n","requests-2.28.1      | 53 KB     | : 100% 1.0/1 [00:00<00:00, 12.50it/s]\n","charset-normalizer-2 | 36 KB     | : 100% 1.0/1 [00:00<00:00, 19.83it/s]\n","s3transfer-0.6.0     | 55 KB     | : 100% 1.0/1 [00:00<00:00, 16.27it/s]\n","libgcc-ng-11.2.0     | 5.3 MB    | : 100% 1.0/1 [00:00<00:00,  3.92it/s]\n","libiconv-1.16        | 1.4 MB    | : 100% 1.0/1 [00:00<00:00,  4.41it/s]\n","libcurl-7.76.1       | 328 KB    | : 100% 1.0/1 [00:00<00:00, 11.26it/s]\n","Preparing transaction: | \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\bdone\n","Verifying transaction: | \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\bdone\n","Executing transaction: - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\bdone\n","#\n","# To activate this environment, use\n","#\n","#     $ conda activate base\n","#\n","# To deactivate an active environment, use\n","#\n","#     $ conda deactivate\n","\n"]}],"source":["! conda env update -n base --file kingfisher-download/kingfisher.yml"]},{"cell_type":"code","execution_count":10,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"elapsed":18386,"status":"ok","timestamp":1661175293645,"user":{"displayName":"Theodore Maximilian Nelson","userId":"03436156380930975060"},"user_tz":240},"id":"fDjCnO4C3NQu","outputId":"a0f0dadf-425a-4dee-c48e-851acb7be089"},"outputs":[{"output_type":"stream","name":"stdout","text":["Collecting package metadata (current_repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\bdone\n","Solving environment: / \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\bdone\n","\n","## Package Plan ##\n","\n","  environment location: /usr/local\n","\n","  added / updated specs:\n","    - aspera-connect\n","\n","\n","The following packages will be downloaded:\n","\n","    package                    |            build\n","    ---------------------------|-----------------\n","    aspera-connect-3.9.6       |                0        34.3 MB  rpetit3\n","    ca-certificates-2022.07.19 |       h06a4308_0         124 KB\n","    certifi-2022.6.15          |   py39h06a4308_0         153 KB\n","    openssl-1.1.1q             |       h7f8727e_0         2.5 MB\n","    ------------------------------------------------------------\n","                                           Total:        37.1 MB\n","\n","The following NEW packages will be INSTALLED:\n","\n","  aspera-connect     rpetit3/linux-64::aspera-connect-3.9.6-0\n","\n","The following packages will be UPDATED:\n","\n","  ca-certificates    conda-forge::ca-certificates-2022.6.1~ --> pkgs/main::ca-certificates-2022.07.19-h06a4308_0\n","  openssl            conda-forge::openssl-1.1.1o-h166bdaf_0 --> pkgs/main::openssl-1.1.1q-h7f8727e_0\n","\n","The following packages will be SUPERSEDED by a higher-priority channel:\n","\n","  certifi            conda-forge::certifi-2022.6.15-py39hf~ --> pkgs/main::certifi-2022.6.15-py39h06a4308_0\n","\n","\n","\n","Downloading and Extracting Packages\n","ca-certificates-2022 | 124 KB    | : 100% 1.0/1 [00:00<00:00,  8.45it/s]\n","openssl-1.1.1q       | 2.5 MB    | : 100% 1.0/1 [00:00<00:00,  4.35it/s]\n","certifi-2022.6.15    | 153 KB    | : 100% 1.0/1 [00:00<00:00, 11.78it/s]\n","aspera-connect-3.9.6 | 34.3 MB   | : 100% 1.0/1 [00:08<00:00,  8.44s/it]\n","Preparing transaction: \\ \b\bdone\n","Verifying transaction: / \b\b- \b\bdone\n","Executing transaction: | \b\bdone\n","Retrieving notices: ...working... done\n"]}],"source":["! conda install -c rpetit3 aspera-connect -y"]},{"cell_type":"code","execution_count":11,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"elapsed":3842,"status":"ok","timestamp":1661175297481,"user":{"displayName":"Theodore Maximilian Nelson","userId":"03436156380930975060"},"user_tz":240},"id":"ksJvQOCtx8BJ","outputId":"7f1e9426-074e-486f-b7e3-916bf8db2755"},"outputs":[{"output_type":"stream","name":"stdout","text":["ibm-aspera-connect-3.9.8.176272-linux-g2.12-64.sh\n"]}],"source":["! wget -qO- https://download.asperasoft.com/download/sw/connect/3.9.8/ibm-aspera-connect-3.9.8.176272-linux-g2.12-64.tar.gz | tar xvz"]},{"cell_type":"markdown","metadata":{"id":"RHjWyX6Ei_jT"},"source":["this command will pop up with an error message; please disregard"]},{"cell_type":"code","execution_count":12,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"elapsed":988,"status":"ok","timestamp":1661175298463,"user":{"displayName":"Theodore Maximilian Nelson","userId":"03436156380930975060"},"user_tz":240},"id":"anXSgAYa3xQU","outputId":"d6db6a06-2682-4a1d-bd2f-06e40cc81f87"},"outputs":[{"output_type":"stream","name":"stdout","text":["\n","Installing IBM Aspera Connect\n","\n","This script cannot be run as root, IBM Aspera Connect must be installed per user.\n"]}],"source":["! ./ibm-aspera-connect-3.9.8.176272-linux-g2.12-64.sh"]},{"cell_type":"markdown","metadata":{"id":"oBdBA_CDdE0z"},"source":["##$\\color{#d42bb4}{\\text{Kingfisher: procurement of sequence files - usage}}$\n"]},{"cell_type":"markdown","metadata":{"id":"IE4T9swGdQY-"},"source":["The Kingfisher program allows for sequence files to be downloaded from the European Nucleotide Archive (documentation: https://github.com/wwood/kingfisher-download)."]},{"cell_type":"code","execution_count":21,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"elapsed":20214,"status":"ok","timestamp":1661175489445,"user":{"displayName":"Theodore Maximilian Nelson","userId":"03436156380930975060"},"user_tz":240},"id":"ReZdoTfnd5uQ","outputId":"ffe81aa3-d967-4547-d2aa-ea904803ec01"},"outputs":[{"output_type":"stream","name":"stdout","text":["08/22/2022 01:37:49 PM INFO: Kingfisher v0.0.1-dev\n","08/22/2022 01:37:49 PM INFO: Attempting download method ena-ascp for run ERR1951293 ..\n","08/22/2022 01:37:49 PM INFO: Using aspera ssh key file: $HOME/.aspera/connect/etc/asperaweb_id_dsa.openssh\n","08/22/2022 01:37:49 PM INFO: Querying ENA for FTP paths for ERR1951293..\n","08/22/2022 01:37:52 PM INFO: Downloading 1 FTP read set(s): ftp.sra.ebi.ac.uk/vol1/fastq/ERR195/003/ERR1951293/ERR1951293.fastq.gz\n","08/22/2022 01:37:52 PM INFO: Running command: ascp -T -l 300m -P33001 -k 2 -i $HOME/.aspera/connect/etc/asperaweb_id_dsa.openssh era-fasp@fasp.sra.ebi.ac.uk:/vol1/fastq/ERR195/003/ERR1951293/ERR1951293.fastq.gz .\n","08/22/2022 01:38:08 PM INFO: Method ena-ascp worked.\n","08/22/2022 01:38:08 PM INFO: Output files: ./ERR1951293.fastq.gz\n","08/22/2022 01:38:08 PM INFO: Kingfisher done.\n"]}],"source":["! cd $PIPELINE_FILE_PATH/long-read-sequencing-pipeline/fastq ; /content/kingfisher-download/bin/kingfisher get -r $ACC -m ena-ascp aws-http prefetch"]},{"cell_type":"code","execution_count":22,"metadata":{"id":"wwpiZ9OU6WPn","executionInfo":{"status":"ok","timestamp":1661175496397,"user_tz":240,"elapsed":6960,"user":{"displayName":"Theodore Maximilian Nelson","userId":"03436156380930975060"}}},"outputs":[],"source":["! cd $PIPELINE_FILE_PATH/long-read-sequencing-pipeline/fastq ; gunzip *.gz"]},{"cell_type":"markdown","metadata":{"id":"izrcNnG0REyD"},"source":["The next two commands simply standardize the file name away from uncommon variants provided by depositors in the European Nucleotide Archive. In rare instances pipeline users may need to directly manipulate these commands to standardize the filename. Please do not be concerned if either command throws an error. "]},{"cell_type":"code","execution_count":23,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"elapsed":20,"status":"ok","timestamp":1661175496398,"user":{"displayName":"Theodore Maximilian Nelson","userId":"03436156380930975060"},"user_tz":240},"id":"4pmkDR2x6SzT","outputId":"ce4365ac-bd1c-4d85-8c05-5a549a98ea69"},"outputs":[{"output_type":"stream","name":"stdout","text":["mv: cannot stat 'ERR1951293_1.fastq': No such file or directory\n"]}],"source":["! cd $PIPELINE_FILE_PATH/long-read-sequencing-pipeline/fastq ; mv ${ACC}_1.fastq $ACC.fastq"]},{"cell_type":"code","execution_count":24,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"elapsed":15,"status":"ok","timestamp":1661175496400,"user":{"displayName":"Theodore Maximilian Nelson","userId":"03436156380930975060"},"user_tz":240},"id":"5_r7Ed86Q-cI","outputId":"10485c8a-f5fe-4a08-d1ca-4a3b043f61e0"},"outputs":[{"output_type":"stream","name":"stdout","text":["mv: cannot stat 'ERR1951293_subreads.fastq': No such file or directory\n"]}],"source":["! cd $PIPELINE_FILE_PATH/long-read-sequencing-pipeline/fastq ; mv ${ACC}_subreads.fastq $ACC.fastq"]},{"cell_type":"markdown","source":["##$\\color{#a7588f}{\\text{Kingfisher: procurement of sequence files - export}}$\n"],"metadata":{"id":"_MislmkXZcz-"}},{"cell_type":"markdown","source":["To Store Resulting Files in your Google Drive: "],"metadata":{"id":"-mPcrcf-Kmfl"}},{"cell_type":"code","source":["! cp $PIPELINE_FILE_PATH/long-read-sequencing-pipeline/fastq/$ACC.fastq $STORAGE_FILE_PATH/$ACC.fastq"],"metadata":{"id":"U0pvTRDjKtqG","executionInfo":{"status":"ok","timestamp":1661175536383,"user_tz":240,"elapsed":9128,"user":{"displayName":"Theodore Maximilian Nelson","userId":"03436156380930975060"}}},"execution_count":25,"outputs":[]},{"cell_type":"markdown","source":["To Store Resulting Files in your Local Hard Drive: "],"metadata":{"id":"txLoObMrKrnJ"}},{"cell_type":"code","source":["! export $ACC\n","! export $PIPELINE_FILE_PATH"],"metadata":{"id":"DIyaWe9PPDJi","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1661175539828,"user_tz":240,"elapsed":555,"user":{"displayName":"Theodore Maximilian Nelson","userId":"03436156380930975060"}},"outputId":"3224eb54-89b4-42f3-fed6-9a16f3199997"},"execution_count":26,"outputs":[{"output_type":"stream","name":"stdout","text":["/bin/bash: line 0: export: `/content/long-read-sequencing-pipeline': not a valid identifier\n"]}]},{"cell_type":"code","source":["from google.colab import files\n","import os\n","\n","files.download(\"\".join([os.environ[\"PIPELINE_FILE_PATH\"],\"/long-read-sequencing-pipeline/fastq/\",os.environ[\"ACC\"],\".fastq\"]))"],"metadata":{"colab":{"base_uri":"https://localhost:8080/","height":17},"id":"sWZhEEl3KuIu","executionInfo":{"status":"ok","timestamp":1661175558971,"user_tz":240,"elapsed":366,"user":{"displayName":"Theodore Maximilian Nelson","userId":"03436156380930975060"}},"outputId":"eb9a2692-c38d-404f-a3c9-06333f3e1799"},"execution_count":27,"outputs":[{"output_type":"display_data","data":{"text/plain":["<IPython.core.display.Javascript object>"],"application/javascript":["\n","    async function download(id, filename, size) {\n","      if (!google.colab.kernel.accessAllowed) {\n","        return;\n","      }\n","      const div = document.createElement('div');\n","      const label = document.createElement('label');\n","      label.textContent = `Downloading \"${filename}\": `;\n","      div.appendChild(label);\n","      const progress = document.createElement('progress');\n","      progress.max = size;\n","      div.appendChild(progress);\n","      document.body.appendChild(div);\n","\n","      const buffers = [];\n","      let downloaded = 0;\n","\n","      const channel = await google.colab.kernel.comms.open(id);\n","      // Send a message to notify the kernel that we're ready.\n","      channel.send({})\n","\n","      for await (const message of channel.messages) {\n","        // Send a message to notify the kernel that we're ready.\n","        channel.send({})\n","        if (message.buffers) {\n","          for (const buffer of message.buffers) {\n","            buffers.push(buffer);\n","            downloaded += buffer.byteLength;\n","            progress.value = downloaded;\n","          }\n","        }\n","      }\n","      const blob = new Blob(buffers, {type: 'application/binary'});\n","      const a = document.createElement('a');\n","      a.href = window.URL.createObjectURL(blob);\n","      a.download = filename;\n","      div.appendChild(a);\n","      a.click();\n","      div.remove();\n","    }\n","  "]},"metadata":{}},{"output_type":"display_data","data":{"text/plain":["<IPython.core.display.Javascript object>"],"application/javascript":["download(\"download_c54f445c-73a6-463b-8b49-dad8ba7c3936\", \"ERR1951293.fastq\", 589549137)"]},"metadata":{}}]},{"cell_type":"markdown","metadata":{"id":"F0hFNwmcv5qr"},"source":["##$\\color{#ff00d5}{\\text{FastQC: quality control tool for high throughput sequence data - installation}}$"]},{"cell_type":"markdown","metadata":{"id":"68fqL9EGNOKE"},"source":["FastQC is a program designed to spot potential problems in high througput sequencing datasets ([documentation](https://github.com/s-andrews/FastQC))."]},{"cell_type":"code","execution_count":28,"metadata":{"id":"_bRfEf9e3Sp-","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1661176061434,"user_tz":240,"elapsed":19395,"user":{"displayName":"Theodore Maximilian Nelson","userId":"03436156380930975060"}},"outputId":"dd430216-7a9c-4eae-b962-00716292a484"},"outputs":[{"output_type":"stream","name":"stdout","text":["Collecting package metadata (current_repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\bdone\n","Solving environment: - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\bdone\n","\n","## Package Plan ##\n","\n","  environment location: /usr/local\n","\n","  added / updated specs:\n","    - fastqc\n","\n","\n","The following packages will be downloaded:\n","\n","    package                    |            build\n","    ---------------------------|-----------------\n","    fastqc-0.11.9              |       hdfd78af_1         9.7 MB  bioconda\n","    font-ttf-dejavu-sans-mono-2.37|       hd3eb1b0_0         335 KB\n","    fontconfig-2.13.1          |       h6c09931_0         250 KB\n","    freetype-2.11.0            |       h70c0345_0         618 KB\n","    libpng-1.6.37              |       hbc83047_0         278 KB\n","    libuuid-1.0.3              |       h7f8727e_2          17 KB\n","    openjdk-8.0.152            |       h7b6447c_3        57.4 MB\n","    perl-5.26.2                |       h14c3975_0        10.5 MB\n","    ------------------------------------------------------------\n","                                           Total:        79.0 MB\n","\n","The following NEW packages will be INSTALLED:\n","\n","  fastqc             bioconda/noarch::fastqc-0.11.9-hdfd78af_1\n","  font-ttf-dejavu-s~ pkgs/main/noarch::font-ttf-dejavu-sans-mono-2.37-hd3eb1b0_0\n","  fontconfig         pkgs/main/linux-64::fontconfig-2.13.1-h6c09931_0\n","  freetype           pkgs/main/linux-64::freetype-2.11.0-h70c0345_0\n","  libpng             pkgs/main/linux-64::libpng-1.6.37-hbc83047_0\n","  libuuid            pkgs/main/linux-64::libuuid-1.0.3-h7f8727e_2\n","  openjdk            pkgs/main/linux-64::openjdk-8.0.152-h7b6447c_3\n","  perl               pkgs/main/linux-64::perl-5.26.2-h14c3975_0\n","\n","\n","\n","Downloading and Extracting Packages\n","perl-5.26.2          | 10.5 MB   | : 100% 1.0/1 [00:00<00:00,  1.16it/s]\n","fontconfig-2.13.1    | 250 KB    | : 100% 1.0/1 [00:00<00:00, 12.08it/s]\n","fastqc-0.11.9        | 9.7 MB    | : 100% 1.0/1 [00:01<00:00,  1.43s/it]               \n","libpng-1.6.37        | 278 KB    | : 100% 1.0/1 [00:00<00:00, 12.16it/s]\n","freetype-2.11.0      | 618 KB    | : 100% 1.0/1 [00:00<00:00,  9.02it/s]\n","libuuid-1.0.3        | 17 KB     | : 100% 1.0/1 [00:00<00:00, 12.21it/s]\n","font-ttf-dejavu-sans | 335 KB    | : 100% 1.0/1 [00:00<00:00, 12.30it/s]\n","openjdk-8.0.152      | 57.4 MB   | : 100% 1.0/1 [00:02<00:00,  2.66s/it]\n","Preparing transaction: - \b\b\\ \b\bdone\n","Verifying transaction: / \b\b- \b\b\\ \b\b| \b\b/ \b\bdone\n","Executing transaction: \\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\bdone\n","Retrieving notices: ...working... done\n"]}],"source":["! conda install -c bioconda fastqc -y"]},{"cell_type":"markdown","metadata":{"id":"SQb5HTiwtfMn"},"source":["##$\\color{#d42bb4}{\\text{FastQC: quality control tool for high throughput sequence data - usage}}$"]},{"cell_type":"markdown","metadata":{"id":"ZoutKrvVtjAY"},"source":["FastQC is a program designed to spot potential problems in high througput sequencing datasets ([documentation](https://github.com/s-andrews/FastQC))."]},{"cell_type":"code","execution_count":29,"metadata":{"id":"bTYA5m273eVN","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1661176080765,"user_tz":240,"elapsed":19338,"user":{"displayName":"Theodore Maximilian Nelson","userId":"03436156380930975060"}},"outputId":"0d481730-3d46-4b62-9625-446461b2d9bc"},"outputs":[{"output_type":"stream","name":"stdout","text":["Started analysis of ERR1951293.fastq\n","Approx 5% complete for ERR1951293.fastq\n","Approx 10% complete for ERR1951293.fastq\n","Approx 15% complete for ERR1951293.fastq\n","Approx 20% complete for ERR1951293.fastq\n","Approx 25% complete for ERR1951293.fastq\n","Approx 30% complete for ERR1951293.fastq\n","Approx 35% complete for ERR1951293.fastq\n","Approx 40% complete for ERR1951293.fastq\n","Approx 45% complete for ERR1951293.fastq\n","Approx 50% complete for ERR1951293.fastq\n","Approx 55% complete for ERR1951293.fastq\n","Approx 60% complete for ERR1951293.fastq\n","Approx 65% complete for ERR1951293.fastq\n","Approx 70% complete for ERR1951293.fastq\n","Approx 75% complete for ERR1951293.fastq\n","Approx 80% complete for ERR1951293.fastq\n","Approx 85% complete for ERR1951293.fastq\n","Approx 90% complete for ERR1951293.fastq\n","Approx 95% complete for ERR1951293.fastq\n","Analysis complete for ERR1951293.fastq\n"]}],"source":["! fastqc $PIPELINE_FILE_PATH/long-read-sequencing-pipeline/fastq/$ACC.fastq --outdir $PIPELINE_FILE_PATH/long-read-sequencing-pipeline/fastqc"]},{"cell_type":"markdown","metadata":{"id":"nHGHIAKCSUen"},"source":["For the best viewing experience, please down the HTML output to your Hard Drive and open in your local browser, such as Google Chrome, Firefox or Edge. "]},{"cell_type":"markdown","source":["##$\\color{#a7588f}{\\text{FastQC: quality control tool for high throughput sequence data - export}}$\n"],"metadata":{"id":"NOrgEp-fbtBt"}},{"cell_type":"markdown","source":["To Store Resulting Files in your Google Drive: "],"metadata":{"id":"bGhAXFSfbtBx"}},{"cell_type":"code","source":["! cp $PIPELINE_FILE_PATH/long-read-sequencing-pipeline/fastqc/${ACC}_fastqc.html $STORAGE_FILE_PATH/${ACC}_fastqc.html"],"metadata":{"id":"z2dsXIK4btBy","executionInfo":{"status":"ok","timestamp":1661176185039,"user_tz":240,"elapsed":2,"user":{"displayName":"Theodore Maximilian Nelson","userId":"03436156380930975060"}}},"execution_count":34,"outputs":[]},{"cell_type":"markdown","source":["To Store Resulting Files in your Local Hard Drive: "],"metadata":{"id":"6CZoOeptbtB0"}},{"cell_type":"code","source":["! export $ACC\n","! export $PIPELINE_FILE_PATH"],"metadata":{"id":"Q7f4LKtgbtB0","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1661176236640,"user_tz":240,"elapsed":18,"user":{"displayName":"Theodore Maximilian Nelson","userId":"03436156380930975060"}},"outputId":"610e7e2a-69e0-46b3-b5aa-b2ae5badb4ed"},"execution_count":35,"outputs":[{"output_type":"stream","name":"stdout","text":["/bin/bash: line 0: export: `/content/long-read-sequencing-pipeline': not a valid identifier\n"]}]},{"cell_type":"code","source":["from google.colab import files\n","import os\n","\n","files.download(\"\".join([os.environ[\"PIPELINE_FILE_PATH\"],\"/long-read-sequencing-pipeline/fastqc/\",os.environ[\"ACC\"],\"_fastqc.html\"]))"],"metadata":{"colab":{"base_uri":"https://localhost:8080/","height":17},"executionInfo":{"status":"ok","timestamp":1661176236642,"user_tz":240,"elapsed":10,"user":{"displayName":"Theodore Maximilian Nelson","userId":"03436156380930975060"}},"outputId":"de158125-a987-435d-d616-074f802e304c","id":"LqiW6frsbtB1"},"execution_count":36,"outputs":[{"output_type":"display_data","data":{"text/plain":["<IPython.core.display.Javascript object>"],"application/javascript":["\n","    async function download(id, filename, size) {\n","      if (!google.colab.kernel.accessAllowed) {\n","        return;\n","      }\n","      const div = document.createElement('div');\n","      const label = document.createElement('label');\n","      label.textContent = `Downloading \"${filename}\": `;\n","      div.appendChild(label);\n","      const progress = document.createElement('progress');\n","      progress.max = size;\n","      div.appendChild(progress);\n","      document.body.appendChild(div);\n","\n","      const buffers = [];\n","      let downloaded = 0;\n","\n","      const channel = await google.colab.kernel.comms.open(id);\n","      // Send a message to notify the kernel that we're ready.\n","      channel.send({})\n","\n","      for await (const message of channel.messages) {\n","        // Send a message to notify the kernel that we're ready.\n","        channel.send({})\n","        if (message.buffers) {\n","          for (const buffer of message.buffers) {\n","            buffers.push(buffer);\n","            downloaded += buffer.byteLength;\n","            progress.value = downloaded;\n","          }\n","        }\n","      }\n","      const blob = new Blob(buffers, {type: 'application/binary'});\n","      const a = document.createElement('a');\n","      a.href = window.URL.createObjectURL(blob);\n","      a.download = filename;\n","      div.appendChild(a);\n","      a.click();\n","      div.remove();\n","    }\n","  "]},"metadata":{}},{"output_type":"display_data","data":{"text/plain":["<IPython.core.display.Javascript object>"],"application/javascript":["download(\"download_18510107-5079-44bb-b605-dd839e6a40a0\", \"ERR1951293_fastqc.html\", 290978)"]},"metadata":{}}]},{"cell_type":"markdown","metadata":{"id":"7uHwZ4Zj1vgC"},"source":["##$\\color{#e74b4b}{\\text{Reference Genome - installation}}$"]},{"cell_type":"markdown","metadata":{"id":"5FvumDB7c6bH"},"source":["The Reference Genome is the reference genome that the long-read sequencing reads are aligned. These commands install the hg38 genome and ensembl annotation availiable from UCSC. You can download more current genomes by utilizing the appropriate links from NCBI RefSeq, Ensembl, or other reference genome providers. If you are unsure of how to find other species, we recommend checking out the list species available in the ```current_fasta``` and ```current_gtf``` folders: http://ftp.ensembl.org/pub/"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"_D5cG18VURVJ"},"outputs":[],"source":["! wget -P $PIPELINE_FILE_PATH/long-read-sequencing-pipeline/prebuilt_indices http://hgdownload.cse.ucsc.edu/goldenPath/hg38/bigZips/hg38.fa.gz"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"Jy4bAiZ6z3Qb"},"outputs":[],"source":["! wget -P $PIPELINE_FILE_PATH/long-read-sequencing-pipeline/prebuilt_indices http://hgdownload.cse.ucsc.edu/goldenPath/hg38/bigZips/genes/hg38.ensGene.gtf.gz"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"bz8hNYEvUXOE"},"outputs":[],"source":["! gunzip -c $PIPELINE_FILE_PATH/long-read-sequencing-pipeline/prebuilt_indices/hg38.fa.gz > $PIPELINE_FILE_PATH/long-read-sequencing-pipeline/prebuilt_indices/hg38.fa"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"z2Hqu_wb0NOv"},"outputs":[],"source":["! gunzip -c $PIPELINE_FILE_PATH/long-read-sequencing-pipeline/prebuilt_indices/hg38.ensGene.gtf.gz > $PIPELINE_FILE_PATH/long-read-sequencing-pipeline/prebuilt_indices/hg38.ensGene.gtf"]},{"cell_type":"markdown","source":["Another example is provided for which installs the UCSC murine mm39 genome. Note that the reference annotation is from RefSeq. "],"metadata":{"id":"iulCArtGA6sw"}},{"cell_type":"code","source":["! wget -P $PIPELINE_FILE_PATH/long-read-sequencing-pipeline/prebuilt_indices ftp://hgdownload.soe.ucsc.edu/goldenPath/mm39/bigZips/mm39.fa.gz"],"metadata":{"id":"rf6DUsMHBUmp","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1661177110457,"user_tz":240,"elapsed":57694,"user":{"displayName":"Theodore Maximilian Nelson","userId":"03436156380930975060"}},"outputId":"74430c98-0333-4c36-faf8-56c940ccd6a7"},"execution_count":40,"outputs":[{"output_type":"stream","name":"stdout","text":["--2022-08-22 14:04:12--  ftp://hgdownload.soe.ucsc.edu/goldenPath/mm39/bigZips/mm39.fa.gz\n","           => ‘/content/long-read-sequencing-pipeline/long-read-sequencing-pipeline/prebuilt_indices/mm39.fa.gz’\n","Resolving hgdownload.soe.ucsc.edu (hgdownload.soe.ucsc.edu)... 128.114.119.163\n","Connecting to hgdownload.soe.ucsc.edu (hgdownload.soe.ucsc.edu)|128.114.119.163|:21... connected.\n","Logging in as anonymous ... Logged in!\n","==> SYST ... done.    ==> PWD ... done.\n","==> TYPE I ... done.  ==> CWD (1) /goldenPath/mm39/bigZips ... done.\n","==> SIZE mm39.fa.gz ... 870543764\n","==> PASV ... done.    ==> RETR mm39.fa.gz ... done.\n","Length: 870543764 (830M) (unauthoritative)\n","\n","mm39.fa.gz          100%[===================>] 830.21M  15.7MB/s    in 55s     \n","\n","2022-08-22 14:05:09 (15.2 MB/s) - ‘/content/long-read-sequencing-pipeline/long-read-sequencing-pipeline/prebuilt_indices/mm39.fa.gz’ saved [870543764]\n","\n"]}]},{"cell_type":"code","source":["! wget -P $PIPELINE_FILE_PATH/long-read-sequencing-pipeline/prebuilt_indices https://hgdownload.soe.ucsc.edu/goldenPath/mm39/bigZips/genes/mm39.ncbiRefSeq.gtf.gz"],"metadata":{"id":"IL949aPzBUsA","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1661177113995,"user_tz":240,"elapsed":3544,"user":{"displayName":"Theodore Maximilian Nelson","userId":"03436156380930975060"}},"outputId":"f6db395e-f06f-4428-d288-185e1a3ee6e1"},"execution_count":41,"outputs":[{"output_type":"stream","name":"stdout","text":["--2022-08-22 14:05:10--  https://hgdownload.soe.ucsc.edu/goldenPath/mm39/bigZips/genes/mm39.ncbiRefSeq.gtf.gz\n","Resolving hgdownload.soe.ucsc.edu (hgdownload.soe.ucsc.edu)... 128.114.119.163\n","Connecting to hgdownload.soe.ucsc.edu (hgdownload.soe.ucsc.edu)|128.114.119.163|:443... connected.\n","HTTP request sent, awaiting response... 200 OK\n","Length: 29617837 (28M) [application/x-gzip]\n","Saving to: ‘/content/long-read-sequencing-pipeline/long-read-sequencing-pipeline/prebuilt_indices/mm39.ncbiRefSeq.gtf.gz’\n","\n","mm39.ncbiRefSeq.gtf 100%[===================>]  28.25M  10.4MB/s    in 2.7s    \n","\n","2022-08-22 14:05:13 (10.4 MB/s) - ‘/content/long-read-sequencing-pipeline/long-read-sequencing-pipeline/prebuilt_indices/mm39.ncbiRefSeq.gtf.gz’ saved [29617837/29617837]\n","\n"]}]},{"cell_type":"code","source":["! gunzip -c $PIPELINE_FILE_PATH/long-read-sequencing-pipeline/prebuilt_indices/mm39.fa.gz > $PIPELINE_FILE_PATH/long-read-sequencing-pipeline/prebuilt_indices/mm39.fa"],"metadata":{"id":"vAqaExDBBUxi","executionInfo":{"status":"ok","timestamp":1661177144116,"user_tz":240,"elapsed":30126,"user":{"displayName":"Theodore Maximilian Nelson","userId":"03436156380930975060"}}},"execution_count":42,"outputs":[]},{"cell_type":"code","source":["! gunzip -c $PIPELINE_FILE_PATH/long-read-sequencing-pipeline/prebuilt_indices/mm39.ncbiRefSeq.gtf.gz > $PIPELINE_FILE_PATH/long-read-sequencing-pipeline/prebuilt_indices/mm39.ncbiRefSeq.gtf"],"metadata":{"id":"suKULpveBU20","executionInfo":{"status":"ok","timestamp":1661177148903,"user_tz":240,"elapsed":4799,"user":{"displayName":"Theodore Maximilian Nelson","userId":"03436156380930975060"}}},"execution_count":43,"outputs":[]},{"cell_type":"markdown","metadata":{"id":"1mvZC8IsijTg"},"source":["##$\\color{#e74b4b}{\\text{minimap2 featherweight alignment - installation}}$"]},{"cell_type":"markdown","metadata":{"id":"_Z4pLWqTe6K7"},"source":["The minimap2 featherweight implementation works by splitting the reference genome into four parts with minimal loss in alignment accuracy (documentation: https://doi.org/10.1038/s41598-019-40739-8)."]},{"cell_type":"code","execution_count":44,"metadata":{"id":"0FCbrpo9ijfj","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1661177374062,"user_tz":240,"elapsed":4465,"user":{"displayName":"Theodore Maximilian Nelson","userId":"03436156380930975060"}},"outputId":"cd2b10d5-fc65-40c5-efbd-866e5aeb3c89"},"outputs":[{"output_type":"stream","name":"stdout","text":["--2022-08-22 14:09:29--  https://github.com/hasindu2008/minimap2-arm/archive/v0.1.tar.gz\n","Resolving github.com (github.com)... 140.82.121.4\n","Connecting to github.com (github.com)|140.82.121.4|:443... connected.\n","HTTP request sent, awaiting response... 302 Found\n","Location: https://codeload.github.com/hasindu2008/minimap2-arm/tar.gz/refs/tags/v0.1 [following]\n","--2022-08-22 14:09:29--  https://codeload.github.com/hasindu2008/minimap2-arm/tar.gz/refs/tags/v0.1\n","Resolving codeload.github.com (codeload.github.com)... 140.82.121.10\n","Connecting to codeload.github.com (codeload.github.com)|140.82.121.10|:443... connected.\n","HTTP request sent, awaiting response... 200 OK\n","Length: unspecified [application/x-gzip]\n","Saving to: ‘v0.1.tar.gz.1’\n","\n","v0.1.tar.gz.1           [ <=>                ] 228.88K  --.-KB/s    in 0.03s   \n","\n","2022-08-22 14:09:29 (6.71 MB/s) - ‘v0.1.tar.gz.1’ saved [234373]\n","\n","minimap2-arm-0.1/\n","minimap2-arm-0.1/.gitignore\n","minimap2-arm-0.1/.travis.yml\n","minimap2-arm-0.1/LICENSE.txt\n","minimap2-arm-0.1/MANIFEST.in\n","minimap2-arm-0.1/Makefile\n","minimap2-arm-0.1/NEWS.md\n","minimap2-arm-0.1/README.md\n","minimap2-arm-0.1/align.c\n","minimap2-arm-0.1/bseq.c\n","minimap2-arm-0.1/bseq.h\n","minimap2-arm-0.1/chain.c\n","minimap2-arm-0.1/cookbook.md\n","minimap2-arm-0.1/esterr.c\n","minimap2-arm-0.1/example.c\n","minimap2-arm-0.1/format.c\n","minimap2-arm-0.1/getopt.c\n","minimap2-arm-0.1/getopt.h\n","minimap2-arm-0.1/hit.c\n","minimap2-arm-0.1/index.c\n","minimap2-arm-0.1/kalloc.c\n","minimap2-arm-0.1/kalloc.h\n","minimap2-arm-0.1/kdq.h\n","minimap2-arm-0.1/khash.h\n","minimap2-arm-0.1/kseq.h\n","minimap2-arm-0.1/ksort.h\n","minimap2-arm-0.1/ksw2.h\n","minimap2-arm-0.1/ksw2_dispatch.c\n","minimap2-arm-0.1/ksw2_extd2_sse.c\n","minimap2-arm-0.1/ksw2_exts2_sse.c\n","minimap2-arm-0.1/ksw2_extz2_sse.c\n","minimap2-arm-0.1/ksw2_ll_sse.c\n","minimap2-arm-0.1/kthread.c\n","minimap2-arm-0.1/kthread.h\n","minimap2-arm-0.1/kvec.h\n","minimap2-arm-0.1/main.c\n","minimap2-arm-0.1/map.c\n","minimap2-arm-0.1/merge.c\n","minimap2-arm-0.1/merge.h\n","minimap2-arm-0.1/minimap.h\n","minimap2-arm-0.1/minimap2.1\n","minimap2-arm-0.1/misc.c\n","minimap2-arm-0.1/misc/\n","minimap2-arm-0.1/misc/README.md\n","minimap2-arm-0.1/misc/idxtools/\n","minimap2-arm-0.1/misc/idxtools/.gitignore\n","minimap2-arm-0.1/misc/idxtools/Makefile\n","minimap2-arm-0.1/misc/idxtools/README.md\n","minimap2-arm-0.1/misc/idxtools/divide.c\n","minimap2-arm-0.1/misc/idxtools/divide_and_index.sh\n","minimap2-arm-0.1/misc/idxtools/eval/\n","minimap2-arm-0.1/misc/idxtools/eval/Mem_usage_with_chr_balance.sh\n","minimap2-arm-0.1/misc/idxtools/eval/README.md\n","minimap2-arm-0.1/misc/idxtools/eval/chimeric_eval.sh\n","minimap2-arm-0.1/misc/idxtools/eval/chimeric_plot.m\n","minimap2-arm-0.1/misc/idxtools/eval/comparesam.c\n","minimap2-arm-0.1/misc/idxtools/eval/gettime.sh\n","minimap2-arm-0.1/misc/idxtools/eval/mem_usage_no_chr_balance.sh\n","minimap2-arm-0.1/misc/idxtools/eval/real_get_stat.sh\n","minimap2-arm-0.1/misc/idxtools/eval/run_comparesam.sh\n","minimap2-arm-0.1/misc/idxtools/eval/synth_eval.sh\n","minimap2-arm-0.1/misc/idxtools/eval/synth_plot.m\n","minimap2-arm-0.1/misc/paftools.js\n","minimap2-arm-0.1/mmpriv.h\n","minimap2-arm-0.1/options.c\n","minimap2-arm-0.1/pe.c\n","minimap2-arm-0.1/python/\n","minimap2-arm-0.1/python/README.rst\n","minimap2-arm-0.1/python/cmappy.h\n","minimap2-arm-0.1/python/cmappy.pxd\n","minimap2-arm-0.1/python/mappy.pyx\n","minimap2-arm-0.1/python/minimap2.py\n","minimap2-arm-0.1/sdust.c\n","minimap2-arm-0.1/sdust.h\n","minimap2-arm-0.1/setup.py\n","minimap2-arm-0.1/sketch.c\n","minimap2-arm-0.1/sse2neon/\n","minimap2-arm-0.1/sse2neon/emmintrin.h\n","minimap2-arm-0.1/test/\n","minimap2-arm-0.1/test/MT-human.fa\n","minimap2-arm-0.1/test/MT-orang.fa\n","minimap2-arm-0.1/test/q-inv.fa\n","minimap2-arm-0.1/test/q2.fa\n","minimap2-arm-0.1/test/t-inv.fa\n","minimap2-arm-0.1/test/t2.fa\n","minimap2-arm-0.1/tex/\n","minimap2-arm-0.1/tex/Makefile\n","minimap2-arm-0.1/tex/bioinfo.cls\n","minimap2-arm-0.1/tex/blasr-mc.eval\n","minimap2-arm-0.1/tex/bowtie2-s3.sam.eval\n","minimap2-arm-0.1/tex/bwa-s3.sam.eval\n","minimap2-arm-0.1/tex/bwa.eval\n","minimap2-arm-0.1/tex/eval2roc.pl\n","minimap2-arm-0.1/tex/graphmap.eval\n","minimap2-arm-0.1/tex/hs38-simu.sh\n","minimap2-arm-0.1/tex/minialign.eval\n","minimap2-arm-0.1/tex/minimap2.bib\n","minimap2-arm-0.1/tex/minimap2.tex\n","minimap2-arm-0.1/tex/mm2-s3.sam.eval\n","minimap2-arm-0.1/tex/mm2.approx.eval\n","minimap2-arm-0.1/tex/mm2.eval\n","minimap2-arm-0.1/tex/natbib.bst\n","minimap2-arm-0.1/tex/natbib.sty\n","minimap2-arm-0.1/tex/ngmlr.eval\n","minimap2-arm-0.1/tex/roc.gp\n","minimap2-arm-0.1/tex/snap-s3.sam.eval\n","cc -c -g -Wall -O2 -Wc++-compat  -msse4.1 -DHAVE_KALLOC -DKSW_CPU_DISPATCH  ksw2_extz2_sse.c -o ksw2_extz2_sse41.o\n","cc -c -g -Wall -O2 -Wc++-compat  -msse4.1 -DHAVE_KALLOC -DKSW_CPU_DISPATCH  ksw2_extd2_sse.c -o ksw2_extd2_sse41.o\n","cc -c -g -Wall -O2 -Wc++-compat  -msse4.1 -DHAVE_KALLOC -DKSW_CPU_DISPATCH  ksw2_exts2_sse.c -o ksw2_exts2_sse41.o\n","cc -c -g -Wall -O2 -Wc++-compat  -msse2 -mno-sse4.1 -DHAVE_KALLOC -DKSW_CPU_DISPATCH -DKSW_SSE2_ONLY  ksw2_extz2_sse.c -o ksw2_extz2_sse2.o\n","cc -c -g -Wall -O2 -Wc++-compat  -msse2 -mno-sse4.1 -DHAVE_KALLOC -DKSW_CPU_DISPATCH -DKSW_SSE2_ONLY  ksw2_extd2_sse.c -o ksw2_extd2_sse2.o\n","cc -c -g -Wall -O2 -Wc++-compat  -msse2 -mno-sse4.1 -DHAVE_KALLOC -DKSW_CPU_DISPATCH -DKSW_SSE2_ONLY  ksw2_exts2_sse.c -o ksw2_exts2_sse2.o\n","cc -c -g -Wall -O2 -Wc++-compat  -msse4.1 -DHAVE_KALLOC -DKSW_CPU_DISPATCH  ksw2_dispatch.c -o ksw2_dispatch.o\n","ar -csru libminimap2.a kthread.o kalloc.o misc.o bseq.o sketch.o sdust.o options.o index.o chain.o align.o hit.o map.o format.o pe.o esterr.o ksw2_ll_sse.o merge.o ksw2_extz2_sse41.o ksw2_extd2_sse41.o ksw2_exts2_sse41.o ksw2_extz2_sse2.o ksw2_extd2_sse2.o ksw2_exts2_sse2.o ksw2_dispatch.o\n","ar: `u' modifier ignored since `D' is the default (see `U')\n","cc -g -Wall -O2 -Wc++-compat  main.o getopt.o -o minimap2 -L. -lminimap2 -lm -lz -lpthread\n"]}],"source":["! wget https://github.com/hasindu2008/minimap2-arm/archive/v0.1.tar.gz\n","! tar xvf v0.1.tar.gz && cd minimap2-arm-0.1 && make"]},{"cell_type":"code","execution_count":45,"metadata":{"id":"sHkWpwJ0js-U","executionInfo":{"status":"ok","timestamp":1661177374063,"user_tz":240,"elapsed":10,"user":{"displayName":"Theodore Maximilian Nelson","userId":"03436156380930975060"}}},"outputs":[],"source":["! chmod u+x /content/minimap2-arm-0.1/misc/idxtools/divide_and_index.sh"]},{"cell_type":"code","execution_count":46,"metadata":{"id":"OSo_TWKelVjr","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1661177385307,"user_tz":240,"elapsed":11251,"user":{"displayName":"Theodore Maximilian Nelson","userId":"03436156380930975060"}},"outputId":"a08ef79e-f04a-49f9-ec59-6c214e90c81a"},"outputs":[{"output_type":"stream","name":"stdout","text":["Reading package lists... Done\n","Building dependency tree       \n","Reading state information... Done\n","The following package was automatically installed and is no longer required:\n","  libnvidia-common-460\n","Use 'apt autoremove' to remove it.\n","The following NEW packages will be installed:\n","  bc\n","0 upgraded, 1 newly installed, 0 to remove and 20 not upgraded.\n","Need to get 86.2 kB of archives.\n","After this operation, 223 kB of additional disk space will be used.\n","Get:1 http://archive.ubuntu.com/ubuntu bionic/main amd64 bc amd64 1.07.1-2 [86.2 kB]\n","Fetched 86.2 kB in 0s (208 kB/s)\n","Selecting previously unselected package bc.\n","(Reading database ... 155676 files and directories currently installed.)\n","Preparing to unpack .../archives/bc_1.07.1-2_amd64.deb ...\n","Unpacking bc (1.07.1-2) ...\n","Setting up bc (1.07.1-2) ...\n","Processing triggers for man-db (2.8.3-2ubuntu0.1) ...\n"]}],"source":["! apt-get install bc"]},{"cell_type":"markdown","metadata":{"id":"oBPQyTcKOPDF"},"source":["##$\\color{#e74b4b}{\\text{minimap2 featherweight alignment - partioning}}$"]},{"cell_type":"markdown","metadata":{"id":"of339UK9OT_0"},"source":["The minimap2 featherweight implementation works by splitting the reference genome into four parts with minimal loss in alignment accuracy (documentation: https://doi.org/10.1038/s41598-019-40739-8). This command will only need to be run once as long as the partioned reference index is not removed from the directory. "]},{"cell_type":"code","execution_count":48,"metadata":{"id":"S2MVZUmwOUID","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1661177566159,"user_tz":240,"elapsed":135205,"user":{"displayName":"Theodore Maximilian Nelson","userId":"03436156380930975060"}},"outputId":"5a5cd1c3-cb1e-4fae-b179-23cd0b7b34bb"},"outputs":[{"output_type":"stream","name":"stdout","text":["Compiling divide.c\n","Running divider\n","INFO : Collecting chromosome stats.\n","INFO : 61 chromosomes parsed.\n","INFO : Determining partitions.\n","INFO : Partition 0 - 6 chromosomes (702432803 non-ambiguous bases). See stat_part0.csv.\n","INFO : Partition 1 - 13 chromosomes (650730000 non-ambiguous bases). See stat_part1.csv.\n","INFO : Partition 2 - 16 chromosomes (650734136 non-ambiguous bases). See stat_part2.csv.\n","INFO : Partition 3 - 26 chromosomes (650724844 non-ambiguous bases). See stat_part3.csv.\n","INFO : Writing partitions.\n","INFO : Done.\n","Minimap2 Indexing\n","Creating partition 0\n","[M::mm_idx_gen::17.129*1.58] collected minimizers\n","[M::mm_idx_gen::20.511*1.81] sorted minimizers\n","[M::main::25.097*1.58] loaded/built the index for 6 target sequence(s)\n","[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 6\n","[M::mm_idx_stat::25.662*1.57] distinct minimizers: 38239488 (62.53% are singletons); average occurrences: 2.413; average spacing: 7.828\n","[M::main] Version: 2.11-r797\n","[M::main] CMD: /content/minimap2-arm-0.1/minimap2 -a -x map-pb -d part0.idx part0.fa\n","[M::main] Real time: 26.014 sec; CPU: 40.687 sec\n","Creating partition 1\n","[M::mm_idx_gen::13.790*1.50] collected minimizers\n","[M::mm_idx_gen::16.532*1.74] sorted minimizers\n","[M::main::21.359*1.52] loaded/built the index for 13 target sequence(s)\n","[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 13\n","[M::mm_idx_stat::21.922*1.51] distinct minimizers: 40118326 (62.16% are singletons); average occurrences: 2.136; average spacing: 7.782\n","[M::main] Version: 2.11-r797\n","[M::main] CMD: /content/minimap2-arm-0.1/minimap2 -a -x map-pb -d part1.idx part1.fa\n","[M::main] Real time: 22.248 sec; CPU: 33.459 sec\n","Creating partition 2\n","[M::mm_idx_gen::14.271*1.47] collected minimizers\n","[M::mm_idx_gen::17.057*1.72] sorted minimizers\n","[M::main::21.202*1.54] loaded/built the index for 16 target sequence(s)\n","[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 16\n","[M::mm_idx_stat::21.750*1.52] distinct minimizers: 38477470 (62.33% are singletons); average occurrences: 2.220; average spacing: 7.857\n","[M::main] Version: 2.11-r797\n","[M::main] CMD: /content/minimap2-arm-0.1/minimap2 -a -x map-pb -d part2.idx part2.fa\n","[M::main] Real time: 22.081 sec; CPU: 33.423 sec\n","Creating partition 3\n","[M::mm_idx_gen::14.243*1.48] collected minimizers\n","[M::mm_idx_gen::17.072*1.73] sorted minimizers\n","[M::main::21.376*1.49] loaded/built the index for 26 target sequence(s)\n","[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 26\n","[M::mm_idx_stat::21.926*1.48] distinct minimizers: 40009642 (62.60% are singletons); average occurrences: 2.139; average spacing: 7.807\n","[M::main] Version: 2.11-r797\n","[M::main] CMD: /content/minimap2-arm-0.1/minimap2 -a -x map-pb -d part3.idx part3.fa\n","[M::main] Real time: 22.284 sec; CPU: 32.829 sec\n","concatenating  part0.idx part1.idx part2.idx part3.idx\n"]}],"source":["! eval /content/minimap2-arm-0.1/misc/idxtools/divide_and_index.sh $INDEX_FILE_PATH 4 $PARTITIONED_INDEX_FILE_PATH /content/minimap2-arm-0.1/minimap2 $PLATFORM_FREE"]},{"cell_type":"markdown","metadata":{"id":"itX8aI_chgmy"},"source":["##$\\color{#e74b4b}{\\text{minimap2 featherweight alignment - usage}}$"]},{"cell_type":"markdown","metadata":{"id":"aY5xHM_gieeF"},"source":["The minimap2 featherweight implementation works by splitting the reference genome into four parts with minimal loss in alignment accuracy (documentation: https://doi.org/10.1038/s41598-019-40739-8)."]},{"cell_type":"code","execution_count":49,"metadata":{"id":"s9pBrxRPi_hU","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1661177954826,"user_tz":240,"elapsed":301986,"user":{"displayName":"Theodore Maximilian Nelson","userId":"03436156380930975060"}},"outputId":"cba91059-72e7-4593-fc58-744a482fbb9a"},"outputs":[{"output_type":"stream","name":"stdout","text":["[WARNING]\u001b[1;31m option --multi-prefix is experimental. Currently works only with uni-segment reads.\u001b[0m\n","[M::main::2.327*1.00] loaded/built the index for 6 target sequence(s)\n","[M::mm_mapopt_update::3.136*1.00] mid_occ = 293\n","[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 6\n","[M::mm_idx_stat::3.672*1.00] distinct minimizers: 38239488 (62.53% are singletons); average occurrences: 2.413; average spacing: 7.828\n","[M::worker_pipeline::21.783*2.51] mapped 301135 sequences\n","[M::main::23.638*2.39] loaded/built the index for 13 target sequence(s)\n","[M::mm_mapopt_update::23.638*2.39] mid_occ = 293\n","[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 13\n","[M::mm_idx_stat::24.172*2.36] distinct minimizers: 40118326 (62.16% are singletons); average occurrences: 2.136; average spacing: 7.782\n","[M::worker_pipeline::49.748*2.63] mapped 301135 sequences\n","[M::main::51.560*2.58] loaded/built the index for 16 target sequence(s)\n","[M::mm_mapopt_update::51.560*2.58] mid_occ = 293\n","[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 16\n","[M::mm_idx_stat::52.117*2.56] distinct minimizers: 38477470 (62.33% are singletons); average occurrences: 2.220; average spacing: 7.857\n","[M::worker_pipeline::118.944*2.76] mapped 301135 sequences\n","[M::main::120.797*2.73] loaded/built the index for 26 target sequence(s)\n","[M::mm_mapopt_update::120.797*2.73] mid_occ = 293\n","[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 26\n","[M::mm_idx_stat::121.322*2.72] distinct minimizers: 40009642 (62.60% are singletons); average occurrences: 2.139; average spacing: 7.807\n","[M::worker_pipeline::296.284*2.87] mapped 301135 sequences\n","[M::main] Version: 2.11-r797\n","[M::main] CMD: /content/minimap2-arm-0.1/minimap2 -a -x map-pb --multi-prefix tmp /content/long-read-sequencing-pipeline/long-read-sequencing-pipeline/prebuilt_indices/mm39.idx /content/long-read-sequencing-pipeline/long-read-sequencing-pipeline/fastq/ERR1951293.fastq\n","[M::main] Real time: 301.354 sec; CPU: 854.100 sec\n"]}],"source":["! eval /content/minimap2-arm-0.1/minimap2 -a -x $PLATFORM_FREE $PARTITIONED_INDEX_FILE_PATH $PIPELINE_FILE_PATH/long-read-sequencing-pipeline/fastq/$ACC.fastq --multi-prefix tmp > $PIPELINE_FILE_PATH/long-read-sequencing-pipeline/sam/$ACC.sam"]},{"cell_type":"markdown","source":["##$\\color{#e74b4b}{\\text{minimap2 featherweight alignment - export}}$\n"],"metadata":{"id":"J_OH_OItuBjO"}},{"cell_type":"markdown","source":["To Store Resulting Files in your Google Drive: "],"metadata":{"id":"tCKOQBzbuBjR"}},{"cell_type":"code","source":["! cp $PIPELINE_FILE_PATH/long-read-sequencing-pipeline/sam/$ACC.sam $STORAGE_FILE_PATH/$ACC.sam"],"metadata":{"id":"xEM-Bg_nuBjS"},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":["To Store Resulting Files in your Local Hard Drive: "],"metadata":{"id":"sfmGI2VruBjS"}},{"cell_type":"code","source":["! export $ACC\n","! export $PIPELINE_FILE_PATH"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"PbpWFTAauBjT","executionInfo":{"status":"ok","timestamp":1661178891596,"user_tz":240,"elapsed":8,"user":{"displayName":"Theodore Maximilian Nelson","userId":"03436156380930975060"}},"outputId":"8f3f140f-a3dd-4a05-d849-44b0293d2a84"},"execution_count":64,"outputs":[{"output_type":"stream","name":"stdout","text":["/bin/bash: line 0: export: `/content/long-read-sequencing-pipeline': not a valid identifier\n"]}]},{"cell_type":"code","source":["from google.colab import files\n","import os\n","\n","files.download(\"\".join([os.environ[\"PIPELINE_FILE_PATH\"],\"/long-read-sequencing-pipeline/sam/\",os.environ[\"ACC\"],\".sam\"]))"],"metadata":{"colab":{"base_uri":"https://localhost:8080/","height":17},"executionInfo":{"status":"ok","timestamp":1661178261164,"user_tz":240,"elapsed":269,"user":{"displayName":"Theodore Maximilian Nelson","userId":"03436156380930975060"}},"outputId":"3f94137c-9105-47e4-8339-c4a230d264cf","id":"zpy-NJKXuBjU"},"execution_count":53,"outputs":[{"output_type":"display_data","data":{"text/plain":["<IPython.core.display.Javascript object>"],"application/javascript":["\n","    async function download(id, filename, size) {\n","      if (!google.colab.kernel.accessAllowed) {\n","        return;\n","      }\n","      const div = document.createElement('div');\n","      const label = document.createElement('label');\n","      label.textContent = `Downloading \"${filename}\": `;\n","      div.appendChild(label);\n","      const progress = document.createElement('progress');\n","      progress.max = size;\n","      div.appendChild(progress);\n","      document.body.appendChild(div);\n","\n","      const buffers = [];\n","      let downloaded = 0;\n","\n","      const channel = await google.colab.kernel.comms.open(id);\n","      // Send a message to notify the kernel that we're ready.\n","      channel.send({})\n","\n","      for await (const message of channel.messages) {\n","        // Send a message to notify the kernel that we're ready.\n","        channel.send({})\n","        if (message.buffers) {\n","          for (const buffer of message.buffers) {\n","            buffers.push(buffer);\n","            downloaded += buffer.byteLength;\n","            progress.value = downloaded;\n","          }\n","        }\n","      }\n","      const blob = new Blob(buffers, {type: 'application/binary'});\n","      const a = document.createElement('a');\n","      a.href = window.URL.createObjectURL(blob);\n","      a.download = filename;\n","      div.appendChild(a);\n","      a.click();\n","      div.remove();\n","    }\n","  "]},"metadata":{}},{"output_type":"display_data","data":{"text/plain":["<IPython.core.display.Javascript object>"],"application/javascript":["download(\"download_1a84c5ef-943f-4fb8-a240-c9498f871182\", \"ERR1951293.sam\", 766801389)"]},"metadata":{}}]},{"cell_type":"markdown","metadata":{"id":"dxspwFM3_1qN"},"source":["##$\\color{#ff00d5}{\\text{samtools: Write/Index SAM to BAM - installation}}$"]},{"cell_type":"markdown","metadata":{"id":"SXphSNhPA9di"},"source":["samtools allows for manipulation of high-throughput sequencing data (documentation: http://www.htslib.org/) \n"]},{"cell_type":"code","execution_count":54,"metadata":{"id":"v0_UcWdBHDGQ","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1661178391529,"user_tz":240,"elapsed":16950,"user":{"displayName":"Theodore Maximilian Nelson","userId":"03436156380930975060"}},"outputId":"cbbbb3e6-d0d8-4a79-a8e0-e8315c692867"},"outputs":[{"output_type":"stream","name":"stdout","text":["Collecting package metadata (current_repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\bdone\n","Solving environment: / \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\bdone\n","\n","## Package Plan ##\n","\n","  environment location: /usr/local\n","\n","  added / updated specs:\n","    - samtools\n","\n","\n","The following packages will be downloaded:\n","\n","    package                    |            build\n","    ---------------------------|-----------------\n","    bzip2-1.0.8                |       h7b6447c_0          78 KB\n","    libgcc-7.2.0               |       h69d50b8_2         269 KB\n","    samtools-1.7               |                1         1.0 MB  bioconda\n","    ------------------------------------------------------------\n","                                           Total:         1.4 MB\n","\n","The following NEW packages will be INSTALLED:\n","\n","  bzip2              pkgs/main/linux-64::bzip2-1.0.8-h7b6447c_0\n","  libgcc             pkgs/main/linux-64::libgcc-7.2.0-h69d50b8_2\n","  samtools           bioconda/linux-64::samtools-1.7-1\n","\n","\n","\n","Downloading and Extracting Packages\n","bzip2-1.0.8          | 78 KB     | : 100% 1.0/1 [00:00<00:00,  7.52it/s]\n","libgcc-7.2.0         | 269 KB    | : 100% 1.0/1 [00:00<00:00,  2.93it/s]               \n","samtools-1.7         | 1.0 MB    | : 100% 1.0/1 [00:00<00:00,  2.37it/s]\n","Preparing transaction: - \b\bdone\n","Verifying transaction: | \b\b/ \b\bdone\n","Executing transaction: \\ \b\bdone\n","Retrieving notices: ...working... done\n"]}],"source":["! conda install -c bioconda samtools -y"]},{"cell_type":"markdown","metadata":{"id":"sjns1XbAkILI"},"source":["##$\\color{#d42bb4}{\\text{samtools: Write/Index SAM to BAM - usage}}$"]},{"cell_type":"markdown","metadata":{"id":"KfgkJ0bJkYDv"},"source":["samtools allows for manipulation of high-throughput sequencing data (documentation: http://www.htslib.org/) \n"]},{"cell_type":"code","execution_count":55,"metadata":{"id":"fAFTMfy2HDOs","executionInfo":{"status":"ok","timestamp":1661178440108,"user_tz":240,"elapsed":45849,"user":{"displayName":"Theodore Maximilian Nelson","userId":"03436156380930975060"}}},"outputs":[],"source":["! samtools view -S -b $PIPELINE_FILE_PATH/long-read-sequencing-pipeline/sam/$ACC.sam > $PIPELINE_FILE_PATH/long-read-sequencing-pipeline/bam/$ACC.bam "]},{"cell_type":"code","execution_count":58,"metadata":{"id":"qYOZbl8PJo-h","executionInfo":{"status":"ok","timestamp":1661178749754,"user_tz":240,"elapsed":49109,"user":{"displayName":"Theodore Maximilian Nelson","userId":"03436156380930975060"}}},"outputs":[],"source":["! samtools sort $PIPELINE_FILE_PATH/long-read-sequencing-pipeline/bam/$ACC.bam -o $PIPELINE_FILE_PATH/long-read-sequencing-pipeline/bam/$ACC.sorted.bam  "]},{"cell_type":"code","execution_count":59,"metadata":{"id":"ypjDsfP7JpM6","executionInfo":{"status":"ok","timestamp":1661178754982,"user_tz":240,"elapsed":5236,"user":{"displayName":"Theodore Maximilian Nelson","userId":"03436156380930975060"}}},"outputs":[],"source":["! samtools index $PIPELINE_FILE_PATH/long-read-sequencing-pipeline/bam/$ACC.sorted.bam "]},{"cell_type":"markdown","source":["##$\\color{#a7588f}{\\text{samtools: Write/Index SAM to BAM - export}}$\n"],"metadata":{"id":"swcyphLAvE52"}},{"cell_type":"markdown","source":["To Store Resulting Files in your Google Drive: "],"metadata":{"id":"GbZSSgbkvE54"}},{"cell_type":"code","source":["! cp $PIPELINE_FILE_PATH/long-read-sequencing-pipeline/bam/$ACC.sorted.bam $STORAGE_FILE_PATH/$ACC.sorted.bam"],"metadata":{"id":"qi5xpmLnvE55","executionInfo":{"status":"ok","timestamp":1661178852235,"user_tz":240,"elapsed":2259,"user":{"displayName":"Theodore Maximilian Nelson","userId":"03436156380930975060"}}},"execution_count":60,"outputs":[]},{"cell_type":"code","source":["! cp $PIPELINE_FILE_PATH/long-read-sequencing-pipeline/bam/$ACC.sorted.bam.bai $STORAGE_FILE_PATH/$ACC.sorted.bam.bai"],"metadata":{"id":"_QPGPrNnXNi_"},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":["To Store Resulting Files in your Local Hard Drive: "],"metadata":{"id":"psCo62ZRvE57"}},{"cell_type":"code","source":["! export $ACC\n","! export $PIPELINE_FILE_PATH"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"pVfnEC_kvE58","executionInfo":{"status":"ok","timestamp":1661178951693,"user_tz":240,"elapsed":11,"user":{"displayName":"Theodore Maximilian Nelson","userId":"03436156380930975060"}},"outputId":"4d95e0ea-d49a-4cd0-9daa-4ce245404b18"},"execution_count":65,"outputs":[{"output_type":"stream","name":"stdout","text":["/bin/bash: line 0: export: `/content/long-read-sequencing-pipeline': not a valid identifier\n"]}]},{"cell_type":"code","source":["from google.colab import files\n","import os\n","\n","files.download(\"\".join([os.environ[\"PIPELINE_FILE_PATH\"],\"/long-read-sequencing-pipeline/bam/\",os.environ[\"ACC\"],\".sorted.bam\"]))"],"metadata":{"colab":{"base_uri":"https://localhost:8080/","height":17},"executionInfo":{"status":"ok","timestamp":1661178958599,"user_tz":240,"elapsed":8,"user":{"displayName":"Theodore Maximilian Nelson","userId":"03436156380930975060"}},"outputId":"071dc642-9547-4e9c-964e-e833c472f369","id":"TSjtffSHvE59"},"execution_count":66,"outputs":[{"output_type":"display_data","data":{"text/plain":["<IPython.core.display.Javascript object>"],"application/javascript":["\n","    async function download(id, filename, size) {\n","      if (!google.colab.kernel.accessAllowed) {\n","        return;\n","      }\n","      const div = document.createElement('div');\n","      const label = document.createElement('label');\n","      label.textContent = `Downloading \"${filename}\": `;\n","      div.appendChild(label);\n","      const progress = document.createElement('progress');\n","      progress.max = size;\n","      div.appendChild(progress);\n","      document.body.appendChild(div);\n","\n","      const buffers = [];\n","      let downloaded = 0;\n","\n","      const channel = await google.colab.kernel.comms.open(id);\n","      // Send a message to notify the kernel that we're ready.\n","      channel.send({})\n","\n","      for await (const message of channel.messages) {\n","        // Send a message to notify the kernel that we're ready.\n","        channel.send({})\n","        if (message.buffers) {\n","          for (const buffer of message.buffers) {\n","            buffers.push(buffer);\n","            downloaded += buffer.byteLength;\n","            progress.value = downloaded;\n","          }\n","        }\n","      }\n","      const blob = new Blob(buffers, {type: 'application/binary'});\n","      const a = document.createElement('a');\n","      a.href = window.URL.createObjectURL(blob);\n","      a.download = filename;\n","      div.appendChild(a);\n","      a.click();\n","      div.remove();\n","    }\n","  "]},"metadata":{}},{"output_type":"display_data","data":{"text/plain":["<IPython.core.display.Javascript object>"],"application/javascript":["download(\"download_0504c0b4-3eef-4d3c-abbd-3d564909f8c4\", \"ERR1951293.sorted.bam\", 334069443)"]},"metadata":{}}]},{"cell_type":"code","source":["from google.colab import files\n","import os\n","\n","files.download(\"\".join([os.environ[\"PIPELINE_FILE_PATH\"],\"/long-read-sequencing-pipeline/bam/\",os.environ[\"ACC\"],\".sorted.bam.bai\"]))"],"metadata":{"colab":{"base_uri":"https://localhost:8080/","height":17},"executionInfo":{"status":"ok","timestamp":1661185413908,"user_tz":240,"elapsed":459,"user":{"displayName":"Theodore Maximilian Nelson","userId":"03436156380930975060"}},"outputId":"867da8c2-68a8-4b85-c25c-13621f38931b","id":"sD5KruF8XV86"},"execution_count":101,"outputs":[{"output_type":"display_data","data":{"text/plain":["<IPython.core.display.Javascript object>"],"application/javascript":["\n","    async function download(id, filename, size) {\n","      if (!google.colab.kernel.accessAllowed) {\n","        return;\n","      }\n","      const div = document.createElement('div');\n","      const label = document.createElement('label');\n","      label.textContent = `Downloading \"${filename}\": `;\n","      div.appendChild(label);\n","      const progress = document.createElement('progress');\n","      progress.max = size;\n","      div.appendChild(progress);\n","      document.body.appendChild(div);\n","\n","      const buffers = [];\n","      let downloaded = 0;\n","\n","      const channel = await google.colab.kernel.comms.open(id);\n","      // Send a message to notify the kernel that we're ready.\n","      channel.send({})\n","\n","      for await (const message of channel.messages) {\n","        // Send a message to notify the kernel that we're ready.\n","        channel.send({})\n","        if (message.buffers) {\n","          for (const buffer of message.buffers) {\n","            buffers.push(buffer);\n","            downloaded += buffer.byteLength;\n","            progress.value = downloaded;\n","          }\n","        }\n","      }\n","      const blob = new Blob(buffers, {type: 'application/binary'});\n","      const a = document.createElement('a');\n","      a.href = window.URL.createObjectURL(blob);\n","      a.download = filename;\n","      div.appendChild(a);\n","      a.click();\n","      div.remove();\n","    }\n","  "]},"metadata":{}},{"output_type":"display_data","data":{"text/plain":["<IPython.core.display.Javascript object>"],"application/javascript":["download(\"download_f451375d-010c-439c-a58b-d4db05118435\", \"ERR1951293.sorted.bam.bai\", 1352560)"]},"metadata":{}}]},{"cell_type":"markdown","metadata":{"id":"Y9qrP4Ip3xwz"},"source":["##$\\color{#ff00d5}{\\text{TranscriptClean: correct mismatches, microindels, and noncanonical splice junctions - installation}}$"]},{"cell_type":"markdown","metadata":{"id":"bB4Zr5Ef38qi"},"source":["TranscriptClean is a command-line program which corrects long-read mismatches, microindels and noncanonical splice junctions (documentation: https://github.com/mortazavilab/TranscriptClean). "]},{"cell_type":"code","execution_count":67,"metadata":{"id":"oXIztTkj3zbu","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1661179256388,"user_tz":240,"elapsed":36340,"user":{"displayName":"Theodore Maximilian Nelson","userId":"03436156380930975060"}},"outputId":"910822cc-c70b-47e8-edb7-a9d90122a415"},"outputs":[{"output_type":"stream","name":"stdout","text":["Collecting package metadata (current_repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\bdone\n","Solving environment: | \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\bfailed with initial frozen solve. Retrying with flexible solve.\n","Solving environment: - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\bfailed with repodata from current_repodata.json, will retry with next repodata source.\n","Collecting package metadata (repodata.json): \\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\bdone\n","Solving environment: / \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\bdone\n","\n","## Package Plan ##\n","\n","  environment location: /usr/local\n","\n","  added / updated specs:\n","    - pyfasta\n","    - pyranges\n","    - samtools\n","\n","\n","The following packages will be downloaded:\n","\n","    package                    |            build\n","    ---------------------------|-----------------\n","    htslib-1.12                |       h9093b5e_1         2.1 MB  bioconda\n","    libdeflate-1.7             |       h27cfd23_5          55 KB\n","    natsort-7.1.1              |     pyhd3eb1b0_0          33 KB\n","    ncls-0.0.64                |   py39hbf8eff0_1         637 KB  bioconda\n","    pyfasta-0.5.2              |             py_1          17 KB  bioconda\n","    pyranges-0.0.117           |     pyh5e36f6f_0         672 KB  bioconda\n","    pyrle-0.0.34               |   py39hbf8eff0_1         467 KB  bioconda\n","    samtools-1.12              |       h9aed4be_1         386 KB  bioconda\n","    sorted_nearest-0.0.33      |   py39hbf8eff0_2         1.4 MB  bioconda\n","    tabulate-0.8.10            |   py39h06a4308_0          45 KB\n","    ------------------------------------------------------------\n","                                           Total:         5.8 MB\n","\n","The following NEW packages will be INSTALLED:\n","\n","  htslib             bioconda/linux-64::htslib-1.12-h9093b5e_1\n","  libdeflate         pkgs/main/linux-64::libdeflate-1.7-h27cfd23_5\n","  natsort            pkgs/main/noarch::natsort-7.1.1-pyhd3eb1b0_0\n","  ncls               bioconda/linux-64::ncls-0.0.64-py39hbf8eff0_1\n","  pyfasta            bioconda/noarch::pyfasta-0.5.2-py_1\n","  pyranges           bioconda/noarch::pyranges-0.0.117-pyh5e36f6f_0\n","  pyrle              bioconda/linux-64::pyrle-0.0.34-py39hbf8eff0_1\n","  sorted_nearest     bioconda/linux-64::sorted_nearest-0.0.33-py39hbf8eff0_2\n","  tabulate           pkgs/main/linux-64::tabulate-0.8.10-py39h06a4308_0\n","\n","The following packages will be UPDATED:\n","\n","  samtools                                            1.7-1 --> 1.12-h9aed4be_1\n","\n","\n","\n","Downloading and Extracting Packages\n","pyrle-0.0.34         | 467 KB    | : 100% 1.0/1 [00:00<00:00,  1.99it/s]\n","pyranges-0.0.117     | 672 KB    | : 100% 1.0/1 [00:00<00:00,  2.54it/s]               \n","tabulate-0.8.10      | 45 KB     | : 100% 1.0/1 [00:00<00:00,  3.76it/s]               \n","libdeflate-1.7       | 55 KB     | : 100% 1.0/1 [00:00<00:00,  4.98it/s]               \n","samtools-1.12        | 386 KB    | : 100% 1.0/1 [00:00<00:00,  4.38it/s]                 \n","natsort-7.1.1        | 33 KB     | : 100% 1.0/1 [00:00<00:00, 11.68it/s]\n","ncls-0.0.64          | 637 KB    | : 100% 1.0/1 [00:00<00:00,  1.41it/s]\n","sorted_nearest-0.0.3 | 1.4 MB    | : 100% 1.0/1 [00:01<00:00,  1.42s/it]\n","pyfasta-0.5.2        | 17 KB     | : 100% 1.0/1 [00:00<00:00,  8.01it/s]               \n","htslib-1.12          | 2.1 MB    | : 100% 1.0/1 [00:00<00:00,  1.84it/s]\n","Preparing transaction: / \b\bdone\n","Verifying transaction: \\ \b\b| \b\bdone\n","Executing transaction: - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\bdone\n","Retrieving notices: ...working... done\n"]}],"source":["! conda install -c bioconda pyfasta pyranges samtools -y"]},{"cell_type":"code","execution_count":68,"metadata":{"id":"BfsI59QG4BH_","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1661179288902,"user_tz":240,"elapsed":32522,"user":{"displayName":"Theodore Maximilian Nelson","userId":"03436156380930975060"}},"outputId":"2a2c6b61-3d66-4cb7-8d56-d597e7979f79"},"outputs":[{"output_type":"stream","name":"stdout","text":["--2022-08-22 14:40:55--  https://github.com/mortazavilab/TranscriptClean/archive/refs/tags/v2.0.3.tar.gz\n","Resolving github.com (github.com)... 140.82.121.4\n","Connecting to github.com (github.com)|140.82.121.4|:443... connected.\n","HTTP request sent, awaiting response... 302 Found\n","Location: https://codeload.github.com/mortazavilab/TranscriptClean/tar.gz/refs/tags/v2.0.3 [following]\n","--2022-08-22 14:40:56--  https://codeload.github.com/mortazavilab/TranscriptClean/tar.gz/refs/tags/v2.0.3\n","Resolving codeload.github.com (codeload.github.com)... 140.82.121.9\n","Connecting to codeload.github.com (codeload.github.com)|140.82.121.9|:443... connected.\n","HTTP request sent, awaiting response... 200 OK\n","Length: unspecified [application/x-gzip]\n","Saving to: ‘v2.0.3.tar.gz’\n","\n","v2.0.3.tar.gz           [            <=>     ] 206.75M  2.03MB/s    in 24s     \n","\n","2022-08-22 14:41:20 (8.50 MB/s) - ‘v2.0.3.tar.gz’ saved [216793874]\n","\n","TranscriptClean-2.0.3/\n","TranscriptClean-2.0.3/.gitignore\n","TranscriptClean-2.0.3/LICENSE.md\n","TranscriptClean-2.0.3/README.md\n","TranscriptClean-2.0.3/TranscriptClean.py\n","TranscriptClean-2.0.3/accessory_scripts/\n","TranscriptClean-2.0.3/accessory_scripts/get_SJs_from_gtf.py\n","TranscriptClean-2.0.3/accessory_scripts/get_TranscriptClean_stats.sh\n","TranscriptClean-2.0.3/archived/\n","TranscriptClean-2.0.3/archived/test_TranscriptClean/\n","TranscriptClean-2.0.3/archived/test_TranscriptClean/sam_files/\n","TranscriptClean-2.0.3/archived/test_TranscriptClean/sam_files/deletion_correctAnswer.sam\n","TranscriptClean-2.0.3/archived/test_TranscriptClean/sam_files/deletion_input.sam\n","TranscriptClean-2.0.3/archived/test_TranscriptClean/sam_files/deletion_insertion_mismatch.sam\n","TranscriptClean-2.0.3/archived/test_TranscriptClean/sam_files/deletion_insertion_mismatch_correctAnswer.sam\n","TranscriptClean-2.0.3/archived/test_TranscriptClean/sam_files/deletion_insertion_mismatch_nc.sam\n","TranscriptClean-2.0.3/archived/test_TranscriptClean/sam_files/deletion_insertion_mismatch_nc_correctAnswer.sam\n","TranscriptClean-2.0.3/archived/test_TranscriptClean/sam_files/deletion_insertion_mismatch_nc_noExtraTags.sam\n","TranscriptClean-2.0.3/archived/test_TranscriptClean/sam_files/deletion_insertion_mismatch_noExtraTags.sam\n","TranscriptClean-2.0.3/archived/test_TranscriptClean/sam_files/deletion_variant.sam\n","TranscriptClean-2.0.3/archived/test_TranscriptClean/sam_files/deletion_variant_correctAnswer.sam\n","TranscriptClean-2.0.3/archived/test_TranscriptClean/sam_files/insertion_correctAnswer.sam\n","TranscriptClean-2.0.3/archived/test_TranscriptClean/sam_files/insertion_input.sam\n","TranscriptClean-2.0.3/archived/test_TranscriptClean/sam_files/insertion_variant.sam\n","TranscriptClean-2.0.3/archived/test_TranscriptClean/sam_files/minusStrand_perfectMatch.sam\n","TranscriptClean-2.0.3/archived/test_TranscriptClean/sam_files/mismatch_correctAnswer.sam\n","TranscriptClean-2.0.3/archived/test_TranscriptClean/sam_files/mismatch_input.sam\n","TranscriptClean-2.0.3/archived/test_TranscriptClean/sam_files/nc_case1.sam\n","TranscriptClean-2.0.3/archived/test_TranscriptClean/sam_files/nc_case1_correctAnswer.sam\n","TranscriptClean-2.0.3/archived/test_TranscriptClean/sam_files/nc_case2.sam\n","TranscriptClean-2.0.3/archived/test_TranscriptClean/sam_files/nc_case2_correctAnswer.sam\n","TranscriptClean-2.0.3/archived/test_TranscriptClean/sam_files/nc_case3.sam\n","TranscriptClean-2.0.3/archived/test_TranscriptClean/sam_files/nc_case3_correctAnswer.sam\n","TranscriptClean-2.0.3/archived/test_TranscriptClean/sam_files/nc_case4.sam\n","TranscriptClean-2.0.3/archived/test_TranscriptClean/sam_files/nc_case4_correctAnswer.sam\n","TranscriptClean-2.0.3/archived/test_TranscriptClean/sam_files/nc_case5.sam\n","TranscriptClean-2.0.3/archived/test_TranscriptClean/sam_files/nc_case5_correctAnswer.sam\n","TranscriptClean-2.0.3/archived/test_TranscriptClean/sam_files/perfectReferenceMatch_noIntrons.sam\n","TranscriptClean-2.0.3/archived/test_TranscriptClean/sam_files/perfectReferenceMatch_noIntrons_noExtraTags.sam\n","TranscriptClean-2.0.3/archived/test_TranscriptClean/sam_files/perfectReferenceMatch_twoIntrons.sam\n","TranscriptClean-2.0.3/archived/test_TranscriptClean/sam_files/perfectReferenceMatch_twoIntrons_noExtraTags.sam\n","TranscriptClean-2.0.3/archived/test_TranscriptClean/test_TranscriptClean.py\n","TranscriptClean-2.0.3/dstruct.py\n","TranscriptClean-2.0.3/example_files/\n","TranscriptClean-2.0.3/example_files/GM12878_SJs_chr1.tab\n","TranscriptClean-2.0.3/example_files/GM12878_chr1.sam\n","TranscriptClean-2.0.3/example_files/GM12878_chr1.vcf.gz\n","TranscriptClean-2.0.3/example_files/chr1.fa.gz\n","TranscriptClean-2.0.3/generate_report.R\n","TranscriptClean-2.0.3/intronBound.py\n","TranscriptClean-2.0.3/spliceJunction.py\n","TranscriptClean-2.0.3/testing_suite/\n","TranscriptClean-2.0.3/testing_suite/Makefile\n","TranscriptClean-2.0.3/testing_suite/answer_files/\n","TranscriptClean-2.0.3/testing_suite/answer_files/DIM_nc_full/\n","TranscriptClean-2.0.3/testing_suite/answer_files/DIM_nc_full/TC_clean.sam\n","TranscriptClean-2.0.3/testing_suite/input_files/\n","TranscriptClean-2.0.3/testing_suite/input_files/GM12878_SJs_chr1.tab\n","TranscriptClean-2.0.3/testing_suite/input_files/chr11_sjs.txt\n","TranscriptClean-2.0.3/testing_suite/input_files/drosophila_example/\n","TranscriptClean-2.0.3/testing_suite/input_files/drosophila_example/bad_correction.sam\n","TranscriptClean-2.0.3/testing_suite/input_files/drosophila_example/chr3R.fa\n","TranscriptClean-2.0.3/testing_suite/input_files/drosophila_example/chr3R_SJs.tsv\n","TranscriptClean-2.0.3/testing_suite/input_files/drosophila_example/input_read.sam\n","TranscriptClean-2.0.3/testing_suite/input_files/drosophila_example/no_SJ_corr.sam\n","TranscriptClean-2.0.3/testing_suite/input_files/hg38_chr1.fa.gz\n","TranscriptClean-2.0.3/testing_suite/input_files/hg38_chr11.fa.gz\n","TranscriptClean-2.0.3/testing_suite/input_files/hg38_chr17.fa\n","TranscriptClean-2.0.3/testing_suite/input_files/init_transcript/\n","TranscriptClean-2.0.3/testing_suite/input_files/init_transcript/non-primary.sam\n","TranscriptClean-2.0.3/testing_suite/input_files/init_transcript/unmapped.sam\n","TranscriptClean-2.0.3/testing_suite/input_files/sams/\n","TranscriptClean-2.0.3/testing_suite/input_files/sams/deletion.sam\n","TranscriptClean-2.0.3/testing_suite/input_files/sams/deletion_insertion_mismatch.sam\n","TranscriptClean-2.0.3/testing_suite/input_files/sams/deletion_insertion_mismatch_nc.sam\n","TranscriptClean-2.0.3/testing_suite/input_files/sams/deletion_insertion_mismatch_withHeader.sam\n","TranscriptClean-2.0.3/testing_suite/input_files/sams/insertion.sam\n","TranscriptClean-2.0.3/testing_suite/input_files/sams/insertion_input.sam\n","TranscriptClean-2.0.3/testing_suite/input_files/sams/microexon.sam\n","TranscriptClean-2.0.3/testing_suite/input_files/sams/mismatch.sam\n","TranscriptClean-2.0.3/testing_suite/input_files/sams/perfectReferenceMatch_noIntrons.sam\n","TranscriptClean-2.0.3/testing_suite/input_files/sams/perfectReferenceMatch_twoIntrons.sam\n","TranscriptClean-2.0.3/testing_suite/input_files/test_junctions.txt\n","TranscriptClean-2.0.3/testing_suite/input_files/toy_sjs_mixed_chroms.txt\n","TranscriptClean-2.0.3/testing_suite/input_files/vcf_test/\n","TranscriptClean-2.0.3/testing_suite/input_files/vcf_test/README.md\n","TranscriptClean-2.0.3/testing_suite/input_files/vcf_test/chr11_variantDel.vcf\n","TranscriptClean-2.0.3/testing_suite/input_files/vcf_test/chr11_variantDel_andMismatch.sam\n","TranscriptClean-2.0.3/testing_suite/input_files/vcf_test/read_with_snps.sam\n","TranscriptClean-2.0.3/testing_suite/input_files/vcf_test/snps.vcf\n","TranscriptClean-2.0.3/testing_suite/pytest.ini\n","TranscriptClean-2.0.3/testing_suite/test_VCF_processing.py\n","TranscriptClean-2.0.3/testing_suite/test_all_jns_annotated.py\n","TranscriptClean-2.0.3/testing_suite/test_attempt_jn_correction.py\n","TranscriptClean-2.0.3/testing_suite/test_canonOnly_mode.py\n","TranscriptClean-2.0.3/testing_suite/test_combined_jn_dist.py\n","TranscriptClean-2.0.3/testing_suite/test_compare_seq_cigar_lengths.py\n","TranscriptClean-2.0.3/testing_suite/test_compute_MD_NM.py\n","TranscriptClean-2.0.3/testing_suite/test_compute_jI_jM.py\n","TranscriptClean-2.0.3/testing_suite/test_correct_deletions.py\n","TranscriptClean-2.0.3/testing_suite/test_correct_insertions.py\n","TranscriptClean-2.0.3/testing_suite/test_correct_mismatches.py\n","TranscriptClean-2.0.3/testing_suite/test_create_tmp_sam.py\n","TranscriptClean-2.0.3/testing_suite/test_example_transcripts.py\n","TranscriptClean-2.0.3/testing_suite/test_example_varDM_monoexon.py\n","TranscriptClean-2.0.3/testing_suite/test_fetch_donor_or_acceptor.py\n","TranscriptClean-2.0.3/testing_suite/test_find_closest_annotated_SJ.py\n","TranscriptClean-2.0.3/testing_suite/test_find_closest_annotated_bound.py\n","TranscriptClean-2.0.3/testing_suite/test_find_ncsj.py\n","TranscriptClean-2.0.3/testing_suite/test_fix_one_side_of_junction.py\n","TranscriptClean-2.0.3/testing_suite/test_full_dryrun.py\n","TranscriptClean-2.0.3/testing_suite/test_high_level_ncsj_correction.py\n","TranscriptClean-2.0.3/testing_suite/test_init_transcript_object.py\n","TranscriptClean-2.0.3/testing_suite/test_prep_refs.py\n","TranscriptClean-2.0.3/testing_suite/test_print_fasta_seq.py\n","TranscriptClean-2.0.3/testing_suite/test_process_splice_annotation.py\n","TranscriptClean-2.0.3/testing_suite/test_rmTmp_option.py\n","TranscriptClean-2.0.3/testing_suite/test_update_post_ncsj_correction.py\n","TranscriptClean-2.0.3/testing_suite/unzip_references.sh\n","TranscriptClean-2.0.3/transcript.py\n","TranscriptClean-2.0.3/transcript.pyc\n"]}],"source":["! wget https://github.com/mortazavilab/TranscriptClean/archive/refs/tags/v2.0.3.tar.gz\n","! tar xvf v2.0.3.tar.gz "]},{"cell_type":"markdown","metadata":{"id":"zqQcBdxa4X7q"},"source":["##$\\color{#d42bb4}{\\text{TranscriptClean: correct mismatches, microindels, and noncanonical splice junctions - usage}}$"]},{"cell_type":"markdown","metadata":{"id":"r5CNnQ5w4k_K"},"source":["TranscriptClean is a command-line program which corrects long-read mismatches, microindels and noncanonical splice junctions (documentation: https://github.com/mortazavilab/TranscriptClean). "]},{"cell_type":"markdown","metadata":{"id":"ZU_kVYSvWXdN"},"source":["Please note that the corrected reads should not be utilized for downstream-level base-calling analysis such as variant calling. "]},{"cell_type":"code","execution_count":69,"metadata":{"id":"sJPb4PSU5LmU","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1661182002359,"user_tz":240,"elapsed":2688233,"user":{"displayName":"Theodore Maximilian Nelson","userId":"03436156380930975060"}},"outputId":"41ac634f-0d8c-4565-d8fa-e6cfcb497d77"},"outputs":[{"output_type":"stream","name":"stdout","text":["Reading genome ..............................\n","No splice annotation provided. Will skip splice junction correction.\n","No variant file provided. Transcript correction will not be variant-aware.\n","Reference file processing took 0:00:03\n","Correcting transcripts...\n","Took 0:44:05 to process transcript batch.\n","Took 0:00:05 to combine all outputs.\n"]}],"source":["! eval python /content/TranscriptClean-2.0.3/TranscriptClean.py --sam $PIPELINE_FILE_PATH/long-read-sequencing-pipeline/sam/$ACC.sam --genome $INDEX_FILE_PATH --outprefix $PIPELINE_FILE_PATH/long-read-sequencing-pipeline/transcriptclean/$ACC"]},{"cell_type":"markdown","source":["##$\\color{#a7588f}{\\text{TranscriptClean: correct mismatches, microindels, and noncanonical splice junctions - export}}$\n"],"metadata":{"id":"iUwbIIRkwWrL"}},{"cell_type":"markdown","source":["To Store Resulting Files in your Google Drive: "],"metadata":{"id":"1K-zlCYDwWrL"}},{"cell_type":"code","source":["! cp $PIPELINE_FILE_PATH/long-read-sequencing-pipeline/transcriptclean/${ACC}_clean.fa $STORAGE_FILE_PATH/${ACC}_clean.fa"],"metadata":{"id":"j_1xnKJJwWrM","executionInfo":{"status":"ok","timestamp":1661182166198,"user_tz":240,"elapsed":14,"user":{"displayName":"Theodore Maximilian Nelson","userId":"03436156380930975060"}}},"execution_count":70,"outputs":[]},{"cell_type":"code","source":["! cp $PIPELINE_FILE_PATH/long-read-sequencing-pipeline/transcriptclean/${ACC}_clean.sam $STORAGE_FILE_PATH/${ACC}_clean.sam"],"metadata":{"id":"Ds664sseLDQL","executionInfo":{"status":"ok","timestamp":1661182190674,"user_tz":240,"elapsed":5112,"user":{"displayName":"Theodore Maximilian Nelson","userId":"03436156380930975060"}}},"execution_count":71,"outputs":[]},{"cell_type":"markdown","source":["To Store Resulting Files in your Local Hard Drive: "],"metadata":{"id":"tcK1cY7pwWrN"}},{"cell_type":"code","source":["! export $ACC\n","! export $PIPELINE_FILE_PATH"],"metadata":{"id":"4ziwR4_YwWrO"},"execution_count":null,"outputs":[]},{"cell_type":"code","source":["from google.colab import files\n","import os\n","\n","files.download(\"\".join([os.environ[\"PIPELINE_FILE_PATH\"],\"/bam/\",os.environ[\"ACC\"],\".sorted.bam\"]))"],"metadata":{"colab":{"base_uri":"https://localhost:8080/","height":34},"executionInfo":{"status":"ok","timestamp":1659643945496,"user_tz":240,"elapsed":336,"user":{"displayName":"Theodore Maximilian Nelson","userId":"03436156380930975060"}},"outputId":"f43d037c-3ef6-432c-c555-f0d0dffd0084","id":"RW7dqOgpwWrP"},"execution_count":null,"outputs":[{"output_type":"display_data","data":{"text/plain":["<IPython.core.display.Javascript object>"],"application/javascript":["\n","    async function download(id, filename, size) {\n","      if (!google.colab.kernel.accessAllowed) {\n","        return;\n","      }\n","      const div = document.createElement('div');\n","      const label = document.createElement('label');\n","      label.textContent = `Downloading \"${filename}\": `;\n","      div.appendChild(label);\n","      const progress = document.createElement('progress');\n","      progress.max = size;\n","      div.appendChild(progress);\n","      document.body.appendChild(div);\n","\n","      const buffers = [];\n","      let downloaded = 0;\n","\n","      const channel = await google.colab.kernel.comms.open(id);\n","      // Send a message to notify the kernel that we're ready.\n","      channel.send({})\n","\n","      for await (const message of channel.messages) {\n","        // Send a message to notify the kernel that we're ready.\n","        channel.send({})\n","        if (message.buffers) {\n","          for (const buffer of message.buffers) {\n","            buffers.push(buffer);\n","            downloaded += buffer.byteLength;\n","            progress.value = downloaded;\n","          }\n","        }\n","      }\n","      const blob = new Blob(buffers, {type: 'application/binary'});\n","      const a = document.createElement('a');\n","      a.href = window.URL.createObjectURL(blob);\n","      a.download = filename;\n","      div.appendChild(a);\n","      a.click();\n","      div.remove();\n","    }\n","  "]},"metadata":{}},{"output_type":"display_data","data":{"text/plain":["<IPython.core.display.Javascript object>"],"application/javascript":["download(\"download_507c0d59-ff35-4289-b691-7b0c5d7ce7ea\", \"ERR1951293.fastq\", 589549137)"]},"metadata":{}}]},{"cell_type":"markdown","source":["##$\\color{#ff00d5}{\\text{FLAME: long-read splice variant annotation - installation}}$"],"metadata":{"id":"3U_BpHSJA9Xe"}},{"cell_type":"code","source":["! git clone https://github.com/marabouboy/FLAME"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"pKvCCPWvBJ2b","executionInfo":{"status":"ok","timestamp":1661184298021,"user_tz":240,"elapsed":1729,"user":{"displayName":"Theodore Maximilian Nelson","userId":"03436156380930975060"}},"outputId":"5cb81fd3-21e2-4e9a-ad84-973115c797a5"},"execution_count":76,"outputs":[{"output_type":"stream","name":"stdout","text":["Cloning into 'FLAME'...\n","remote: Enumerating objects: 298, done.\u001b[K\n","remote: Counting objects: 100% (296/296), done.\u001b[K\n","remote: Compressing objects: 100% (182/182), done.\u001b[K\n","remote: Total 298 (delta 127), reused 267 (delta 104), pack-reused 2\u001b[K\n","Receiving objects: 100% (298/298), 11.94 MiB | 21.68 MiB/s, done.\n","Resolving deltas: 100% (127/127), done.\n"]}]},{"cell_type":"code","source":["! chmod 755 /content/FLAME/FLAME/FLAME.py"],"metadata":{"id":"TZaGFSOoVUCo","executionInfo":{"status":"ok","timestamp":1661184862163,"user_tz":240,"elapsed":382,"user":{"displayName":"Theodore Maximilian Nelson","userId":"03436156380930975060"}}},"execution_count":88,"outputs":[]},{"cell_type":"code","source":["! chmod 755 /content/FLAME/setup.py"],"metadata":{"executionInfo":{"status":"ok","timestamp":1661184941662,"user_tz":240,"elapsed":6,"user":{"displayName":"Theodore Maximilian Nelson","userId":"03436156380930975060"}},"id":"7Erq_G3QVlz9"},"execution_count":90,"outputs":[]},{"cell_type":"code","source":["! cd /content/FLAME/ ; python3 setup.py install"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"grBakyZXVv3s","executionInfo":{"status":"ok","timestamp":1661185017523,"user_tz":240,"elapsed":633,"user":{"displayName":"Theodore Maximilian Nelson","userId":"03436156380930975060"}},"outputId":"04e3672c-7722-425f-a826-8546a36be437"},"execution_count":95,"outputs":[{"output_type":"stream","name":"stdout","text":["running install\n","/usr/local/lib/python3.9/site-packages/setuptools/command/install.py:34: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.\n","  warnings.warn(\n","/usr/local/lib/python3.9/site-packages/setuptools/command/easy_install.py:144: EasyInstallDeprecationWarning: easy_install command is deprecated. Use build and pip and other standards-based tools.\n","  warnings.warn(\n","running bdist_egg\n","running egg_info\n","creating FLAME_Full_Length_Adjecency_Matrix_Enumeration.egg-info\n","writing FLAME_Full_Length_Adjecency_Matrix_Enumeration.egg-info/PKG-INFO\n","writing dependency_links to FLAME_Full_Length_Adjecency_Matrix_Enumeration.egg-info/dependency_links.txt\n","writing top-level names to FLAME_Full_Length_Adjecency_Matrix_Enumeration.egg-info/top_level.txt\n","writing manifest file 'FLAME_Full_Length_Adjecency_Matrix_Enumeration.egg-info/SOURCES.txt'\n","reading manifest file 'FLAME_Full_Length_Adjecency_Matrix_Enumeration.egg-info/SOURCES.txt'\n","adding license file 'LICENSE'\n","writing manifest file 'FLAME_Full_Length_Adjecency_Matrix_Enumeration.egg-info/SOURCES.txt'\n","installing library code to build/bdist.linux-x86_64/egg\n","running install_lib\n","warning: install_lib: 'build/lib' does not exist -- no Python modules to install\n","\n","creating build\n","creating build/bdist.linux-x86_64\n","creating build/bdist.linux-x86_64/egg\n","creating build/bdist.linux-x86_64/egg/EGG-INFO\n","copying FLAME_Full_Length_Adjecency_Matrix_Enumeration.egg-info/PKG-INFO -> build/bdist.linux-x86_64/egg/EGG-INFO\n","copying FLAME_Full_Length_Adjecency_Matrix_Enumeration.egg-info/SOURCES.txt -> build/bdist.linux-x86_64/egg/EGG-INFO\n","copying FLAME_Full_Length_Adjecency_Matrix_Enumeration.egg-info/dependency_links.txt -> build/bdist.linux-x86_64/egg/EGG-INFO\n","copying FLAME_Full_Length_Adjecency_Matrix_Enumeration.egg-info/top_level.txt -> build/bdist.linux-x86_64/egg/EGG-INFO\n","zip_safe flag not set; analyzing archive contents...\n","creating dist\n","creating 'dist/FLAME_Full_Length_Adjecency_Matrix_Enumeration-0.3.1-py3.9.egg' and adding 'build/bdist.linux-x86_64/egg' to it\n","removing 'build/bdist.linux-x86_64/egg' (and everything under it)\n","Processing FLAME_Full_Length_Adjecency_Matrix_Enumeration-0.3.1-py3.9.egg\n","Copying FLAME_Full_Length_Adjecency_Matrix_Enumeration-0.3.1-py3.9.egg to /usr/local/lib/python3.9/site-packages\n","Adding FLAME-Full-Length-Adjecency-Matrix-Enumeration 0.3.1 to easy-install.pth file\n","\n","Installed /usr/local/lib/python3.9/site-packages/FLAME_Full_Length_Adjecency_Matrix_Enumeration-0.3.1-py3.9.egg\n","Processing dependencies for FLAME-Full-Length-Adjecency-Matrix-Enumeration==0.3.1\n","Finished processing dependencies for FLAME-Full-Length-Adjecency-Matrix-Enumeration==0.3.1\n"]}]},{"cell_type":"markdown","metadata":{"id":"y3bt2k_tBGnT"},"source":["##$\\color{#d42bb4}{\\text{FLAME: long-read splice variant annotation - usage}}$"]},{"cell_type":"code","execution_count":103,"metadata":{"executionInfo":{"status":"ok","timestamp":1661185929654,"user_tz":240,"elapsed":15399,"user":{"displayName":"Theodore Maximilian Nelson","userId":"03436156380930975060"}},"id":"JB86-GcXZHIT"},"outputs":[],"source":["! bedtools bamtobed -bed12 -i $PIPELINE_FILE_PATH/long-read-sequencing-pipeline/bam/$ACC.sorted.bam > $PIPELINE_FILE_PATH/long-read-sequencing-pipeline/bed/$ACC.sorted.bed12"]},{"cell_type":"code","source":["! eval python3 /content/FLAME/FLAME/FLAME.py -I $PIPELINE_FILE_PATH/long-read-sequencing-pipeline/bed/$ACC.sorted.bed12 -GTF $ANNOTATION_FILE_PATH -G $REGION_NAME -O $ACC"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"_Y4DKdo5LhBB","executionInfo":{"status":"ok","timestamp":1661190668728,"user_tz":240,"elapsed":210481,"user":{"displayName":"Theodore Maximilian Nelson","userId":"03436156380930975060"}},"outputId":"6300cdfd-e94e-488e-aae0-04c1aa845722"},"execution_count":107,"outputs":[{"output_type":"stream","name":"stdout","text":["\n","-------------------------------------------------------------------------------------------\n","\n","-----------\tFLAME: Full Length Adjacency Matrix Enumeration\t\t\t-----------\n","\n","-------------------------------------------------------------------------------------------\n","\n","-----------\tInitiating FLAME\t\t\t\t\t\t-----------\n","Input:\t\t/content/long-read-sequencing-pipeline/long-read-sequencing-pipeline/bed/ERR1951293.sorted.bed12\n","GTF:\t\t/content/long-read-sequencing-pipeline/long-read-sequencing-pipeline/prebuilt_indices/mm39.ncbiRefSeq.gtf\n","Gene:\t\tGm30604\n","Range:\t\t20\n","Output:\t\tFlame-[Suffix]\n","-------------------------------------------------------------------------------------------\n","\n","\n","\n","-----------\tInitiate Filter Function\t\t\t\t\t-----------\n","Status: [##################################################] 100% Done...\n","-----------\tInitiate Translate Function, Incongruent\t\t\t-----------\n","Status: [##################################################] 100% Done...\n","-----------\tInitiate Creation of Empty Adjacency Matrix\t\t\t-----------\n","Status: [##################################################] 100% Done...\n","-----------\tInitiate Creation of Adjacency Matrix, Annotated\t\t-----------\n","Status: [] Error... Empty Input\n","-----------\tInitiate Singling of Incongruent Exons\t\t\t\t-----------\n","Status: [##################################################] 100% Done...\n","-----------\tInitiate Novel Splice Site Detection, Part1: Frequency\t\t-----------\n","tcmalloc: large alloc 1233903616 bytes == 0x564e52dfa000 @  0x7f6553635615 0x564dd221dda2 0x564dd221de2a 0x564dd2252703 0x564dd22a074c 0x564dd2251b53 0x564dd22a46ac 0x564dd2250d4c 0x564dd230245c 0x564dd2251ddb 0x564dd23024bb 0x564dd233bde5 0x564dd21e559e 0x564dd23404e9 0x564dd2340a76 0x564dd2340c79 0x7f6552665c87 0x564dd22ca175\n","Status: [##################################################] 100% Done...\n","-----------\tInitiate Novel Splice Site Detection, Part2: Threshold\t\t-----------\n","Status: [##################################################] 100% Done...\n","-----------\tInitiate Quantification of each Splice Permutaiton, Annotated\t-----------\n","Status: [] Error... Empty Input\n","-----------\tInitiate Quantification of each Splice Permutaiton, Incongruent\t-----------\n","Status: [##################################################] 100% Done...\n","-----------\tInitiate Creation of Empty Adjacency Matrix\t\t\t-----------\n","Status: [##################################################] 100% Done...\n","-----------\tInitiate Creation of Adjacency Matrix, Incongruent\t\t-----------\n","Status: [##################################################] 100% Done...\n"]}]},{"cell_type":"markdown","metadata":{"id":"yhR96h7O5CHN"},"source":["##$\\color{#ff00d5}{\\text{featureCounts: an efficient general purpose program for assigning sequence reads to genomic features - installation}}$"]},{"cell_type":"markdown","metadata":{"id":"VnlTGqwG5JYz"},"source":["featureCounts is a read summarization program suitable for counting reads generated from either RNA or genomic DNA sequencing experiments (documentation: http://subread.sourceforge.net/). "]},{"cell_type":"code","execution_count":null,"metadata":{"id":"Vd48uWN45VkA"},"outputs":[],"source":["! conda install -c bioconda subread -y"]},{"cell_type":"markdown","metadata":{"id":"7yVbaSBc5cu7"},"source":["##$\\color{#d42bb4}{\\text{featureCounts: an efficient general purpose program for assigning sequence reads to genomic features - usage}}$"]},{"cell_type":"markdown","metadata":{"id":"-2kbK23S5oi5"},"source":["featureCounts is a read summarization program suitable for counting reads generated from either RNA or genomic DNA sequencing experiments (documentation: http://subread.sourceforge.net/). "]},{"cell_type":"code","execution_count":null,"metadata":{"colab":{"background_save":true},"id":"1ycdSYt_5plb"},"outputs":[],"source":["! eval featureCounts -M -O -T 24 -L -a $ANNOTATION_FILE_PATH -t exon -g gene_id -o $PIPELINE_FILE_PATH/featureCounts/$ACC.txt $PIPELINE_FILE_PATH/bam/$ACC.bam "]},{"cell_type":"markdown","source":["##$\\color{#a7588f}{\\text{featureCounts: an efficient general purpose program for assigning sequence reads to genomic features - export}}$\n"],"metadata":{"id":"a2s7KTWRw3ry"}},{"cell_type":"markdown","source":["To Store Resulting Files in your Google Drive: "],"metadata":{"id":"ZsTqqT1aw3r0"}},{"cell_type":"code","source":["! cp $PIPELINE_FILE_PATH/bam/$ACC.sorted.bam $STORAGE_FILE_PATH/$ACC.sorted.bam"],"metadata":{"id":"StNofMQFw3r1"},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":["To Store Resulting Files in your Local Hard Drive: "],"metadata":{"id":"jqBqz7Ztw3r2"}},{"cell_type":"code","source":["! export $ACC\n","! export $PIPELINE_FILE_PATH"],"metadata":{"id":"d_8Me1b_w3r3"},"execution_count":null,"outputs":[]},{"cell_type":"code","source":["from google.colab import files\n","import os\n","\n","files.download(\"\".join([os.environ[\"PIPELINE_FILE_PATH\"],\"/bam/\",os.environ[\"ACC\"],\".sorted.bam\"]))"],"metadata":{"colab":{"base_uri":"https://localhost:8080/","height":34},"executionInfo":{"status":"ok","timestamp":1659643945496,"user_tz":240,"elapsed":336,"user":{"displayName":"Theodore Maximilian Nelson","userId":"03436156380930975060"}},"outputId":"f43d037c-3ef6-432c-c555-f0d0dffd0084","id":"I38-6oFow3r4"},"execution_count":null,"outputs":[{"output_type":"display_data","data":{"text/plain":["<IPython.core.display.Javascript object>"],"application/javascript":["\n","    async function download(id, filename, size) {\n","      if (!google.colab.kernel.accessAllowed) {\n","        return;\n","      }\n","      const div = document.createElement('div');\n","      const label = document.createElement('label');\n","      label.textContent = `Downloading \"${filename}\": `;\n","      div.appendChild(label);\n","      const progress = document.createElement('progress');\n","      progress.max = size;\n","      div.appendChild(progress);\n","      document.body.appendChild(div);\n","\n","      const buffers = [];\n","      let downloaded = 0;\n","\n","      const channel = await google.colab.kernel.comms.open(id);\n","      // Send a message to notify the kernel that we're ready.\n","      channel.send({})\n","\n","      for await (const message of channel.messages) {\n","        // Send a message to notify the kernel that we're ready.\n","        channel.send({})\n","        if (message.buffers) {\n","          for (const buffer of message.buffers) {\n","            buffers.push(buffer);\n","            downloaded += buffer.byteLength;\n","            progress.value = downloaded;\n","          }\n","        }\n","      }\n","      const blob = new Blob(buffers, {type: 'application/binary'});\n","      const a = document.createElement('a');\n","      a.href = window.URL.createObjectURL(blob);\n","      a.download = filename;\n","      div.appendChild(a);\n","      a.click();\n","      div.remove();\n","    }\n","  "]},"metadata":{}},{"output_type":"display_data","data":{"text/plain":["<IPython.core.display.Javascript object>"],"application/javascript":["download(\"download_507c0d59-ff35-4289-b691-7b0c5d7ce7ea\", \"ERR1951293.fastq\", 589549137)"]},"metadata":{}}]},{"cell_type":"markdown","metadata":{"id":"npZwllsupfEJ"},"source":["##svist4get: a simple visualization tool for genomic tracks from sequencing experiments - installation"]},{"cell_type":"markdown","metadata":{"id":"niesqrQ1nHMM"},"source":["svist4get allows you to view read coverage at a defined region on a chromosome (documentation: https://bitbucket.org/artegorov/svist4get/src/master/)"]},{"cell_type":"code","execution_count":77,"metadata":{"id":"mq3WmdTxpeft","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1661184344529,"user_tz":240,"elapsed":6789,"user":{"displayName":"Theodore Maximilian Nelson","userId":"03436156380930975060"}},"outputId":"c654ec0c-221b-4d85-b2f1-0049cdb6a4e5"},"outputs":[{"output_type":"stream","name":"stdout","text":["Reading package lists... Done\n","Building dependency tree       \n","Reading state information... Done\n","The following package was automatically installed and is no longer required:\n","  libnvidia-common-460\n","Use 'sudo apt autoremove' to remove it.\n","The following NEW packages will be installed:\n","  bedtools\n","0 upgraded, 1 newly installed, 0 to remove and 20 not upgraded.\n","Need to get 577 kB of archives.\n","After this operation, 2,040 kB of additional disk space will be used.\n","Get:1 http://archive.ubuntu.com/ubuntu bionic/universe amd64 bedtools amd64 2.26.0+dfsg-5 [577 kB]\n","Fetched 577 kB in 0s (5,207 kB/s)\n","debconf: unable to initialize frontend: Dialog\n","debconf: (No usable dialog-like program is installed, so the dialog based frontend cannot be used. at /usr/share/perl5/Debconf/FrontEnd/Dialog.pm line 76, <> line 1.)\n","debconf: falling back to frontend: Readline\n","debconf: unable to initialize frontend: Readline\n","debconf: (This frontend requires a controlling tty.)\n","debconf: falling back to frontend: Teletype\n","dpkg-preconfigure: unable to re-open stdin: \n","Selecting previously unselected package bedtools.\n","(Reading database ... 155693 files and directories currently installed.)\n","Preparing to unpack .../bedtools_2.26.0+dfsg-5_amd64.deb ...\n","Unpacking bedtools (2.26.0+dfsg-5) ...\n","Setting up bedtools (2.26.0+dfsg-5) ...\n"]}],"source":["! sudo apt-get install bedtools"]},{"cell_type":"code","execution_count":null,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1661184344529,"user_tz":240,"elapsed":6789,"user":{"displayName":"Theodore Maximilian Nelson","userId":"03436156380930975060"}},"outputId":"c654ec0c-221b-4d85-b2f1-0049cdb6a4e5","id":"C-1v4kxAxlwm"},"outputs":[{"output_type":"stream","name":"stdout","text":["Reading package lists... Done\n","Building dependency tree       \n","Reading state information... Done\n","The following package was automatically installed and is no longer required:\n","  libnvidia-common-460\n","Use 'sudo apt autoremove' to remove it.\n","The following NEW packages will be installed:\n","  bedtools\n","0 upgraded, 1 newly installed, 0 to remove and 20 not upgraded.\n","Need to get 577 kB of archives.\n","After this operation, 2,040 kB of additional disk space will be used.\n","Get:1 http://archive.ubuntu.com/ubuntu bionic/universe amd64 bedtools amd64 2.26.0+dfsg-5 [577 kB]\n","Fetched 577 kB in 0s (5,207 kB/s)\n","debconf: unable to initialize frontend: Dialog\n","debconf: (No usable dialog-like program is installed, so the dialog based frontend cannot be used. at /usr/share/perl5/Debconf/FrontEnd/Dialog.pm line 76, <> line 1.)\n","debconf: falling back to frontend: Readline\n","debconf: unable to initialize frontend: Readline\n","debconf: (This frontend requires a controlling tty.)\n","debconf: falling back to frontend: Teletype\n","dpkg-preconfigure: unable to re-open stdin: \n","Selecting previously unselected package bedtools.\n","(Reading database ... 155693 files and directories currently installed.)\n","Preparing to unpack .../bedtools_2.26.0+dfsg-5_amd64.deb ...\n","Unpacking bedtools (2.26.0+dfsg-5) ...\n","Setting up bedtools (2.26.0+dfsg-5) ...\n"]}],"source":["! sudo apt-get install bedtools"]},{"cell_type":"code","execution_count":78,"metadata":{"id":"iaRNkYPEoa0T","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1661184348311,"user_tz":240,"elapsed":3790,"user":{"displayName":"Theodore Maximilian Nelson","userId":"03436156380930975060"}},"outputId":"97166f3b-6233-49f8-a6b1-092820c76285"},"outputs":[{"output_type":"stream","name":"stdout","text":["\r0% [Working]\r            \rHit:1 http://archive.ubuntu.com/ubuntu bionic InRelease\n","\r0% [Waiting for headers] [Connecting to security.ubuntu.com (91.189.91.39)] [Co\r                                                                               \rGet:2 http://archive.ubuntu.com/ubuntu bionic-updates InRelease [88.7 kB]\n","\r                                                                               \rGet:3 http://ppa.launchpad.net/c2d4u.team/c2d4u4.0+/ubuntu bionic InRelease [15.9 kB]\n","\r0% [2 InRelease 21.4 kB/88.7 kB 24%] [Connecting to security.ubuntu.com (91.189\r0% [1 InRelease gpgv 242 kB] [2 InRelease 21.4 kB/88.7 kB 24%] [Connecting to s\r                                                                               \rHit:4 http://ppa.launchpad.net/cran/libgit2/ubuntu bionic InRelease\n","\r0% [1 InRelease gpgv 242 kB] [2 InRelease 85.1 kB/88.7 kB 96%] [Connecting to s\r0% [1 InRelease gpgv 242 kB] [Connecting to security.ubuntu.com (91.189.91.39)]\r                                                                               \rGet:5 http://archive.ubuntu.com/ubuntu bionic-backports InRelease [74.6 kB]\n","\r0% [1 InRelease gpgv 242 kB] [5 InRelease 20.0 kB/74.6 kB 27%] [Connecting to s\r0% [1 InRelease gpgv 242 kB] [Connecting to security.ubuntu.com (91.189.91.39)]\r                                                                               \rHit:6 http://ppa.launchpad.net/deadsnakes/ppa/ubuntu bionic InRelease\n","\r0% [1 InRelease gpgv 242 kB] [Waiting for headers] [Connecting to cloud.r-proje\r                                                                               \rHit:7 http://ppa.launchpad.net/graphics-drivers/ppa/ubuntu bionic InRelease\n","\r0% [1 InRelease gpgv 242 kB] [Waiting for headers] [Connected to cloud.r-projec\r                                                                               \rGet:8 https://cloud.r-project.org/bin/linux/ubuntu bionic-cran40/ InRelease [3,626 B]\n","Get:9 http://security.ubuntu.com/ubuntu bionic-security InRelease [88.7 kB]\n","Get:10 http://ppa.launchpad.net/c2d4u.team/c2d4u4.0+/ubuntu bionic/main Sources [2,095 kB]\n","Get:11 http://ppa.launchpad.net/c2d4u.team/c2d4u4.0+/ubuntu bionic/main amd64 Packages [1,073 kB]\n","Ign:12 https://developer.download.nvidia.com/compute/machine-learning/repos/ubuntu1804/x86_64  InRelease\n","Get:13 https://developer.download.nvidia.com/compute/cuda/repos/ubuntu1804/x86_64  InRelease [1,581 B]\n","Hit:14 https://developer.download.nvidia.com/compute/machine-learning/repos/ubuntu1804/x86_64  Release\n","Get:15 http://archive.ubuntu.com/ubuntu bionic-updates/universe amd64 Packages [2,311 kB]\n","Get:16 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 Packages [3,369 kB]\n","Get:17 http://security.ubuntu.com/ubuntu bionic-security/main amd64 Packages [2,937 kB]\n","Get:18 https://developer.download.nvidia.com/compute/cuda/repos/ubuntu1804/x86_64  Packages [910 kB]\n","Get:20 http://security.ubuntu.com/ubuntu bionic-security/universe amd64 Packages [1,533 kB]\n","Fetched 14.5 MB in 3s (4,993 kB/s)\n","Reading package lists... Done\n"]}],"source":["! apt-get update"]},{"cell_type":"code","execution_count":79,"metadata":{"id":"-gNi5jxXA5oq","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1661184374847,"user_tz":240,"elapsed":26541,"user":{"displayName":"Theodore Maximilian Nelson","userId":"03436156380930975060"}},"outputId":"498d9f78-4d81-491b-bb8d-62d953ff8a15"},"outputs":[{"output_type":"stream","name":"stdout","text":["Reading package lists... Done\n","Building dependency tree       \n","Reading state information... Done\n","The following package was automatically installed and is no longer required:\n","  libnvidia-common-460\n","Use 'apt autoremove' to remove it.\n","The following additional packages will be installed:\n","  fonts-droid-fallback fonts-noto-mono ghostscript gir1.2-freedesktop\n","  gir1.2-gdkpixbuf-2.0 gir1.2-rsvg-2.0 gsfonts imagemagick-6-common\n","  libcairo-script-interpreter2 libcairo2-dev libcupsfilters1 libcupsimage2\n","  libdjvulibre-dev libdjvulibre-text libdjvulibre21 libgdk-pixbuf2.0-dev\n","  libgs9 libgs9-common libijs-0.35 libjbig2dec0 liblcms2-dev liblqr-1-0\n","  liblqr-1-0-dev libmagickcore-6-arch-config libmagickcore-6-headers\n","  libmagickcore-6.q16-3 libmagickcore-6.q16-3-extra libmagickcore-6.q16-dev\n","  libmagickwand-6-headers libmagickwand-6.q16-3 libmagickwand-6.q16-dev\n","  libpixman-1-dev librsvg2-dev libwmf-dev libwmf0.2-7 libxcb-shm0-dev\n","  poppler-data\n","Suggested packages:\n","  fonts-noto ghostscript-x libcairo2-doc inkscape libjxr-tools librsvg2-doc\n","  libwmf-doc libwmf0.2-7-gtk poppler-utils fonts-japanese-mincho\n","  | fonts-ipafont-mincho fonts-japanese-gothic | fonts-ipafont-gothic\n","  fonts-arphic-ukai fonts-arphic-uming fonts-nanum\n","The following NEW packages will be installed:\n","  fonts-droid-fallback fonts-noto-mono ghostscript gir1.2-freedesktop\n","  gir1.2-gdkpixbuf-2.0 gir1.2-rsvg-2.0 gsfonts imagemagick-6-common\n","  libcairo-script-interpreter2 libcairo2-dev libcupsfilters1 libcupsimage2\n","  libdjvulibre-dev libdjvulibre-text libdjvulibre21 libgdk-pixbuf2.0-dev\n","  libgs9 libgs9-common libijs-0.35 libjbig2dec0 liblcms2-dev liblqr-1-0\n","  liblqr-1-0-dev libmagickcore-6-arch-config libmagickcore-6-headers\n","  libmagickcore-6.q16-3 libmagickcore-6.q16-3-extra libmagickcore-6.q16-dev\n","  libmagickwand-6-headers libmagickwand-6.q16-3 libmagickwand-6.q16-dev\n","  libmagickwand-dev libpixman-1-dev librsvg2-dev libwmf-dev libwmf0.2-7\n","  libxcb-shm0-dev poppler-data\n","0 upgraded, 38 newly installed, 0 to remove and 26 not upgraded.\n","Need to get 31.0 MB of archives.\n","After this operation, 91.0 MB of additional disk space will be used.\n","Get:1 http://archive.ubuntu.com/ubuntu bionic/main amd64 fonts-droid-fallback all 1:6.0.1r16-1.1 [1,805 kB]\n","Get:2 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 imagemagick-6-common all 8:6.9.7.4+dfsg-16ubuntu6.13 [60.3 kB]\n","Get:3 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 libmagickcore-6-arch-config amd64 8:6.9.7.4+dfsg-16ubuntu6.13 [25.3 kB]\n","Get:4 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 libmagickcore-6-headers all 8:6.9.7.4+dfsg-16ubuntu6.13 [47.2 kB]\n","Get:5 http://archive.ubuntu.com/ubuntu bionic/main amd64 liblqr-1-0 amd64 0.4.2-2.1 [27.7 kB]\n","Get:6 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 libmagickcore-6.q16-3 amd64 8:6.9.7.4+dfsg-16ubuntu6.13 [1,620 kB]\n","Get:7 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 libmagickwand-6-headers all 8:6.9.7.4+dfsg-16ubuntu6.13 [10.5 kB]\n","Get:8 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 libmagickwand-6.q16-3 amd64 8:6.9.7.4+dfsg-16ubuntu6.13 [292 kB]\n","Get:9 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 libdjvulibre-text all 3.5.27.1-8ubuntu0.4 [49.4 kB]\n","Get:10 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 libdjvulibre21 amd64 3.5.27.1-8ubuntu0.4 [561 kB]\n","Get:11 http://archive.ubuntu.com/ubuntu bionic/main amd64 libwmf0.2-7 amd64 0.2.8.4-12 [150 kB]\n","Get:12 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 libmagickcore-6.q16-3-extra amd64 8:6.9.7.4+dfsg-16ubuntu6.13 [62.3 kB]\n","Get:13 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 libdjvulibre-dev amd64 3.5.27.1-8ubuntu0.4 [2,382 kB]\n","Get:14 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 liblcms2-dev amd64 2.9-1ubuntu0.1 [9,096 kB]\n","Get:15 http://archive.ubuntu.com/ubuntu bionic/main amd64 liblqr-1-0-dev amd64 0.4.2-2.1 [69.1 kB]\n","Get:16 http://archive.ubuntu.com/ubuntu bionic/main amd64 gir1.2-freedesktop amd64 1.56.1-1 [9,080 B]\n","Get:17 http://archive.ubuntu.com/ubuntu bionic/main amd64 gir1.2-gdkpixbuf-2.0 amd64 2.36.11-2 [7,748 B]\n","Get:18 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 gir1.2-rsvg-2.0 amd64 2.40.20-2ubuntu0.2 [3,764 B]\n","Get:19 http://archive.ubuntu.com/ubuntu bionic/main amd64 libgdk-pixbuf2.0-dev amd64 2.36.11-2 [46.8 kB]\n","Get:20 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 libcairo-script-interpreter2 amd64 1.15.10-2ubuntu0.1 [53.5 kB]\n","Get:21 http://archive.ubuntu.com/ubuntu bionic/main amd64 libpixman-1-dev amd64 0.34.0-2 [244 kB]\n","Get:22 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 libxcb-shm0-dev amd64 1.13-2~ubuntu18.04 [6,684 B]\n","Get:23 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 libcairo2-dev amd64 1.15.10-2ubuntu0.1 [626 kB]\n","Get:24 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 librsvg2-dev amd64 2.40.20-2ubuntu0.2 [10.7 kB]\n","Get:25 http://archive.ubuntu.com/ubuntu bionic/main amd64 libwmf-dev amd64 0.2.8.4-12 [172 kB]\n","Get:26 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 libmagickcore-6.q16-dev amd64 8:6.9.7.4+dfsg-16ubuntu6.13 [974 kB]\n","Get:27 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 libmagickwand-6.q16-dev amd64 8:6.9.7.4+dfsg-16ubuntu6.13 [290 kB]\n","Get:28 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 libmagickwand-dev all 8:6.9.7.4+dfsg-16ubuntu6.13 [1,368 B]\n","Get:29 http://archive.ubuntu.com/ubuntu bionic/main amd64 poppler-data all 0.4.8-2 [1,479 kB]\n","Get:30 http://archive.ubuntu.com/ubuntu bionic/main amd64 fonts-noto-mono all 20171026-2 [75.5 kB]\n","Get:31 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 libcupsimage2 amd64 2.2.7-1ubuntu2.9 [18.6 kB]\n","Get:32 http://archive.ubuntu.com/ubuntu bionic/main amd64 libijs-0.35 amd64 0.35-13 [15.5 kB]\n","Get:33 http://archive.ubuntu.com/ubuntu bionic/main amd64 libjbig2dec0 amd64 0.13-6 [55.9 kB]\n","Get:34 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 libgs9-common all 9.26~dfsg+0-0ubuntu0.18.04.16 [5,093 kB]\n","Get:35 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 libgs9 amd64 9.26~dfsg+0-0ubuntu0.18.04.16 [2,265 kB]\n","Get:36 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 ghostscript amd64 9.26~dfsg+0-0ubuntu0.18.04.16 [51.3 kB]\n","Get:37 http://archive.ubuntu.com/ubuntu bionic/main amd64 gsfonts all 1:8.11+urwcyr1.0.7~pre44-4.4 [3,120 kB]\n","Get:38 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 libcupsfilters1 amd64 1.20.2-0ubuntu3.1 [108 kB]\n","Fetched 31.0 MB in 1s (48.0 MB/s)\n","Extracting templates from packages: 100%\n","Selecting previously unselected package fonts-droid-fallback.\n","(Reading database ... 155744 files and directories currently installed.)\n","Preparing to unpack .../00-fonts-droid-fallback_1%3a6.0.1r16-1.1_all.deb ...\n","Unpacking fonts-droid-fallback (1:6.0.1r16-1.1) ...\n","Selecting previously unselected package imagemagick-6-common.\n","Preparing to unpack .../01-imagemagick-6-common_8%3a6.9.7.4+dfsg-16ubuntu6.13_all.deb ...\n","Unpacking imagemagick-6-common (8:6.9.7.4+dfsg-16ubuntu6.13) ...\n","Selecting previously unselected package libmagickcore-6-arch-config:amd64.\n","Preparing to unpack .../02-libmagickcore-6-arch-config_8%3a6.9.7.4+dfsg-16ubuntu6.13_amd64.deb ...\n","Unpacking libmagickcore-6-arch-config:amd64 (8:6.9.7.4+dfsg-16ubuntu6.13) ...\n","Selecting previously unselected package libmagickcore-6-headers.\n","Preparing to unpack .../03-libmagickcore-6-headers_8%3a6.9.7.4+dfsg-16ubuntu6.13_all.deb ...\n","Unpacking libmagickcore-6-headers (8:6.9.7.4+dfsg-16ubuntu6.13) ...\n","Selecting previously unselected package liblqr-1-0:amd64.\n","Preparing to unpack .../04-liblqr-1-0_0.4.2-2.1_amd64.deb ...\n","Unpacking liblqr-1-0:amd64 (0.4.2-2.1) ...\n","Selecting previously unselected package libmagickcore-6.q16-3:amd64.\n","Preparing to unpack .../05-libmagickcore-6.q16-3_8%3a6.9.7.4+dfsg-16ubuntu6.13_amd64.deb ...\n","Unpacking libmagickcore-6.q16-3:amd64 (8:6.9.7.4+dfsg-16ubuntu6.13) ...\n","Selecting previously unselected package libmagickwand-6-headers.\n","Preparing to unpack .../06-libmagickwand-6-headers_8%3a6.9.7.4+dfsg-16ubuntu6.13_all.deb ...\n","Unpacking libmagickwand-6-headers (8:6.9.7.4+dfsg-16ubuntu6.13) ...\n","Selecting previously unselected package libmagickwand-6.q16-3:amd64.\n","Preparing to unpack .../07-libmagickwand-6.q16-3_8%3a6.9.7.4+dfsg-16ubuntu6.13_amd64.deb ...\n","Unpacking libmagickwand-6.q16-3:amd64 (8:6.9.7.4+dfsg-16ubuntu6.13) ...\n","Selecting previously unselected package libdjvulibre-text.\n","Preparing to unpack .../08-libdjvulibre-text_3.5.27.1-8ubuntu0.4_all.deb ...\n","Unpacking libdjvulibre-text (3.5.27.1-8ubuntu0.4) ...\n","Selecting previously unselected package libdjvulibre21:amd64.\n","Preparing to unpack .../09-libdjvulibre21_3.5.27.1-8ubuntu0.4_amd64.deb ...\n","Unpacking libdjvulibre21:amd64 (3.5.27.1-8ubuntu0.4) ...\n","Selecting previously unselected package libwmf0.2-7:amd64.\n","Preparing to unpack .../10-libwmf0.2-7_0.2.8.4-12_amd64.deb ...\n","Unpacking libwmf0.2-7:amd64 (0.2.8.4-12) ...\n","Selecting previously unselected package libmagickcore-6.q16-3-extra:amd64.\n","Preparing to unpack .../11-libmagickcore-6.q16-3-extra_8%3a6.9.7.4+dfsg-16ubuntu6.13_amd64.deb ...\n","Unpacking libmagickcore-6.q16-3-extra:amd64 (8:6.9.7.4+dfsg-16ubuntu6.13) ...\n","Selecting previously unselected package libdjvulibre-dev:amd64.\n","Preparing to unpack .../12-libdjvulibre-dev_3.5.27.1-8ubuntu0.4_amd64.deb ...\n","Unpacking libdjvulibre-dev:amd64 (3.5.27.1-8ubuntu0.4) ...\n","Selecting previously unselected package liblcms2-dev:amd64.\n","Preparing to unpack .../13-liblcms2-dev_2.9-1ubuntu0.1_amd64.deb ...\n","Unpacking liblcms2-dev:amd64 (2.9-1ubuntu0.1) ...\n","Selecting previously unselected package liblqr-1-0-dev:amd64.\n","Preparing to unpack .../14-liblqr-1-0-dev_0.4.2-2.1_amd64.deb ...\n","Unpacking liblqr-1-0-dev:amd64 (0.4.2-2.1) ...\n","Selecting previously unselected package gir1.2-freedesktop:amd64.\n","Preparing to unpack .../15-gir1.2-freedesktop_1.56.1-1_amd64.deb ...\n","Unpacking gir1.2-freedesktop:amd64 (1.56.1-1) ...\n","Selecting previously unselected package gir1.2-gdkpixbuf-2.0:amd64.\n","Preparing to unpack .../16-gir1.2-gdkpixbuf-2.0_2.36.11-2_amd64.deb ...\n","Unpacking gir1.2-gdkpixbuf-2.0:amd64 (2.36.11-2) ...\n","Selecting previously unselected package gir1.2-rsvg-2.0:amd64.\n","Preparing to unpack .../17-gir1.2-rsvg-2.0_2.40.20-2ubuntu0.2_amd64.deb ...\n","Unpacking gir1.2-rsvg-2.0:amd64 (2.40.20-2ubuntu0.2) ...\n","Selecting previously unselected package libgdk-pixbuf2.0-dev.\n","Preparing to unpack .../18-libgdk-pixbuf2.0-dev_2.36.11-2_amd64.deb ...\n","Unpacking libgdk-pixbuf2.0-dev (2.36.11-2) ...\n","Selecting previously unselected package libcairo-script-interpreter2:amd64.\n","Preparing to unpack .../19-libcairo-script-interpreter2_1.15.10-2ubuntu0.1_amd64.deb ...\n","Unpacking libcairo-script-interpreter2:amd64 (1.15.10-2ubuntu0.1) ...\n","Selecting previously unselected package libpixman-1-dev:amd64.\n","Preparing to unpack .../20-libpixman-1-dev_0.34.0-2_amd64.deb ...\n","Unpacking libpixman-1-dev:amd64 (0.34.0-2) ...\n","Selecting previously unselected package libxcb-shm0-dev:amd64.\n","Preparing to unpack .../21-libxcb-shm0-dev_1.13-2~ubuntu18.04_amd64.deb ...\n","Unpacking libxcb-shm0-dev:amd64 (1.13-2~ubuntu18.04) ...\n","Selecting previously unselected package libcairo2-dev:amd64.\n","Preparing to unpack .../22-libcairo2-dev_1.15.10-2ubuntu0.1_amd64.deb ...\n","Unpacking libcairo2-dev:amd64 (1.15.10-2ubuntu0.1) ...\n","Selecting previously unselected package librsvg2-dev:amd64.\n","Preparing to unpack .../23-librsvg2-dev_2.40.20-2ubuntu0.2_amd64.deb ...\n","Unpacking librsvg2-dev:amd64 (2.40.20-2ubuntu0.2) ...\n","Selecting previously unselected package libwmf-dev.\n","Preparing to unpack .../24-libwmf-dev_0.2.8.4-12_amd64.deb ...\n","Unpacking libwmf-dev (0.2.8.4-12) ...\n","Selecting previously unselected package libmagickcore-6.q16-dev:amd64.\n","Preparing to unpack .../25-libmagickcore-6.q16-dev_8%3a6.9.7.4+dfsg-16ubuntu6.13_amd64.deb ...\n","Unpacking libmagickcore-6.q16-dev:amd64 (8:6.9.7.4+dfsg-16ubuntu6.13) ...\n","Selecting previously unselected package libmagickwand-6.q16-dev:amd64.\n","Preparing to unpack .../26-libmagickwand-6.q16-dev_8%3a6.9.7.4+dfsg-16ubuntu6.13_amd64.deb ...\n","Unpacking libmagickwand-6.q16-dev:amd64 (8:6.9.7.4+dfsg-16ubuntu6.13) ...\n","Selecting previously unselected package libmagickwand-dev.\n","Preparing to unpack .../27-libmagickwand-dev_8%3a6.9.7.4+dfsg-16ubuntu6.13_all.deb ...\n","Unpacking libmagickwand-dev (8:6.9.7.4+dfsg-16ubuntu6.13) ...\n","Selecting previously unselected package poppler-data.\n","Preparing to unpack .../28-poppler-data_0.4.8-2_all.deb ...\n","Unpacking poppler-data (0.4.8-2) ...\n","Selecting previously unselected package fonts-noto-mono.\n","Preparing to unpack .../29-fonts-noto-mono_20171026-2_all.deb ...\n","Unpacking fonts-noto-mono (20171026-2) ...\n","Selecting previously unselected package libcupsimage2:amd64.\n","Preparing to unpack .../30-libcupsimage2_2.2.7-1ubuntu2.9_amd64.deb ...\n","Unpacking libcupsimage2:amd64 (2.2.7-1ubuntu2.9) ...\n","Selecting previously unselected package libijs-0.35:amd64.\n","Preparing to unpack .../31-libijs-0.35_0.35-13_amd64.deb ...\n","Unpacking libijs-0.35:amd64 (0.35-13) ...\n","Selecting previously unselected package libjbig2dec0:amd64.\n","Preparing to unpack .../32-libjbig2dec0_0.13-6_amd64.deb ...\n","Unpacking libjbig2dec0:amd64 (0.13-6) ...\n","Selecting previously unselected package libgs9-common.\n","Preparing to unpack .../33-libgs9-common_9.26~dfsg+0-0ubuntu0.18.04.16_all.deb ...\n","Unpacking libgs9-common (9.26~dfsg+0-0ubuntu0.18.04.16) ...\n","Selecting previously unselected package libgs9:amd64.\n","Preparing to unpack .../34-libgs9_9.26~dfsg+0-0ubuntu0.18.04.16_amd64.deb ...\n","Unpacking libgs9:amd64 (9.26~dfsg+0-0ubuntu0.18.04.16) ...\n","Selecting previously unselected package ghostscript.\n","Preparing to unpack .../35-ghostscript_9.26~dfsg+0-0ubuntu0.18.04.16_amd64.deb ...\n","Unpacking ghostscript (9.26~dfsg+0-0ubuntu0.18.04.16) ...\n","Selecting previously unselected package gsfonts.\n","Preparing to unpack .../36-gsfonts_1%3a8.11+urwcyr1.0.7~pre44-4.4_all.deb ...\n","Unpacking gsfonts (1:8.11+urwcyr1.0.7~pre44-4.4) ...\n","Selecting previously unselected package libcupsfilters1:amd64.\n","Preparing to unpack .../37-libcupsfilters1_1.20.2-0ubuntu3.1_amd64.deb ...\n","Unpacking libcupsfilters1:amd64 (1.20.2-0ubuntu3.1) ...\n","Setting up libgs9-common (9.26~dfsg+0-0ubuntu0.18.04.16) ...\n","Setting up imagemagick-6-common (8:6.9.7.4+dfsg-16ubuntu6.13) ...\n","Setting up fonts-droid-fallback (1:6.0.1r16-1.1) ...\n","Setting up liblcms2-dev:amd64 (2.9-1ubuntu0.1) ...\n","Setting up gsfonts (1:8.11+urwcyr1.0.7~pre44-4.4) ...\n","Setting up libcairo-script-interpreter2:amd64 (1.15.10-2ubuntu0.1) ...\n","Setting up poppler-data (0.4.8-2) ...\n","Setting up libdjvulibre-text (3.5.27.1-8ubuntu0.4) ...\n","Setting up gir1.2-freedesktop:amd64 (1.56.1-1) ...\n","Setting up libxcb-shm0-dev:amd64 (1.13-2~ubuntu18.04) ...\n","Setting up libmagickcore-6-arch-config:amd64 (8:6.9.7.4+dfsg-16ubuntu6.13) ...\n","Setting up gir1.2-gdkpixbuf-2.0:amd64 (2.36.11-2) ...\n","Setting up fonts-noto-mono (20171026-2) ...\n","Setting up libcupsfilters1:amd64 (1.20.2-0ubuntu3.1) ...\n","Setting up libcupsimage2:amd64 (2.2.7-1ubuntu2.9) ...\n","Setting up liblqr-1-0:amd64 (0.4.2-2.1) ...\n","Setting up libjbig2dec0:amd64 (0.13-6) ...\n","Setting up libgdk-pixbuf2.0-dev (2.36.11-2) ...\n","Setting up libpixman-1-dev:amd64 (0.34.0-2) ...\n","Setting up gir1.2-rsvg-2.0:amd64 (2.40.20-2ubuntu0.2) ...\n","Setting up libijs-0.35:amd64 (0.35-13) ...\n","Setting up libmagickcore-6-headers (8:6.9.7.4+dfsg-16ubuntu6.13) ...\n","Setting up libgs9:amd64 (9.26~dfsg+0-0ubuntu0.18.04.16) ...\n","Setting up libwmf0.2-7:amd64 (0.2.8.4-12) ...\n","Setting up libwmf-dev (0.2.8.4-12) ...\n","Setting up libmagickcore-6.q16-3:amd64 (8:6.9.7.4+dfsg-16ubuntu6.13) ...\n","Setting up liblqr-1-0-dev:amd64 (0.4.2-2.1) ...\n","Setting up libcairo2-dev:amd64 (1.15.10-2ubuntu0.1) ...\n","Setting up libdjvulibre21:amd64 (3.5.27.1-8ubuntu0.4) ...\n","Setting up libmagickwand-6-headers (8:6.9.7.4+dfsg-16ubuntu6.13) ...\n","Setting up libdjvulibre-dev:amd64 (3.5.27.1-8ubuntu0.4) ...\n","Setting up librsvg2-dev:amd64 (2.40.20-2ubuntu0.2) ...\n","Setting up ghostscript (9.26~dfsg+0-0ubuntu0.18.04.16) ...\n","Setting up libmagickwand-6.q16-3:amd64 (8:6.9.7.4+dfsg-16ubuntu6.13) ...\n","Setting up libmagickcore-6.q16-3-extra:amd64 (8:6.9.7.4+dfsg-16ubuntu6.13) ...\n","Setting up libmagickcore-6.q16-dev:amd64 (8:6.9.7.4+dfsg-16ubuntu6.13) ...\n","Setting up libmagickwand-6.q16-dev:amd64 (8:6.9.7.4+dfsg-16ubuntu6.13) ...\n","Setting up libmagickwand-dev (8:6.9.7.4+dfsg-16ubuntu6.13) ...\n","Processing triggers for man-db (2.8.3-2ubuntu0.1) ...\n","Processing triggers for fontconfig (2.12.6-0ubuntu2) ...\n","Processing triggers for libc-bin (2.27-3ubuntu1.5) ...\n","/sbin/ldconfig.real: file /usr/local/lib/libjawt.so is truncated\n","\n","/sbin/ldconfig.real: file /usr/local/lib/libnet.so is truncated\n","\n","/sbin/ldconfig.real: file /usr/local/lib/libjli.so is truncated\n","\n"]}],"source":["! apt-get install libmagickwand-dev"]},{"cell_type":"code","execution_count":80,"metadata":{"id":"3HS96PJ45f26","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1661184374848,"user_tz":240,"elapsed":21,"user":{"displayName":"Theodore Maximilian Nelson","userId":"03436156380930975060"}},"outputId":"6e967fed-c72b-411b-f41c-056752b3aae5"},"outputs":[{"output_type":"stream","name":"stdout","text":["cp: cannot stat '/content/long-read-sequencing-pipeline/svist4get/policy_revised.xml': No such file or directory\n"]}],"source":["! cp -r $PIPELINE_FILE_PATH/svist4get/policy_revised.xml /etc/ImageMagick-6/policy.xml"]},{"cell_type":"code","execution_count":81,"metadata":{"id":"ZNVRJydu1wIV","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1661184383380,"user_tz":240,"elapsed":8545,"user":{"displayName":"Theodore Maximilian Nelson","userId":"03436156380930975060"}},"outputId":"1e5259e8-4bc0-474d-9a7c-15a6f0395edd"},"outputs":[{"output_type":"stream","name":"stdout","text":["Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/\n","Collecting svist4get\n","  Downloading svist4get-1.3-py3-none-any.whl (5.8 MB)\n","\u001b[K     |████████████████████████████████| 5.8 MB 13.8 MB/s \n","\u001b[?25hCollecting reportlab\n","  Downloading reportlab-3.6.11-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (2.8 MB)\n","\u001b[K     |████████████████████████████████| 2.8 MB 65.1 MB/s \n","\u001b[?25hCollecting wand\n","  Downloading Wand-0.6.10-py2.py3-none-any.whl (142 kB)\n","\u001b[K     |████████████████████████████████| 142 kB 47.2 MB/s \n","\u001b[?25hRequirement already satisfied: Pybedtools in /usr/local/lib/python3.9/site-packages (from svist4get) (0.9.0)\n","Collecting statistics\n","  Downloading statistics-1.0.3.5.tar.gz (8.3 kB)\n","Collecting argparse\n","  Downloading argparse-1.4.0-py2.py3-none-any.whl (23 kB)\n","Collecting biopython\n","  Downloading biopython-1.79-cp39-cp39-manylinux_2_5_x86_64.manylinux1_x86_64.whl (2.3 MB)\n","\u001b[K     |████████████████████████████████| 2.3 MB 47.4 MB/s \n","\u001b[?25hCollecting configs\n","  Downloading configs-3.0.3-py3-none-any.whl (7.1 kB)\n","Collecting pillow>=9.0.0\n","  Downloading Pillow-9.2.0-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (3.1 MB)\n","\u001b[K     |████████████████████████████████| 3.1 MB 57.3 MB/s \n","\u001b[?25hRequirement already satisfied: pysam in /usr/local/lib/python3.9/site-packages (from Pybedtools->svist4get) (0.19.1)\n","Requirement already satisfied: six in /usr/local/lib/python3.9/site-packages (from Pybedtools->svist4get) (1.16.0)\n","Requirement already satisfied: docutils>=0.3 in /usr/local/lib/python3.9/site-packages (from statistics->svist4get) (0.16)\n","Requirement already satisfied: numpy in /usr/local/lib/python3.9/site-packages (from biopython->svist4get) (1.22.3)\n","Building wheels for collected packages: statistics\n","  Building wheel for statistics (setup.py) ... \u001b[?25l\u001b[?25hdone\n","  Created wheel for statistics: filename=statistics-1.0.3.5-py3-none-any.whl size=7439 sha256=c97330605cd0a8daeeaa41f6c4e9676667abb8cc11e39ad0bbcbe3d0622eaa76\n","  Stored in directory: /root/.cache/pip/wheels/26/3c/70/9467407f3aa90862061eadcd286627b23a8bab6789b667776f\n","Successfully built statistics\n","Installing collected packages: pillow, reportlab, wand, statistics, argparse, biopython, configs, svist4get\n","Successfully installed argparse-1.4.0 biopython-1.79 configs-3.0.3 pillow-9.2.0 reportlab-3.6.11 statistics-1.0.3.5 svist4get-1.3 wand-0.6.10\n"]}],"source":["! python3 -m pip install svist4get"]},{"cell_type":"markdown","metadata":{"id":"S9r4MCB0lvAf"},"source":["##svist4get: a simple visualization tool for genomic tracks from sequencing experiments - usage"]},{"cell_type":"markdown","metadata":{"id":"3T2Y3dcAnaWu"},"source":["svist4get allows you to view read coverage at a defined region on a chromosome (documentation: https://bitbucket.org/artegorov/svist4get/src/master/)"]},{"cell_type":"code","execution_count":83,"metadata":{"id":"ehHeatJQCZ42","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1661184477400,"user_tz":240,"elapsed":38230,"user":{"displayName":"Theodore Maximilian Nelson","userId":"03436156380930975060"}},"outputId":"5ff21474-c911-4477-98a2-6198e8f97ed5"},"outputs":[{"output_type":"stream","name":"stdout","text":["tcmalloc: large alloc 1561239552 bytes == 0x5604f94ba000 @  0x7f68545d8887 0x5604f7fd653a 0x5604f7fd5659 0x5604f7fd613c 0x5604f7fda960 0x5604f7f10ab3 0x7f6853c81c87 0x5604f7f1548a\n","tcmalloc: large alloc 1454047232 bytes == 0x5604f94ba000 @  0x7f68545d8887 0x5604f7fd653a 0x5604f7fd5659 0x5604f7fd613c 0x5604f7fda960 0x5604f7f10ab3 0x7f6853c81c87 0x5604f7f1548a\n"]}],"source":["! bedtools genomecov -ibam $PIPELINE_FILE_PATH/long-read-sequencing-pipeline/bam/$ACC.sorted.bam -bg > $PIPELINE_FILE_PATH/long-read-sequencing-pipeline/bed/$ACC.sorted.bedgraph"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"OpQMW3gI15ZI"},"outputs":[],"source":["! eval svist4get -bg $PIPELINE_FILE_PATH/bed/$ACC.sorted.bedgraph -gtf $ANNOTATION_FILE_PATH -fa $INDEX_FILE_PATH -bl Long-Read Coverage -w $CHROMOSOME $CHROMOSOME_START $CHROMOSOME_FINISH -it \"$REGION_NAME\" -o $PIPELINE_FILE_PATH/svist4get/$ACC"]},{"cell_type":"markdown","metadata":{"id":"5CwywN4tc74s"},"source":["##Pistis: Quality control plotting for long reads - installation"]},{"cell_type":"markdown","metadata":{"id":"ckgZcyKcpFyC"},"source":["Pistis generates long-read specific quality control graphs, including a plot demonstrating read alignment percentage to the reference genome (documentation: https://github.com/mbhall88/pistis)"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"S5mHHUr1d50q"},"outputs":[],"source":["! pip3 install pistis"]},{"cell_type":"markdown","metadata":{"id":"MjZ_tYz4pEHX"},"source":["##Pistis: Quality control plotting for long reads - usage"]},{"cell_type":"markdown","metadata":{"id":"RFhhJITCphA_"},"source":["Pistis generates long-read specific quality control graphs, including a plot demonstrating read alignment percentage to the reference genome (documentation: https://github.com/mbhall88/pistis)"]},{"cell_type":"markdown","metadata":{"id":"B5Zv-aSzVPwK"},"source":["Please note that the report generates assuming alignment of more than 50,000 reads. If this is not the case, please add the flag `--downsample INTEGER` replacing `INTEGER` with a number less than the number of aligned reads. "]},{"cell_type":"code","execution_count":null,"metadata":{"id":"ik-boaBxeNjP"},"outputs":[],"source":["! pistis -f $PIPELINE_FILE_PATH/fastq/$ACC.fastq -b $PIPELINE_FILE_PATH/bam/$ACC.sorted.bam  -o $PIPELINE_FILE_PATH/pistis/$ACC.pdf"]},{"cell_type":"markdown","metadata":{"id":"EQn9Ikb3YkFC"},"source":["##MakeHub: Fully automated generation of UCSC assembly hubs - installation"]},{"cell_type":"markdown","metadata":{"id":"X7J1qXU1bmXd"},"source":["MakeHub is a command line tool for the fully automatic generation of of track data hubs for visualizing genomes with the UCSC genome browser (documentation: https://github.com/Gaius-Augustus/MakeHub)."]},{"cell_type":"code","execution_count":null,"metadata":{"id":"_eYWwdsmdsBa"},"outputs":[],"source":["! python3.7 -m pip install biopython"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"Fu1IcmMbdsxV"},"outputs":[],"source":["! sudo apt install samtools"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"yudzuOTW2IQ-"},"outputs":[],"source":["! sudo apt install augustus augustus-data augustus-doc"]},{"cell_type":"markdown","metadata":{"id":"POLtVVCEKndf"},"source":["##MakeHub: Fully automated generation of UCSC assembly hubs - usage"]},{"cell_type":"markdown","metadata":{"id":"ZcW7zXOoKrCy"},"source":["MakeHub is a command line tool for the fully automatic generation of of track data hubs for visualizing genomes with the UCSC genome browser (documentation: https://github.com/Gaius-Augustus/MakeHub)."]},{"cell_type":"code","execution_count":null,"metadata":{"id":"fMN2TYdniufR"},"outputs":[],"source":["! chmod 755 $PIPELINE_FILE_PATH/makehub/make_hub.py"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"pVA2mrrjZ2yM"},"outputs":[],"source":["! eval $PIPELINE_FILE_PATH/makehub/make_hub.py -l $HUB_KEYWORD -L $HUB_NAME -g $INDEX_FILE_PATH -e \\\n","  $HUB_EMAIL -a $ANNOTATION_FILE_PATH -b $PIPELINE_FILE_PATH/bam/$ACC.sorted.bam -o $PIPELINE_FILE_PATH/makehub/"]},{"cell_type":"markdown","metadata":{"id":"kpfzCCTKuqMt"},"source":["##MultiQC: Aggregate results from bioinformatics analyses across many samples into a single report - installation\n","\n"]},{"cell_type":"markdown","metadata":{"id":"Jluk65BgNoEJ"},"source":["MultiQC is a program which allows you to combine reports for as many samples as you wish ([documentation](https://multiqc.info/docs/))"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"B6OuaTChNs4J"},"outputs":[],"source":["! pip install multiqc"]},{"cell_type":"markdown","metadata":{"id":"m5pU9i4DvGxQ"},"source":["##MultiQC: Aggregate results from bioinformatics analyses across many samples into a single report - usage"]},{"cell_type":"markdown","metadata":{"id":"sPB5udNzvVdJ"},"source":["MultiQC is a program which allows you to combine reports for as many samples as you wish ([documentation](https://multiqc.info/docs/))"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"SEBsMVQ5OG4U"},"outputs":[],"source":["! multiqc $PIPELINE_FILE_PATH -o $PIPELINE_FILE_PATH/multiqc"]},{"cell_type":"markdown","metadata":{"id":"pquILHZedOAx"},"source":["#$\\text{L-RAP: Long Read RNA Sequencing Analysis Pipeline - Pro Features}$\n"]},{"cell_type":"markdown","metadata":{"id":"gyh52FxVTSQt"},"source":["##minimap2: A versatile pairwise aligner for genomic and spliced nucleotide sequences (Google Colab Pro required) - installation"]},{"cell_type":"markdown","metadata":{"id":"0bXGX4mEeegN"},"source":["minimap2 is a long-read sequencing aligner (documentation: https://github.com/lh3/minimap2). "]},{"cell_type":"code","execution_count":null,"metadata":{"id":"yZz8ktFYTQ1F"},"outputs":[],"source":["! conda install -c bioconda minimap2 -y"]},{"cell_type":"markdown","metadata":{"id":"f8EfwQZze29B"},"source":["##minimap2: A versatile pairwise aligner for genomic and spliced nucleotide sequences (Google Colab Pro required) - usage"]},{"cell_type":"markdown","metadata":{"id":"xlj72R1ofOAz"},"source":["minimap2 is a long-read sequencing aligner (documentation: https://github.com/lh3/minimap2). "]},{"cell_type":"markdown","metadata":{"id":"nZvCDgAqY5dZ"},"source":["Google Colab will typically provide virtual machines with 12 GB of RAM. minimap2 in the case of the human genome usually requires more than 12 GB of RAM. Therefore this command will fail to execute within the free version of Google Colab. However, this pipeline also implements a less memory-intensive version of minimap2 for free tier users. "]},{"cell_type":"code","execution_count":null,"metadata":{"id":"YjPnVj5PTRtK"},"outputs":[],"source":["! eval minimap2 -ax splice $INDEX_FILE_PATH $PIPELINE_FILE_PATH/fastq/$ACC.fastq > $PIPELINE_FILE_PATH/sam/$ACC.sam"]},{"cell_type":"markdown","metadata":{"id":"Wj85WdDNwraY"},"source":["##TAMA: Transcriptome Annotation by Modular Algorithms (Google Colab Pro required) - installation\n","\n"]},{"cell_type":"markdown","metadata":{"id":"bAO18hfruXPm"},"source":["TAMA allows for the construction of a polished transcriptome by collapsing runs into individual transcripts (documentation: https://github.com/GenomeRIK/tama/wiki). "]},{"cell_type":"code","execution_count":null,"metadata":{"id":"GKv-Sz-IwsXj"},"outputs":[],"source":["! git clone https://github.com/MakeTheBrainHappy/tama"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"FqjqBPkgkLfX"},"outputs":[],"source":["! python -m pip install Bio"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"F4vJnRJCkVhy"},"outputs":[],"source":["! pip install pysam"]},{"cell_type":"markdown","metadata":{"id":"F8ECo-DYue_P"},"source":["##TAMA: Transcriptome Annotation by Modular Algorithms (Google Colab Pro required) - usage"]},{"cell_type":"markdown","metadata":{"id":"8kgDVWffuslu"},"source":["TAMA allows for the construction of a polished transcriptome by collapsing runs into individual transcripts (documentation: https://github.com/GenomeRIK/tama/wiki). "]},{"cell_type":"code","execution_count":null,"metadata":{"id":"wsqCqeIqwsbH"},"outputs":[],"source":["! eval python tama/tama_collapse.py -s $PIPELINE_FILE_PATH/bam/$ACC.sorted.bam -f $INDEX_FILE_PATH -p $PIPELINE_FILE_PATH/tama/$ACC.polished -b BAM -rm low_mem"]}],"metadata":{"colab":{"collapsed_sections":["RrTcAlHtbWfg","24vu4E_8vbKa","z1lysGn4hO-i","oBdBA_CDdE0z","_MislmkXZcz-","F0hFNwmcv5qr","SQb5HTiwtfMn","NOrgEp-fbtBt","7uHwZ4Zj1vgC","1mvZC8IsijTg","dxspwFM3_1qN","sjns1XbAkILI","swcyphLAvE52","Y9qrP4Ip3xwz","zqQcBdxa4X7q","iUwbIIRkwWrL","yhR96h7O5CHN","7yVbaSBc5cu7","a2s7KTWRw3ry","npZwllsupfEJ","S9r4MCB0lvAf","5CwywN4tc74s","MjZ_tYz4pEHX","EQn9Ikb3YkFC","POLtVVCEKndf","kpfzCCTKuqMt","m5pU9i4DvGxQ","pquILHZedOAx","gyh52FxVTSQt","f8EfwQZze29B","Wj85WdDNwraY","F8ECo-DYue_P"],"machine_shape":"hm","name":"long_read_rna_seq_analysis_prebuilt_indices.ipynb","provenance":[]},"kernelspec":{"display_name":"Python 3","name":"python3"},"language_info":{"name":"python"}},"nbformat":4,"nbformat_minor":0}