{"cells":[{"cell_type":"markdown","metadata":{"id":"P5k_WL74Y2Ei"},"source":["#Long Read RNA Sequencing Analysis Pipeline (Requires Pre-Built Indices)\n"]},{"cell_type":"markdown","metadata":{"id":"IaygDNNEZnph"},"source":["A pipeline to analyze Oxford Nanopore and PacBio third-generation long transcriptomic sequencing reads"]},{"cell_type":"markdown","metadata":{"id":"hIjUpSQCZlw1"},"source":["*Theodore Nelson (tmn2126)* - Columbia University Irving Medical Center\n"]},{"cell_type":"markdown","metadata":{"id":"AUKHTlobUcf_"},"source":["##Parameter Input and User Instructions"]},{"cell_type":"markdown","source":["Please define where the file structure is within your Google Drive:"],"metadata":{"id":"oM-NaZDK63YC"}},{"cell_type":"markdown","source":["<ul type=disc>\n","<li><b>PIPELINE_FILE_PATH</b>: file path to location of long-read RNA sequencing analysis pipeline within your Google Drive/general file system - required for most applications. This must be defined and initialized first.</li>\n","</ul>"],"metadata":{"id":"uc1rzY2x6tvu"}},{"cell_type":"code","source":["%env PIPELINE_FILE_PATH=/content/drive/MyDrive/long-read-sequencing-pipeline"],"metadata":{"id":"_P4OAFNrzifE"},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":["Please modify the following parameters within the code box below to fit your own study requirements:  "],"metadata":{"id":"2USxlBU47A0L"}},{"cell_type":"markdown","metadata":{"id":"f99vaAE0Z0AG"},"source":["<li><b>ACC</b>: Run accession number for reads within the [European Nucleotide Archive](https://www.ebi.ac.uk/ena/browser/) (SRR...) or file path to location of long-read RNA sequencing data within your Google Drive/general file system - required for most applications</li>\n","<li><b>PLATFORM_FREE</b>: choose either `map-ont` (Nanopore reads) or `map-pac` (PacBio reads) - required for minimap2 featherweight version</li>\n","<li><b>INDEX_FILE_PATH</b>: file path to location of reference genome (e.g. .FASTA) within your Google Drive/general file system - required for most applications</li>\n","<li><b>ANNOTATION_FILE_PATH</b>: file path to location of reference annotation (e.g. .GTF) within your Google Drive/general file system - required for most applications</li>\n","<li><b>PARTITIONED_INDEX_FILE_PATH</b>: file path to location of partioned reference annotation (e.g. .idx) within your Google Drive/general file system - required for minimap2 featherweight version</li>\n","<li><b>CHROMOSOME</b>: Name of Chromosome of Interest matching the name of the Chromosome within your Reference Annotation - required for svist4get</li>\n","<li><b>CHROMOSOME_START</b>: Starting Location of Interest on the Chromosome - required for svist4get</li>\n","<li><b>CHROMOSOME_FINISH</b>: Ending Location of Interest on the Chromosome - required for svist4get</li>\n","<li><b>REGION_NAME</b>: Gene Name (does not need to match annotation file) - required for svist4get</li>\n","<li><b>HUB_KEYWORD</b>: Short Keyword for your UCSC Track Hub - required for MakeHub</li>\n","<li><b>HUB_NAME</b>: Longer Title for your UCSC Track Hub - required for MakeHub</li>\n","<li><b>HUB_EMAIL</b>: Email for your UCSC Track Hub (if you publish your track hub then this email will be public) - required for MakeHub</li>\n"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"tqGM1OFhUcoo"},"outputs":[],"source":["%env ACC=ERR1951293  \n","%env PLATFORM_FREE=map-ont\n","%env INDEX_FILE_PATH=${PIPELINE_FILE_PATH}/prebuilt_indices/mm39.fa\n","%env ANNOTATION_FILE_PATH=${PIPELINE_FILE_PATH}/prebuilt_indices/mm39.ncbiRefSeq.gtf\n","%env PARTITIONED_INDEX_FILE_PATH=${PIPELINE_FILE_PATH}/prebuilt_indices/mm39.idx\n","%env CHROMOSOME=chr12 \n","%env CHROMOSOME_START=116533435 \n","%env CHROMOSOME_FINISH=116536513\n","%env REGION_NAME=LINC00173\n","%env HUB_KEYWORD=lnc173\n","%env HUB_NAME=\"293T LINC00173\"\n","%env HUB_EMAIL=tmn2126@columbia.edu"]},{"cell_type":"markdown","source":["##Mounting your Google Drive"],"metadata":{"id":"RrTcAlHtbWfg"}},{"cell_type":"markdown","source":["This step allows for permeanent storage of your bioinformatics analysis in Google Drive"],"metadata":{"id":"C5_iJ_KkbYt3"}},{"cell_type":"code","source":["from google.colab import drive\n","drive.mount('/content/drive')"],"metadata":{"id":"aQOezcA5ZrFA"},"execution_count":null,"outputs":[]},{"cell_type":"markdown","metadata":{"id":"24vu4E_8vbKa"},"source":["##Managing Software via BioConda"]},{"cell_type":"markdown","metadata":{"id":"lz3I3YU06E6z"},"source":["BioConda is a software environment and package manager, providing acess to over 8,000 different software packages related to bioinformatics (documentation: [BioConda](https://bioconda.github.io/user/install.html) and [Managing Environments via Conda](https://docs.conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html)). "]},{"cell_type":"code","execution_count":null,"metadata":{"id":"oxEPAO-_wpbz"},"outputs":[],"source":["! wget https://repo.anaconda.com/miniconda/Miniconda3-py37_4.8.2-Linux-x86_64.sh\n","! chmod +x Miniconda3-py37_4.8.2-Linux-x86_64.sh\n","! bash ./Miniconda3-py37_4.8.2-Linux-x86_64.sh -b -f -p /usr/local\n","import sys\n","sys.path.append('/usr/local/lib/python3.7/site-packages/')"]},{"cell_type":"markdown","source":["##Kingfisher: fast and flexible program for procurement of sequence files - installation "],"metadata":{"id":"z1lysGn4hO-i"}},{"cell_type":"markdown","source":["The Kingfisher program allows for sequence files to be downloaded from the European Nucleotide Archive (documentation: https://github.com/wwood/kingfisher-download)."],"metadata":{"id":"X4iuuuqPkKcT"}},{"cell_type":"code","execution_count":null,"metadata":{"id":"_Kn0KYlyOi2r"},"outputs":[],"source":["! git clone https://github.com/MakeTheBrainHappy/kingfisher-download"]},{"cell_type":"code","source":["! conda env update -n base --file kingfisher-download/kingfisher.yml"],"metadata":{"id":"NNtGMks8OkmZ"},"execution_count":null,"outputs":[]},{"cell_type":"code","execution_count":null,"metadata":{"id":"fDjCnO4C3NQu"},"outputs":[],"source":["! conda install -c rpetit3 aspera-connect -y"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"ksJvQOCtx8BJ"},"outputs":[],"source":["! wget -qO- https://download.asperasoft.com/download/sw/connect/3.9.8/ibm-aspera-connect-3.9.8.176272-linux-g2.12-64.tar.gz | tar xvz"]},{"cell_type":"markdown","source":["this command will pop up with an error message; please disregard"],"metadata":{"id":"RHjWyX6Ei_jT"}},{"cell_type":"code","execution_count":null,"metadata":{"id":"anXSgAYa3xQU"},"outputs":[],"source":["! ./ibm-aspera-connect-3.9.8.176272-linux-g2.12-64.sh"]},{"cell_type":"markdown","source":["##Kingfisher: fast and flexible program for procurement of sequence files - usage \n"],"metadata":{"id":"oBdBA_CDdE0z"}},{"cell_type":"markdown","source":["The Kingfisher program allows for sequence files to be downloaded from the European Nucleotide Archive (documentation: https://github.com/wwood/kingfisher-download)."],"metadata":{"id":"IE4T9swGdQY-"}},{"cell_type":"code","source":["! cd $PIPELINE_FILE_PATH/fastq ; /content/kingfisher-download/bin/kingfisher get -r $ACC -m ena-ascp aws-http prefetch"],"metadata":{"id":"ReZdoTfnd5uQ"},"execution_count":null,"outputs":[]},{"cell_type":"code","source":["! cd $PIPELINE_FILE_PATH/fastq ; gunzip *.gz"],"metadata":{"id":"wwpiZ9OU6WPn"},"execution_count":null,"outputs":[]},{"cell_type":"code","source":["! cd $PIPELINE_FILE_PATH/fastq ; mv ${ACC}_1.fastq $ACC.fastq"],"metadata":{"id":"4pmkDR2x6SzT"},"execution_count":null,"outputs":[]},{"cell_type":"markdown","metadata":{"id":"F0hFNwmcv5qr"},"source":["##FastQC: A quality control tool for high throughput sequence data - installation"]},{"cell_type":"markdown","metadata":{"id":"68fqL9EGNOKE"},"source":["FastQC is a program designed to spot potential problems in high througput sequencing datasets ([documentation](https://github.com/s-andrews/FastQC))."]},{"cell_type":"code","metadata":{"id":"_bRfEf9e3Sp-"},"source":["! conda install -c bioconda fastqc -y"],"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":["##FastQC: A quality control tool for high throughput sequence data - usage"],"metadata":{"id":"SQb5HTiwtfMn"}},{"cell_type":"markdown","source":["FastQC is a program designed to spot potential problems in high througput sequencing datasets ([documentation](https://github.com/s-andrews/FastQC))."],"metadata":{"id":"ZoutKrvVtjAY"}},{"cell_type":"code","metadata":{"id":"bTYA5m273eVN"},"source":["! fastqc $PIPELINE_FILE_PATH/fastq/$ACC.fastq --outdir $PIPELINE_FILE_PATH/fastqc"],"execution_count":null,"outputs":[]},{"cell_type":"markdown","metadata":{"id":"gyh52FxVTSQt"},"source":["##minimap2: A versatile pairwise aligner for genomic and spliced nucleotide sequences (Google Colab Pro required) - installation"]},{"cell_type":"markdown","metadata":{"id":"0bXGX4mEeegN"},"source":["minimap2 is a long-read sequencing aligner (documentation: https://github.com/lh3/minimap2). "]},{"cell_type":"code","execution_count":null,"metadata":{"id":"yZz8ktFYTQ1F"},"outputs":[],"source":["! conda install -c bioconda minimap2 -y"]},{"cell_type":"markdown","source":["##minimap2: A versatile pairwise aligner for genomic and spliced nucleotide sequences (Google Colab Pro required) - usage"],"metadata":{"id":"f8EfwQZze29B"}},{"cell_type":"markdown","metadata":{"id":"xlj72R1ofOAz"},"source":["minimap2 is a long-read sequencing aligner (documentation: https://github.com/lh3/minimap2). "]},{"cell_type":"markdown","metadata":{"id":"nZvCDgAqY5dZ"},"source":["Google Colab will typically provide virtual machines with 12 GB of RAM. minimap2 in the case of the human genome usually requires more than 12 GB of RAM. Therefore this command will fail to execute within the free version of Google Colab. However, this pipeline also implements a less memory-intensive version of minimap2 for free tier users. "]},{"cell_type":"code","execution_count":null,"metadata":{"id":"YjPnVj5PTRtK"},"outputs":[],"source":["! eval minimap2 -ax splice $INDEX_FILE_PATH $PIPELINE_FILE_PATH/fastq/$ACC.fastq > $PIPELINE_FILE_PATH/sam/$ACC.sam"]},{"cell_type":"markdown","metadata":{"id":"1mvZC8IsijTg"},"source":["##minimap2 featherweight alignment - installation"]},{"cell_type":"markdown","metadata":{"id":"_Z4pLWqTe6K7"},"source":["The minimap2 featherweight implementation works by splitting the reference genome into four parts with minimal loss in alignment accuracy (documentation: https://doi.org/10.1038/s41598-019-40739-8)."]},{"cell_type":"code","execution_count":null,"metadata":{"id":"0FCbrpo9ijfj"},"outputs":[],"source":["! wget https://github.com/hasindu2008/minimap2-arm/archive/v0.1.tar.gz\n","! tar xvf v0.1.tar.gz && cd minimap2-arm-0.1 && make"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"sHkWpwJ0js-U"},"outputs":[],"source":["! chmod u+x /content/minimap2-arm-0.1/misc/idxtools/divide_and_index.sh"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"OSo_TWKelVjr"},"outputs":[],"source":["! apt-get install bc"]},{"cell_type":"markdown","source":["##minimap2 featherweight alignment - partioning "],"metadata":{"id":"oBPQyTcKOPDF"}},{"cell_type":"markdown","source":["The minimap2 featherweight implementation works by splitting the reference genome into four parts with minimal loss in alignment accuracy (documentation: https://doi.org/10.1038/s41598-019-40739-8). This command will only need to be run once as long as the partioned reference index is not removed from the directory. "],"metadata":{"id":"of339UK9OT_0"}},{"cell_type":"code","source":["! eval /content/minimap2-arm-0.1/misc/idxtools/divide_and_index.sh $INDEX_FILE_PATH 4 $PARTITIONED_INDEX_FILE_PATH /content/minimap2-arm-0.1/minimap2 $PLATFORM_FREE"],"metadata":{"id":"S2MVZUmwOUID"},"execution_count":null,"outputs":[]},{"cell_type":"markdown","metadata":{"id":"itX8aI_chgmy"},"source":["##minimap2 featherweight alignment - usage"]},{"cell_type":"markdown","metadata":{"id":"aY5xHM_gieeF"},"source":["The minimap2 featherweight implementation works by splitting the reference genome into four parts with minimal loss in alignment accuracy (documentation: https://doi.org/10.1038/s41598-019-40739-8)."]},{"cell_type":"code","execution_count":null,"metadata":{"id":"s9pBrxRPi_hU"},"outputs":[],"source":["! eval /content/minimap2-arm-0.1/minimap2 -a -x $PLATFORM_FREE $PARTITIONED_INDEX_FILE_PATH $PIPELINE_FILE_PATH/fastq/$ACC.fastq --multi-prefix tmp > $PIPELINE_FILE_PATH/sam/$ACC.sam"]},{"cell_type":"markdown","metadata":{"id":"dxspwFM3_1qN"},"source":["##samtools: Reading/writing/editing/indexing/viewing SAM/BAM/CRAM format - installation"]},{"cell_type":"markdown","metadata":{"id":"SXphSNhPA9di"},"source":["samtools allows for manipulation of high-throughput sequencing data (documentation: http://www.htslib.org/) \n"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"v0_UcWdBHDGQ"},"outputs":[],"source":["! conda install -c bioconda samtools -y"]},{"cell_type":"markdown","source":["##samtools: Reading/writing/editing/indexing/viewing SAM/BAM/CRAM format - usage"],"metadata":{"id":"sjns1XbAkILI"}},{"cell_type":"markdown","metadata":{"id":"KfgkJ0bJkYDv"},"source":["samtools allows for manipulation of high-throughput sequencing data (documentation: http://www.htslib.org/) \n"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"fAFTMfy2HDOs"},"outputs":[],"source":["! samtools view -S -b $PIPELINE_FILE_PATH/sam/$ACC.sam > $PIPELINE_FILE_PATH/bam/$ACC.bam "]},{"cell_type":"code","execution_count":null,"metadata":{"id":"qYOZbl8PJo-h"},"outputs":[],"source":["! samtools sort $PIPELINE_FILE_PATH/bam/$ACC.bam -o $PIPELINE_FILE_PATH/bam/$ACC.sorted.bam  "]},{"cell_type":"code","execution_count":null,"metadata":{"id":"ypjDsfP7JpM6"},"outputs":[],"source":["! samtools index $PIPELINE_FILE_PATH/bam/$ACC.sorted.bam "]},{"cell_type":"markdown","source":["##TranscriptClean: correct mismatches, microindels, and noncanonical splice junctions - installation"],"metadata":{"id":"Y9qrP4Ip3xwz"}},{"cell_type":"markdown","source":["TranscriptClean is a command-line program which corrects long-read mismatches, microindels and noncanonical splice junctions (documentation: https://github.com/mortazavilab/TranscriptClean). "],"metadata":{"id":"bB4Zr5Ef38qi"}},{"cell_type":"code","source":["! conda install -c bioconda pyfasta pyranges samtools -y"],"metadata":{"id":"oXIztTkj3zbu"},"execution_count":null,"outputs":[]},{"cell_type":"code","source":["! wget https://github.com/mortazavilab/TranscriptClean/archive/refs/tags/v2.0.3.tar.gz\n","! tar xvf v2.0.3.tar.gz "],"metadata":{"id":"BfsI59QG4BH_"},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":["##TranscriptClean: correct mismatches, microindels, and noncanonical splice junctions - usage"],"metadata":{"id":"zqQcBdxa4X7q"}},{"cell_type":"markdown","source":["TranscriptClean is a command-line program which corrects long-read mismatches, microindels and noncanonical splice junctions (documentation: https://github.com/mortazavilab/TranscriptClean). "],"metadata":{"id":"r5CNnQ5w4k_K"}},{"cell_type":"code","source":["! eval python /content/TranscriptClean-2.0.3/TranscriptClean.py --sam $PIPELINE_FILE_PATH/sam/$ACC.sam --genome $INDEX_FILE_PATH --outprefix $PIPELINE_FILE_PATH/transcriptclean/$ACC"],"metadata":{"id":"sJPb4PSU5LmU"},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":["##featureCounts: an efficient general purpose program for assigning sequence reads to genomic features - installation"],"metadata":{"id":"yhR96h7O5CHN"}},{"cell_type":"markdown","source":["featureCounts is a read summarization program suitable for counting reads generated from either RNA or genomic DNA sequencing experiments (documentation: http://subread.sourceforge.net/). "],"metadata":{"id":"VnlTGqwG5JYz"}},{"cell_type":"code","source":["! conda install -c bioconda subread -y"],"metadata":{"id":"Vd48uWN45VkA"},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":["##featureCounts: an efficient general purpose program for assigning sequence reads to genomic features - usage"],"metadata":{"id":"7yVbaSBc5cu7"}},{"cell_type":"markdown","source":["featureCounts is a read summarization program suitable for counting reads generated from either RNA or genomic DNA sequencing experiments (documentation: http://subread.sourceforge.net/). "],"metadata":{"id":"-2kbK23S5oi5"}},{"cell_type":"code","source":["! eval featureCounts -M -O -T 24 -L -a $ANNOTATION_FILE_PATH -t exon -g gene_id -o $PIPELINE_FILE_PATH/featureCounts/$ACC.txt $PIPELINE_FILE_PATH/bam/$ACC.bam "],"metadata":{"id":"1ycdSYt_5plb"},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":["##TAMA: Transcriptome Annotation by Modular Algorithms (Google Colab Pro required) - installation\n","\n"],"metadata":{"id":"Wj85WdDNwraY"}},{"cell_type":"markdown","source":["TAMA allows for the construction of a polished transcriptome by collapsing runs into individual transcripts (documentation: https://github.com/GenomeRIK/tama/wiki). "],"metadata":{"id":"bAO18hfruXPm"}},{"cell_type":"code","source":["! git clone https://github.com/MakeTheBrainHappy/tama"],"metadata":{"id":"GKv-Sz-IwsXj"},"execution_count":null,"outputs":[]},{"cell_type":"code","source":["! python -m pip install Bio"],"metadata":{"id":"FqjqBPkgkLfX"},"execution_count":null,"outputs":[]},{"cell_type":"code","source":["! pip install pysam"],"metadata":{"id":"F4vJnRJCkVhy"},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":["##TAMA: Transcriptome Annotation by Modular Algorithms (Google Colab Pro required) - usage"],"metadata":{"id":"F8ECo-DYue_P"}},{"cell_type":"markdown","source":["TAMA allows for the construction of a polished transcriptome by collapsing runs into individual transcripts (documentation: https://github.com/GenomeRIK/tama/wiki). "],"metadata":{"id":"8kgDVWffuslu"}},{"cell_type":"code","source":["! eval python tama/tama_collapse.py -s $PIPELINE_FILE_PATH/bam/$ACC.sorted.bam -f $INDEX_FILE_PATH -p $PIPELINE_FILE_PATH/tama/$ACC.polished -b BAM -rm low_mem"],"metadata":{"id":"wsqCqeIqwsbH"},"execution_count":null,"outputs":[]},{"cell_type":"markdown","metadata":{"id":"npZwllsupfEJ"},"source":["##svist4get: a simple visualization tool for genomic tracks from sequencing experiments - installation"]},{"cell_type":"markdown","source":["svist4get allows you to view read coverage at a defined region on a chromosome (documentation: https://bitbucket.org/artegorov/svist4get/src/master/)"],"metadata":{"id":"niesqrQ1nHMM"}},{"cell_type":"code","execution_count":null,"metadata":{"id":"mq3WmdTxpeft"},"outputs":[],"source":["! sudo apt-get install bedtools"]},{"cell_type":"code","source":["! apt-get update"],"metadata":{"id":"iaRNkYPEoa0T"},"execution_count":null,"outputs":[]},{"cell_type":"code","execution_count":null,"metadata":{"id":"-gNi5jxXA5oq"},"outputs":[],"source":["! apt-get install libmagickwand-dev"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"3HS96PJ45f26"},"outputs":[],"source":["! cp -r $PIPELINE_FILE_PATH/svist4get/policy_revised.xml /etc/ImageMagick-6/policy.xml"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"ZNVRJydu1wIV"},"outputs":[],"source":["! python3 -m pip install svist4get"]},{"cell_type":"markdown","source":["##svist4get: a simple visualization tool for genomic tracks from sequencing experiments - usage"],"metadata":{"id":"S9r4MCB0lvAf"}},{"cell_type":"markdown","source":["svist4get allows you to view read coverage at a defined region on a chromosome (documentation: https://bitbucket.org/artegorov/svist4get/src/master/)"],"metadata":{"id":"3T2Y3dcAnaWu"}},{"cell_type":"code","execution_count":null,"metadata":{"id":"ehHeatJQCZ42"},"outputs":[],"source":["! bedtools genomecov -ibam $PIPELINE_FILE_PATH/bam/$ACC.sorted.bam -bg > $PIPELINE_FILE_PATH/bed/$ACC.sorted.bedgraph"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"OpQMW3gI15ZI"},"outputs":[],"source":["! eval svist4get -bg $PIPELINE_FILE_PATH/bed/$ACC.sorted.bedgraph -gtf $ANNOTATION_FILE_PATH -fa $INDEX_FILE_PATH -bl Long-Read Coverage -w $CHROMOSOME $CHROMOSOME_START $CHROMOSOME_FINISH -it \"$REGION_NAME\" -o $PIPELINE_FILE_PATH/svist4get/$ACC"]},{"cell_type":"markdown","source":["##Pistis: Quality control plotting for long reads - installation"],"metadata":{"id":"5CwywN4tc74s"}},{"cell_type":"markdown","source":["Pistis generates long-read specific quality control graphs, including a plot demonstrating read alignment percentage to the reference genome (documentation: https://github.com/mbhall88/pistis)"],"metadata":{"id":"ckgZcyKcpFyC"}},{"cell_type":"code","source":["! pip3 install pistis"],"metadata":{"id":"S5mHHUr1d50q"},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":["##Pistis: Quality control plotting for long reads - usage"],"metadata":{"id":"MjZ_tYz4pEHX"}},{"cell_type":"markdown","source":["Pistis generates long-read specific quality control graphs, including a plot demonstrating read alignment percentage to the reference genome (documentation: https://github.com/mbhall88/pistis)"],"metadata":{"id":"RFhhJITCphA_"}},{"cell_type":"code","source":["! pistis -f $PIPELINE_FILE_PATH/fastq/$ACC.fastq -b $PIPELINE_FILE_PATH/bam/$ACC.sorted.bam  -o $PIPELINE_FILE_PATH/pistis/$ACC.pdf"],"metadata":{"id":"ik-boaBxeNjP"},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":["##MakeHub: Fully automated generation of UCSC assembly hubs - installation"],"metadata":{"id":"EQn9Ikb3YkFC"}},{"cell_type":"markdown","source":["MakeHub is a command line tool for the fully automatic generation of of track data hubs for visualizing genomes with the UCSC genome browser (documentation: https://github.com/Gaius-Augustus/MakeHub)."],"metadata":{"id":"X7J1qXU1bmXd"}},{"cell_type":"code","source":["! python3.7 -m pip install biopython"],"metadata":{"id":"_eYWwdsmdsBa"},"execution_count":null,"outputs":[]},{"cell_type":"code","source":["! sudo apt install samtools"],"metadata":{"id":"Fu1IcmMbdsxV"},"execution_count":null,"outputs":[]},{"cell_type":"code","source":["! sudo apt install augustus augustus-data augustus-doc"],"metadata":{"id":"yudzuOTW2IQ-"},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":["##MakeHub: Fully automated generation of UCSC assembly hubs - usage"],"metadata":{"id":"POLtVVCEKndf"}},{"cell_type":"markdown","source":["MakeHub is a command line tool for the fully automatic generation of of track data hubs for visualizing genomes with the UCSC genome browser (documentation: https://github.com/Gaius-Augustus/MakeHub)."],"metadata":{"id":"ZcW7zXOoKrCy"}},{"cell_type":"code","source":["! chmod 755 $PIPELINE_FILE_PATH/makehub/make_hub.py"],"metadata":{"id":"fMN2TYdniufR"},"execution_count":null,"outputs":[]},{"cell_type":"code","source":["! eval $PIPELINE_FILE_PATH/makehub/make_hub.py -l $HUB_KEYWORD -L $HUB_NAME -g $INDEX_FILE_PATH -e \\\n","  $HUB_EMAIL -a $ANNOTATION_FILE_PATH -b $PIPELINE_FILE_PATH/bam/$ACC.sorted.bam -o $PIPELINE_FILE_PATH/makehub/"],"metadata":{"id":"pVA2mrrjZ2yM"},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":["##MultiQC: Aggregate results from bioinformatics analyses across many samples into a single report - installation\n","\n"],"metadata":{"id":"kpfzCCTKuqMt"}},{"cell_type":"markdown","metadata":{"id":"Jluk65BgNoEJ"},"source":["MultiQC is a program which allows you to combine reports for as many samples as you wish ([documentation](https://multiqc.info/docs/))"]},{"cell_type":"code","metadata":{"id":"B6OuaTChNs4J"},"source":["! pip install multiqc"],"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":["##MultiQC: Aggregate results from bioinformatics analyses across many samples into a single report - usage"],"metadata":{"id":"m5pU9i4DvGxQ"}},{"cell_type":"markdown","source":["MultiQC is a program which allows you to combine reports for as many samples as you wish ([documentation](https://multiqc.info/docs/))"],"metadata":{"id":"sPB5udNzvVdJ"}},{"cell_type":"code","metadata":{"id":"SEBsMVQ5OG4U"},"source":["! multiqc $PIPELINE_FILE_PATH -o $PIPELINE_FILE_PATH/multiqc"],"execution_count":null,"outputs":[]}],"metadata":{"colab":{"name":"long_read_rna_seq_analysis_prebuilt_indices.ipynb","provenance":[],"collapsed_sections":["24vu4E_8vbKa","z1lysGn4hO-i","oBdBA_CDdE0z","F0hFNwmcv5qr","SQb5HTiwtfMn","gyh52FxVTSQt","1mvZC8IsijTg","oBPQyTcKOPDF","itX8aI_chgmy","dxspwFM3_1qN","sjns1XbAkILI","Y9qrP4Ip3xwz","zqQcBdxa4X7q","yhR96h7O5CHN","7yVbaSBc5cu7","Wj85WdDNwraY","F8ECo-DYue_P","npZwllsupfEJ","S9r4MCB0lvAf","oaWQESsSLnIj","f3VjubgsKGJl","m5pU9i4DvGxQ"],"machine_shape":"hm"},"kernelspec":{"display_name":"Python 3","name":"python3"},"language_info":{"name":"python"}},"nbformat":4,"nbformat_minor":0}